{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Annotate peaks for:\n",
    "* `/data/reddylab/Alex/encode4_duke/ipynbs/jamborees/20220706_fcc_capstone/data/tewhey_mpra_orientation_dep_call_all.merged.bed`\n",
    "* `/data/reddylab/Alex/encode4_duke/ipynbs/jamborees/20220706_fcc_capstone/data/lenti_mpra_orientation_dep_call_all.merged.bed` \n",
    "* `/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_enhancer_calls_5Oct/processed/merged_peak/merged_sig_up_in_each_orientation.bed` \n",
    "* `/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/K562_wgSTARR_active_cradle_peaks.txt` \n",
    "* CRISPR_HCRFlowFISH: `/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/encode_data/CRISPR/K562.merged_peaks_from_HCRFF_SabetiLab.peakwise.bed`\n",
    "* CRISPR_CERES_screen: `/data/reddylab/Alex/collab/20181203_Tyler/data/allen_order_beta_test/k562-wgCERES-discovery.dhs_allen_test_v2_df.bnd_3.fdr_0.05.hg38.bed`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"tewhey_mpra_orientation_dep_call_all\"\n",
      ">> loading peak file...\t\t\t\t 2022-11-02 11:38:30 PM \n",
      ">> preparing features information...\t\t 2022-11-02 11:38:30 PM \n",
      ">> identifying nearest features...\t\t 2022-11-02 11:38:30 PM \n",
      ">> calculating distance from peak to TSS...\t 2022-11-02 11:38:30 PM \n",
      ">> assigning genomic annotation...\t\t 2022-11-02 11:38:30 PM \n",
      ">> adding gene annotation...\t\t\t 2022-11-02 11:38:37 PM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:1 mapping between keys and columns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2022-11-02 11:38:37 PM \n",
      ">> done...\t\t\t\t\t 2022-11-02 11:38:37 PM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "4004/4004  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "              Feature  Frequency\n",
      "9    Promoter (<=1kb) 11.4885115\n",
      "10   Promoter (1-2kb)  7.2677323\n",
      "4              5' UTR  0.1248751\n",
      "3              3' UTR  1.8231768\n",
      "1            1st Exon  0.9740260\n",
      "7          Other Exon  4.9700300\n",
      "2          1st Intron  7.7672328\n",
      "8        Other Intron 36.0139860\n",
      "6  Downstream (<=300)  1.0239760\n",
      "5   Distal Intergenic 28.5464535\n",
      "[1] \"lenti_mpra_orientation_dep_call_all\"\n",
      ">> loading peak file...\t\t\t\t 2022-11-02 11:38:37 PM \n",
      ">> preparing features information...\t\t 2022-11-02 11:38:37 PM \n",
      ">> identifying nearest features...\t\t 2022-11-02 11:38:37 PM \n",
      ">> calculating distance from peak to TSS...\t 2022-11-02 11:38:38 PM \n",
      ">> assigning genomic annotation...\t\t 2022-11-02 11:38:38 PM \n",
      ">> adding gene annotation...\t\t\t 2022-11-02 11:38:43 PM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2022-11-02 11:38:44 PM \n",
      ">> done...\t\t\t\t\t 2022-11-02 11:38:44 PM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "12439/12439  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "              Feature  Frequency\n",
      "9    Promoter (<=1kb) 42.7928290\n",
      "10   Promoter (1-2kb)  1.9052979\n",
      "4              5' UTR  0.1768631\n",
      "3              3' UTR  1.2139239\n",
      "1            1st Exon  1.6962778\n",
      "7          Other Exon  1.5435324\n",
      "2          1st Intron  9.4139400\n",
      "8        Other Intron 17.1396415\n",
      "6  Downstream (<=300)  0.9003939\n",
      "5   Distal Intergenic 23.2173004\n",
      "[1] \"merged_sig_up_in_each_orientation\"\n",
      ">> loading peak file...\t\t\t\t 2022-11-02 11:38:44 PM \n",
      ">> preparing features information...\t\t 2022-11-02 11:38:47 PM \n",
      ">> identifying nearest features...\t\t 2022-11-02 11:38:47 PM \n",
      ">> calculating distance from peak to TSS...\t 2022-11-02 11:38:50 PM \n",
      ">> assigning genomic annotation...\t\t 2022-11-02 11:38:50 PM \n",
      ">> adding gene annotation...\t\t\t 2022-11-02 11:38:57 PM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2022-11-02 11:38:57 PM \n",
      ">> done...\t\t\t\t\t 2022-11-02 11:38:57 PM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "84996/84996  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "              Feature   Frequency\n",
      "9    Promoter (<=1kb) 35.00870629\n",
      "10   Promoter (1-2kb)  6.55442609\n",
      "4              5' UTR  0.07882724\n",
      "3              3' UTR  1.50477670\n",
      "1            1st Exon  1.31888559\n",
      "7          Other Exon  1.63537108\n",
      "2          1st Intron 12.34175726\n",
      "8        Other Intron 19.53856652\n",
      "6  Downstream (<=300)  0.96592781\n",
      "5   Distal Intergenic 21.05275542\n",
      "[1] \"K562_wgSTARR_active_cradle_peaks\"\n",
      ">> loading peak file...\t\t\t\t 2022-11-02 11:38:59 PM \n",
      ">> preparing features information...\t\t 2022-11-02 11:38:59 PM \n",
      ">> identifying nearest features...\t\t 2022-11-02 11:38:59 PM \n",
      ">> calculating distance from peak to TSS...\t 2022-11-02 11:39:00 PM \n",
      ">> assigning genomic annotation...\t\t 2022-11-02 11:39:00 PM \n",
      ">> adding gene annotation...\t\t\t 2022-11-02 11:39:09 PM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2022-11-02 11:39:09 PM \n",
      ">> done...\t\t\t\t\t 2022-11-02 11:39:09 PM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "19054/19054  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "              Feature  Frequency\n",
      "9    Promoter (<=1kb) 23.2549596\n",
      "10   Promoter (1-2kb)  4.3875302\n",
      "4              5' UTR  0.1417025\n",
      "3              3' UTR  1.5062454\n",
      "1            1st Exon  1.5167419\n",
      "7          Other Exon  2.5926315\n",
      "2          1st Intron 11.4254225\n",
      "8        Other Intron 23.9372310\n",
      "6  Downstream (<=300)  0.9604283\n",
      "5   Distal Intergenic 30.2771072\n",
      "[1] \"K562_merged_peaks_from_HCRFF_SabetiLab\"\n",
      ">> loading peak file...\t\t\t\t 2022-11-02 11:39:09 PM \n",
      ">> preparing features information...\t\t 2022-11-02 11:39:09 PM \n",
      ">> identifying nearest features...\t\t 2022-11-02 11:39:09 PM \n",
      ">> calculating distance from peak to TSS...\t 2022-11-02 11:39:10 PM \n",
      ">> assigning genomic annotation...\t\t 2022-11-02 11:39:10 PM \n",
      ">> adding gene annotation...\t\t\t 2022-11-02 11:39:16 PM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:1 mapping between keys and columns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2022-11-02 11:39:16 PM \n",
      ">> done...\t\t\t\t\t 2022-11-02 11:39:16 PM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "98/98  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "             Feature Frequency\n",
      "8   Promoter (<=1kb) 37.755102\n",
      "9   Promoter (1-2kb)  5.102041\n",
      "3             5' UTR  3.061224\n",
      "2             3' UTR  1.020408\n",
      "6         Other Exon  4.081633\n",
      "1         1st Intron 19.387755\n",
      "7       Other Intron 18.367347\n",
      "5 Downstream (<=300)  2.040816\n",
      "4  Distal Intergenic  9.183673\n",
      "[1] \"k562-wgCERES-discovery\"\n",
      ">> loading peak file...\t\t\t\t 2022-11-02 11:39:16 PM \n",
      ">> preparing features information...\t\t 2022-11-02 11:39:16 PM \n",
      ">> identifying nearest features...\t\t 2022-11-02 11:39:16 PM \n",
      ">> calculating distance from peak to TSS...\t 2022-11-02 11:39:17 PM \n",
      ">> assigning genomic annotation...\t\t 2022-11-02 11:39:17 PM \n",
      ">> adding gene annotation...\t\t\t 2022-11-02 11:39:23 PM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2022-11-02 11:39:23 PM \n",
      ">> done...\t\t\t\t\t 2022-11-02 11:39:23 PM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "7768/7770  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "              Feature  Frequency\n",
      "9    Promoter (<=1kb) 36.9078270\n",
      "10   Promoter (1-2kb)  6.6297631\n",
      "4              5' UTR  0.2703399\n",
      "3              3' UTR  2.7162719\n",
      "1            1st Exon  2.0854789\n",
      "7          Other Exon  3.4371782\n",
      "2          1st Intron  9.9253347\n",
      "8        Other Intron 17.5463440\n",
      "6  Downstream (<=300)  1.6220391\n",
      "5   Distal Intergenic 18.8594233\n"
     ]
    }
   ],
   "source": [
    "suppressMessages(suppressWarnings(library(TxDb.Hsapiens.UCSC.hg38.knownGene)))\n",
    "suppressMessages(suppressWarnings(library(ChIPseeker)))\n",
    "\n",
    "txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene\n",
    "\n",
    "\n",
    "set_plot_dimensions <- function(width_choice, height_choice) {\n",
    "        options(repr.plot.width=width_choice, repr.plot.height=height_choice)\n",
    "        }\n",
    "\n",
    "files <- list(\"/data/reddylab/Alex/encode4_duke/ipynbs/jamborees/20220706_fcc_capstone/data/tewhey_mpra_orientation_dep_call_all.merged.bed\",\n",
    "              \"/data/reddylab/Alex/encode4_duke/ipynbs/jamborees/20220706_fcc_capstone/data/lenti_mpra_orientation_dep_call_all.merged.bed\",\n",
    "              \"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_enhancer_calls_5Oct/processed/merged_peak/merged_sig_up_in_each_orientation.bed\",\n",
    "              \"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/K562_wgSTARR_active_cradle_peaks.txt\",\n",
    "              \"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/encode_data/CRISPR/K562_merged_peaks_from_HCRFF_SabetiLab.cut.peakwise.bed\",\n",
    "              \"/data/reddylab/Alex/collab/20181203_Tyler/data/allen_order_beta_test/k562-wgCERES-discovery.dhs_allen_test_v2_df.bnd_3.fdr_0.05.hg38.bed\")\n",
    "\n",
    "for (file in files){\n",
    "    filename <- sub('\\\\..*$', '', basename(file))\n",
    "    print(filename)\n",
    "    peakAnnoEnrich <- annotatePeak(file, tssRegion=c(-2000, 2000),\n",
    "                                TxDb=txdb, annoDb=\"org.Hs.eg.db\")\n",
    "    print(peakAnnoEnrich)\n",
    "    foo <- as.data.frame(peakAnnoEnrich)\n",
    "    write.table(foo, \n",
    "             file=paste0('/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/enhancer_call_annotation/',filename, '.annotated.txt'), \n",
    "            quote=F, sep='\\t')\n",
    "    \n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Promoter/Non-promoter annotations for the random 5k hits used for the heatmaps"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "data: \n",
    "\n",
    "`/data/reddylab/Alex/encode4_duke/ipynbs/jamborees/20220706_fcc_capstone/data/tewhey_mpra_orientation_dep_call_all.merged.bed`\n",
    "        `/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/lenti_mpra_orientation_dep_call_all.merged.random5000.bed`\n",
    "        `/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_ASTARR.30bp_bin.merged_sig_up_in_each_orientation.random5000.bed`\n",
    "        `/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/K562_wgSTARR_active_cradle_peaks.random5000.bed `\n",
    "       `/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/k562_wgCERES_dhsHits_all_HCRFF_Sabeti_Lab.random5000.bed`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"lenti_mpra_orientation_dep_call_all\"\n",
      ">> loading peak file...\t\t\t\t 2022-11-09 01:54:21 PM \n",
      ">> preparing features information...\t\t 2022-11-09 01:54:21 PM \n",
      ">> identifying nearest features...\t\t 2022-11-09 01:54:25 PM \n",
      ">> calculating distance from peak to TSS...\t 2022-11-09 01:54:27 PM \n",
      ">> assigning genomic annotation...\t\t 2022-11-09 01:54:27 PM \n",
      ">> adding gene annotation...\t\t\t 2022-11-09 01:55:13 PM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: org.Hs.eg.db\n",
      "\n",
      "'select()' returned 1:many mapping between keys and columns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2022-11-09 01:55:14 PM \n",
      ">> done...\t\t\t\t\t 2022-11-09 01:55:14 PM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "5000/5000  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "              Feature Frequency\n",
      "9    Promoter (<=1kb)     42.72\n",
      "10   Promoter (1-2kb)      1.96\n",
      "4              5' UTR      0.22\n",
      "3              3' UTR      1.30\n",
      "1            1st Exon      1.76\n",
      "7          Other Exon      1.26\n",
      "2          1st Intron      9.48\n",
      "8        Other Intron     16.94\n",
      "6  Downstream (<=300)      0.86\n",
      "5   Distal Intergenic     23.50\n",
      "[1] \"Junke_ASTARR\"\n",
      ">> loading peak file...\t\t\t\t 2022-11-09 01:55:14 PM \n",
      ">> preparing features information...\t\t 2022-11-09 01:55:15 PM \n",
      ">> identifying nearest features...\t\t 2022-11-09 01:55:15 PM \n",
      ">> calculating distance from peak to TSS...\t 2022-11-09 01:55:15 PM \n",
      ">> assigning genomic annotation...\t\t 2022-11-09 01:55:15 PM \n",
      ">> adding gene annotation...\t\t\t 2022-11-09 01:55:22 PM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2022-11-09 01:55:22 PM \n",
      ">> done...\t\t\t\t\t 2022-11-09 01:55:22 PM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "5000/5000  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "              Feature Frequency\n",
      "9    Promoter (<=1kb)     34.88\n",
      "10   Promoter (1-2kb)      6.82\n",
      "4              5' UTR      0.10\n",
      "3              3' UTR      1.54\n",
      "1            1st Exon      1.10\n",
      "7          Other Exon      1.70\n",
      "2          1st Intron     11.86\n",
      "8        Other Intron     19.16\n",
      "6  Downstream (<=300)      0.92\n",
      "5   Distal Intergenic     21.92\n",
      "[1] \"K562_wgSTARR_active_cradle_peaks\"\n",
      ">> loading peak file...\t\t\t\t 2022-11-09 01:55:22 PM \n",
      ">> preparing features information...\t\t 2022-11-09 01:55:22 PM \n",
      ">> identifying nearest features...\t\t 2022-11-09 01:55:22 PM \n",
      ">> calculating distance from peak to TSS...\t 2022-11-09 01:55:23 PM \n",
      ">> assigning genomic annotation...\t\t 2022-11-09 01:55:23 PM \n",
      ">> adding gene annotation...\t\t\t 2022-11-09 01:55:29 PM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2022-11-09 01:55:29 PM \n",
      ">> done...\t\t\t\t\t 2022-11-09 01:55:30 PM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "5000/5000  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "              Feature Frequency\n",
      "9    Promoter (<=1kb)     23.70\n",
      "10   Promoter (1-2kb)      4.24\n",
      "4              5' UTR      0.16\n",
      "3              3' UTR      1.50\n",
      "1            1st Exon      1.50\n",
      "7          Other Exon      2.50\n",
      "2          1st Intron     11.60\n",
      "8        Other Intron     23.42\n",
      "6  Downstream (<=300)      0.90\n",
      "5   Distal Intergenic     30.48\n",
      "[1] \"k562_wgCERES_dhsHits_all_HCRFF_Sabeti_Lab\"\n",
      ">> loading peak file...\t\t\t\t 2022-11-09 01:55:30 PM \n",
      ">> preparing features information...\t\t 2022-11-09 01:55:30 PM \n",
      ">> identifying nearest features...\t\t 2022-11-09 01:55:30 PM \n",
      ">> calculating distance from peak to TSS...\t 2022-11-09 01:55:31 PM \n",
      ">> assigning genomic annotation...\t\t 2022-11-09 01:55:31 PM \n",
      ">> adding gene annotation...\t\t\t 2022-11-09 01:55:37 PM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2022-11-09 01:55:37 PM \n",
      ">> done...\t\t\t\t\t 2022-11-09 01:55:37 PM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "4998/5000  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "              Feature  Frequency\n",
      "9    Promoter (<=1kb) 37.0548219\n",
      "10   Promoter (1-2kb)  6.6626651\n",
      "4              5' UTR  0.3601441\n",
      "3              3' UTR  2.6610644\n",
      "1            1st Exon  2.1208483\n",
      "7          Other Exon  3.0212085\n",
      "2          1st Intron 10.4241697\n",
      "8        Other Intron 18.0272109\n",
      "6  Downstream (<=300)  1.5006002\n",
      "5   Distal Intergenic 18.1672669\n"
     ]
    }
   ],
   "source": [
    "suppressMessages(suppressWarnings(library(TxDb.Hsapiens.UCSC.hg38.knownGene)))\n",
    "suppressMessages(suppressWarnings(library(ChIPseeker)))\n",
    "\n",
    "txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene\n",
    "\n",
    "\n",
    "set_plot_dimensions <- function(width_choice, height_choice) {\n",
    "        options(repr.plot.width=width_choice, repr.plot.height=height_choice)\n",
    "        }\n",
    "\n",
    "files <- list(\n",
    "              \"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/lenti_mpra_orientation_dep_call_all.merged.random5000.bed\",\n",
    "              \"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_ASTARR.30bp_bin.merged_sig_up_in_each_orientation.random5000.bed\",\n",
    "              \"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/K562_wgSTARR_active_cradle_peaks.random5000.bed\",\n",
    "              \"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/k562_wgCERES_dhsHits_all_HCRFF_Sabeti_Lab.random5000.bed\"\n",
    ")\n",
    "\n",
    "for (file in files){\n",
    "    filename <- sub('\\\\..*$', '', basename(file))\n",
    "    print(filename)\n",
    "    peakAnnoEnrich <- annotatePeak(file, tssRegion=c(-2000, 2000),\n",
    "                                TxDb=txdb, annoDb=\"org.Hs.eg.db\")\n",
    "    print(peakAnnoEnrich)\n",
    "    foo <- as.data.frame(peakAnnoEnrich)\n",
    "    write.table(foo, \n",
    "             file=paste0('/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/enhancer_call_annotation/',filename, '.random5k.annotated.txt'), \n",
    "            quote=F, sep='\\t')\n",
    "    \n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Annotations for active_vs_inactive peaks \n",
    "\n",
    "DATA:\n",
    "* LentiMPRA:\n",
    "`/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/lenti_mpra_orientation_dep_call_all.merged.random5000.bed`\n",
    "`/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/New_enhancer_call/lentiMPRA/lenti_mpra_tested_element_all.unique_peaks.random5k.bed`\n",
    "        \n",
    "\n",
    "* ASTARR:\n",
    "`/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_ASTARR.30bp_bin.merged_sig_up_in_each_orientation.random5000.bed`\n",
    "`/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_enhancer_calls_5Oct/processed/bin_level/inactive_bins_with_raw_count.merged.random5k.bed`\n",
    "        \n",
    "\n",
    "* wgSTARR:\n",
    "`/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_wgSTARR.30bp_10bp_bin.sig_up.random5000.bed`\n",
    "`/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_enhancer_calls_5Oct/wgSTARR/processed/bin_level/inactive_bins_with_raw_count.merged.random5000.bed`\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"lenti_mpra_orientation_dep_call_all\"\n",
      ">> loading peak file...\t\t\t\t 2023-02-14 01:04:47 AM \n",
      ">> preparing features information...\t\t 2023-02-14 01:04:47 AM \n",
      ">> identifying nearest features...\t\t 2023-02-14 01:04:47 AM \n",
      ">> calculating distance from peak to TSS...\t 2023-02-14 01:04:48 AM \n",
      ">> assigning genomic annotation...\t\t 2023-02-14 01:04:48 AM \n",
      ">> adding gene annotation...\t\t\t 2023-02-14 01:04:55 AM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2023-02-14 01:04:56 AM \n",
      ">> done...\t\t\t\t\t 2023-02-14 01:04:56 AM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "5000/5000  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "             Feature Frequency\n",
      "9           Promoter     42.72\n",
      "4             5' UTR      0.22\n",
      "3             3' UTR      1.36\n",
      "1           1st Exon      1.78\n",
      "7         Other Exon      1.38\n",
      "2         1st Intron      9.96\n",
      "8       Other Intron     18.04\n",
      "6 Downstream (<=300)      0.90\n",
      "5  Distal Intergenic     23.64\n",
      "[1] \"lenti_mpra_tested_element_all\"\n",
      ">> loading peak file...\t\t\t\t 2023-02-14 01:04:56 AM \n",
      ">> preparing features information...\t\t 2023-02-14 01:04:56 AM \n",
      ">> identifying nearest features...\t\t 2023-02-14 01:04:56 AM \n",
      ">> calculating distance from peak to TSS...\t 2023-02-14 01:04:58 AM \n",
      ">> assigning genomic annotation...\t\t 2023-02-14 01:04:58 AM \n",
      ">> adding gene annotation...\t\t\t 2023-02-14 01:05:04 AM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2023-02-14 01:05:04 AM \n",
      ">> done...\t\t\t\t\t 2023-02-14 01:05:04 AM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "5000/5000  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "             Feature Frequency\n",
      "9           Promoter     18.20\n",
      "4             5' UTR      0.28\n",
      "3             3' UTR      2.22\n",
      "1           1st Exon      1.76\n",
      "7         Other Exon      3.34\n",
      "2         1st Intron     14.16\n",
      "8       Other Intron     29.22\n",
      "6 Downstream (<=300)      1.18\n",
      "5  Distal Intergenic     29.64\n",
      "[1] \"Junke_ASTARR\"\n",
      ">> loading peak file...\t\t\t\t 2023-02-14 01:05:04 AM \n",
      ">> preparing features information...\t\t 2023-02-14 01:05:04 AM \n",
      ">> identifying nearest features...\t\t 2023-02-14 01:05:04 AM \n",
      ">> calculating distance from peak to TSS...\t 2023-02-14 01:05:05 AM \n",
      ">> assigning genomic annotation...\t\t 2023-02-14 01:05:05 AM \n",
      ">> adding gene annotation...\t\t\t 2023-02-14 01:05:11 AM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2023-02-14 01:05:11 AM \n",
      ">> done...\t\t\t\t\t 2023-02-14 01:05:11 AM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "5000/5000  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "             Feature Frequency\n",
      "9           Promoter     34.88\n",
      "4             5' UTR      0.16\n",
      "3             3' UTR      1.78\n",
      "1           1st Exon      1.42\n",
      "7         Other Exon      2.14\n",
      "2         1st Intron     14.56\n",
      "8       Other Intron     21.08\n",
      "6 Downstream (<=300)      1.10\n",
      "5  Distal Intergenic     22.88\n",
      "[1] \"inactive_bins_with_raw_count\"\n",
      ">> loading peak file...\t\t\t\t 2023-02-14 01:05:12 AM \n",
      ">> preparing features information...\t\t 2023-02-14 01:05:12 AM \n",
      ">> identifying nearest features...\t\t 2023-02-14 01:05:12 AM \n",
      ">> calculating distance from peak to TSS...\t 2023-02-14 01:05:12 AM \n",
      ">> assigning genomic annotation...\t\t 2023-02-14 01:05:12 AM \n",
      ">> adding gene annotation...\t\t\t 2023-02-14 01:05:19 AM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2023-02-14 01:05:19 AM \n",
      ">> done...\t\t\t\t\t 2023-02-14 01:05:19 AM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "4987/5000  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "             Feature  Frequency\n",
      "9           Promoter 40.9464608\n",
      "4             5' UTR  0.3208342\n",
      "3             3' UTR  3.1080810\n",
      "1           1st Exon  1.6242230\n",
      "7         Other Exon  3.7698015\n",
      "2         1st Intron 12.6127933\n",
      "8       Other Intron 16.4828554\n",
      "6 Downstream (<=300)  1.5841187\n",
      "5  Distal Intergenic 19.5508322\n",
      "[1] \"Junke_wgSTARR\"\n",
      ">> loading peak file...\t\t\t\t 2023-02-14 01:05:19 AM \n",
      ">> preparing features information...\t\t 2023-02-14 01:05:19 AM \n",
      ">> identifying nearest features...\t\t 2023-02-14 01:05:19 AM \n",
      ">> calculating distance from peak to TSS...\t 2023-02-14 01:05:20 AM \n",
      ">> assigning genomic annotation...\t\t 2023-02-14 01:05:20 AM \n",
      ">> adding gene annotation...\t\t\t 2023-02-14 01:05:27 AM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2023-02-14 01:05:27 AM \n",
      ">> done...\t\t\t\t\t 2023-02-14 01:05:27 AM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "5000/5000  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "             Feature Frequency\n",
      "9           Promoter     28.70\n",
      "4             5' UTR      0.34\n",
      "3             3' UTR      2.04\n",
      "1           1st Exon      1.64\n",
      "7         Other Exon      3.42\n",
      "2         1st Intron     12.24\n",
      "8       Other Intron     23.60\n",
      "6 Downstream (<=300)      0.86\n",
      "5  Distal Intergenic     27.16\n",
      "[1] \"wgSTARR_inactive_bins_with_raw_count\"\n",
      ">> loading peak file...\t\t\t\t 2023-02-14 01:05:27 AM \n",
      ">> preparing features information...\t\t 2023-02-14 01:05:27 AM \n",
      ">> identifying nearest features...\t\t 2023-02-14 01:05:27 AM \n",
      ">> calculating distance from peak to TSS...\t 2023-02-14 01:05:28 AM \n",
      ">> assigning genomic annotation...\t\t 2023-02-14 01:05:28 AM \n",
      ">> adding gene annotation...\t\t\t 2023-02-14 01:05:34 AM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2023-02-14 01:05:35 AM \n",
      ">> done...\t\t\t\t\t 2023-02-14 01:05:35 AM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "5000/5000  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "             Feature Frequency\n",
      "9           Promoter      5.46\n",
      "4             5' UTR      0.26\n",
      "3             3' UTR      1.86\n",
      "1           1st Exon      0.86\n",
      "7         Other Exon      2.84\n",
      "2         1st Intron     15.04\n",
      "8       Other Intron     34.76\n",
      "6 Downstream (<=300)      0.44\n",
      "5  Distal Intergenic     38.48\n"
     ]
    }
   ],
   "source": [
    "suppressMessages(suppressWarnings(library(TxDb.Hsapiens.UCSC.hg38.knownGene)))\n",
    "suppressMessages(suppressWarnings(library(ChIPseeker)))\n",
    "\n",
    "txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene\n",
    "\n",
    "\n",
    "set_plot_dimensions <- function(width_choice, height_choice) {\n",
    "        options(repr.plot.width=width_choice, repr.plot.height=height_choice)\n",
    "        }\n",
    "\n",
    "files <- list(\n",
    "              \"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/lenti_mpra_orientation_dep_call_all.merged.random5000.bed\",\n",
    "              \"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/New_enhancer_call/lentiMPRA/lenti_mpra_tested_element_all.unique_peaks.random5k.bed\",\n",
    "              \"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_ASTARR.30bp_bin.merged_sig_up_in_each_orientation.random5000.bed\", \n",
    "              \"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_enhancer_calls_5Oct/processed/bin_level/inactive_bins_with_raw_count.merged.random5k.bed\",\n",
    "              \"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_wgSTARR.30bp_10bp_bin.sig_up.random5000.bed\",\n",
    "              \"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_enhancer_calls_5Oct/wgSTARR/processed/bin_level/wgSTARR_inactive_bins_with_raw_count.merged.random5000.bed\"\n",
    ")\n",
    "\n",
    "for (file in files){\n",
    "    filename <- sub('\\\\..*$', '', basename(file))\n",
    "    print(filename)\n",
    "    peakAnnoEnrich <- annotatePeak(file, tssRegion=c(-1000, 1000),\n",
    "                                TxDb=txdb, annoDb=\"org.Hs.eg.db\")\n",
    "    print(peakAnnoEnrich)\n",
    "    foo <- as.data.frame(peakAnnoEnrich)\n",
    "    write.table(foo, \n",
    "             file=paste0('/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/enhancer_call_annotation/new/',filename, '.random5k.annotated.txt'), \n",
    "            quote=F, sep='\\t')\n",
    "    \n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"lentiMPRA_merged_enhancer_peak_orientation_independent.summits.random5000\"\n",
      ">> loading peak file...\t\t\t\t 2023-05-03 10:00:23 PM \n",
      ">> preparing features information...\t\t 2023-05-03 10:00:23 PM \n",
      ">> identifying nearest features...\t\t 2023-05-03 10:00:25 PM \n",
      ">> calculating distance from peak to TSS...\t 2023-05-03 10:00:27 PM \n",
      ">> assigning genomic annotation...\t\t 2023-05-03 10:00:27 PM \n",
      ">> adding gene annotation...\t\t\t 2023-05-03 10:01:11 PM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: org.Hs.eg.db\n",
      "\n",
      "'select()' returned 1:many mapping between keys and columns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2023-05-03 10:01:12 PM \n",
      ">> done...\t\t\t\t\t 2023-05-03 10:01:12 PM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "5000/5000  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "             Feature Frequency\n",
      "9           Promoter     31.82\n",
      "4             5' UTR      0.20\n",
      "3             3' UTR      1.84\n",
      "1           1st Exon      1.50\n",
      "7         Other Exon      2.14\n",
      "2         1st Intron     12.28\n",
      "8       Other Intron     22.48\n",
      "6 Downstream (<=300)      0.88\n",
      "5  Distal Intergenic     26.86\n",
      "[1] \"ASTARR_wgSTARR_LentiMPRA_100%_intersected.100bp_bins.logFC_filtered.with_ATAC_peaks.merged.random5000\"\n",
      ">> loading peak file...\t\t\t\t 2023-05-03 10:01:12 PM \n",
      ">> preparing features information...\t\t 2023-05-03 10:01:12 PM \n",
      ">> identifying nearest features...\t\t 2023-05-03 10:01:12 PM \n",
      ">> calculating distance from peak to TSS...\t 2023-05-03 10:01:13 PM \n",
      ">> assigning genomic annotation...\t\t 2023-05-03 10:01:13 PM \n",
      ">> adding gene annotation...\t\t\t 2023-05-03 10:01:20 PM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2023-05-03 10:01:20 PM \n",
      ">> done...\t\t\t\t\t 2023-05-03 10:01:20 PM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "5004/5004  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "             Feature  Frequency\n",
      "9           Promoter 18.7649880\n",
      "4             5' UTR  0.2398082\n",
      "3             3' UTR  3.2773781\n",
      "1           1st Exon  2.2581934\n",
      "7         Other Exon  3.3972822\n",
      "2         1st Intron 16.5067946\n",
      "8       Other Intron 24.0007994\n",
      "6 Downstream (<=300)  1.8185452\n",
      "5  Distal Intergenic 29.7362110\n"
     ]
    }
   ],
   "source": [
    "suppressMessages(suppressWarnings(library(TxDb.Hsapiens.UCSC.hg38.knownGene)))\n",
    "suppressMessages(suppressWarnings(library(ChIPseeker)))\n",
    "\n",
    "txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene\n",
    "\n",
    "\n",
    "set_plot_dimensions <- function(width_choice, height_choice) {\n",
    "        options(repr.plot.width=width_choice, repr.plot.height=height_choice)\n",
    "        }\n",
    "\n",
    "files <- list(\n",
    "#               \"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_enhancer_calls_5Oct/z_score_processed/ASTARR/merged_peak/ASTARR_merged_enhancer_peak_orientation_independent.summits.bed\",\n",
    "#               \"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_enhancer_calls_5Oct/z_score_processed/wgSTARR/merged_peak/merged_enhancer_peak_orientation_independent.summits.random5000.bed\",\n",
    "                \"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_enhancer_calls_5Oct/z_score_processed/lentiMPRA/merged_peak/lentiMPRA_merged_enhancer_peak_orientation_independent.summits.random5000.bed\",\n",
    "                \"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/inputs/intersections/ASTARR_wgSTARR_LentiMPRA_100%_intersected.100bp_bins.logFC_filtered.with_ATAC_peaks.merged.random5000.bed\"\n",
    ")\n",
    "\n",
    "for (file in files){\n",
    "    filename <- sub('[.][^.]+$', '', basename(file))\n",
    "    print(filename)\n",
    "    peakAnnoEnrich <- annotatePeak(file, tssRegion=c(-1000, 1000),\n",
    "                                TxDb=txdb, annoDb=\"org.Hs.eg.db\")\n",
    "    print(peakAnnoEnrich)\n",
    "    foo <- as.data.frame(peakAnnoEnrich)\n",
    "    write.table(foo, \n",
    "             file=paste0('/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/enhancer_call_annotation/new/',filename, '.annotated.txt'), \n",
    "            quote=F, sep='\\t')\n",
    "    \n",
    "}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Annotate all ASTARR, LentiMPRA, wgSTARR and common background peaks into promoter and distal -- V5"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "   `/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_enhancer_calls_5Oct/z_score_processed/ASTARR/merged_peak/merged_enhancer_peaks_in_each_orientation.summits.random5000.bed`\n",
    "   `/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_enhancer_calls_5Oct/z_score_processed/lentiMPRA/merged_peak/merged_enhancer_peaks_in_either_orientation.summits.random5000.bed`\n",
    "   `/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_enhancer_calls_5Oct/z_score_processed/wgSTARR/merged_peak/merged_enhancer_peaks_in_each_orientation.summits.random5000.bed`\n",
    "   `/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/inputs/intersections/ASTARR_wgSTARR_LentiMPRA_100%_intersected.100bp_bins.logFC_filtered.with_ATAC_peaks.merged.random5000.bed`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"ASTARR_merged_enhancer_peaks_in_each_orientation.summits.random5000\"\n",
      ">> loading peak file...\t\t\t\t 2023-06-13 12:03:05 PM \n",
      ">> preparing features information...\t\t 2023-06-13 12:03:05 PM \n",
      ">> identifying nearest features...\t\t 2023-06-13 12:03:05 PM \n",
      ">> calculating distance from peak to TSS...\t 2023-06-13 12:03:06 PM \n",
      ">> assigning genomic annotation...\t\t 2023-06-13 12:03:06 PM \n",
      ">> adding gene annotation...\t\t\t 2023-06-13 12:03:12 PM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2023-06-13 12:03:12 PM \n",
      ">> done...\t\t\t\t\t 2023-06-13 12:03:12 PM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "5000/5000  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "             Feature Frequency\n",
      "9           Promoter     28.80\n",
      "4             5' UTR      0.14\n",
      "3             3' UTR      1.28\n",
      "1           1st Exon      1.48\n",
      "7         Other Exon      1.90\n",
      "2         1st Intron     15.90\n",
      "8       Other Intron     24.72\n",
      "6 Downstream (<=300)      0.80\n",
      "5  Distal Intergenic     24.98\n",
      "[1] \"lentiMPRA_merged_enhancer_peaks_in_either_orientation.summits.random5000\"\n",
      ">> loading peak file...\t\t\t\t 2023-06-13 12:03:12 PM \n",
      ">> preparing features information...\t\t 2023-06-13 12:03:12 PM \n",
      ">> identifying nearest features...\t\t 2023-06-13 12:03:12 PM \n",
      ">> calculating distance from peak to TSS...\t 2023-06-13 12:03:13 PM \n",
      ">> assigning genomic annotation...\t\t 2023-06-13 12:03:13 PM \n",
      ">> adding gene annotation...\t\t\t 2023-06-13 12:03:20 PM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2023-06-13 12:03:20 PM \n",
      ">> done...\t\t\t\t\t 2023-06-13 12:03:20 PM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "5000/5000  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "             Feature Frequency\n",
      "9           Promoter     26.02\n",
      "4             5' UTR      0.28\n",
      "3             3' UTR      2.04\n",
      "1           1st Exon      1.68\n",
      "7         Other Exon      3.08\n",
      "2         1st Intron     12.86\n",
      "8       Other Intron     24.56\n",
      "6 Downstream (<=300)      1.00\n",
      "5  Distal Intergenic     28.48\n",
      "[1] \"wgSTARR_merged_enhancer_peaks_in_each_orientation.summits.random5000\"\n",
      ">> loading peak file...\t\t\t\t 2023-06-13 12:03:20 PM \n",
      ">> preparing features information...\t\t 2023-06-13 12:03:20 PM \n",
      ">> identifying nearest features...\t\t 2023-06-13 12:03:20 PM \n",
      ">> calculating distance from peak to TSS...\t 2023-06-13 12:03:21 PM \n",
      ">> assigning genomic annotation...\t\t 2023-06-13 12:03:21 PM \n",
      ">> adding gene annotation...\t\t\t 2023-06-13 12:03:27 PM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2023-06-13 12:03:28 PM \n",
      ">> done...\t\t\t\t\t 2023-06-13 12:03:28 PM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "5000/5000  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "             Feature Frequency\n",
      "9           Promoter     25.54\n",
      "4             5' UTR      0.12\n",
      "3             3' UTR      1.24\n",
      "1           1st Exon      1.58\n",
      "7         Other Exon      2.12\n",
      "2         1st Intron     11.78\n",
      "8       Other Intron     24.10\n",
      "6 Downstream (<=300)      0.92\n",
      "5  Distal Intergenic     32.60\n",
      "[1] \"ASTARR_wgSTARR_LentiMPRA_100%_intersected.100bp_bins.logFC_filtered.with_ATAC_peaks.merged.random5000\"\n",
      ">> loading peak file...\t\t\t\t 2023-06-13 12:03:28 PM \n",
      ">> preparing features information...\t\t 2023-06-13 12:03:28 PM \n",
      ">> identifying nearest features...\t\t 2023-06-13 12:03:28 PM \n",
      ">> calculating distance from peak to TSS...\t 2023-06-13 12:03:29 PM \n",
      ">> assigning genomic annotation...\t\t 2023-06-13 12:03:29 PM \n",
      ">> adding gene annotation...\t\t\t 2023-06-13 12:03:36 PM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2023-06-13 12:03:36 PM \n",
      ">> done...\t\t\t\t\t 2023-06-13 12:03:36 PM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "4999/4999  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "             Feature Frequency\n",
      "9           Promoter 18.903781\n",
      "4             5' UTR  0.440088\n",
      "3             3' UTR  3.380676\n",
      "1           1st Exon  2.360472\n",
      "7         Other Exon  3.380676\n",
      "2         1st Intron 15.983197\n",
      "8       Other Intron 24.384877\n",
      "6 Downstream (<=300)  1.980396\n",
      "5  Distal Intergenic 29.185837\n"
     ]
    }
   ],
   "source": [
    "suppressMessages(suppressWarnings(library(TxDb.Hsapiens.UCSC.hg38.knownGene)))\n",
    "suppressMessages(suppressWarnings(library(ChIPseeker)))\n",
    "\n",
    "txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene\n",
    "\n",
    "\n",
    "set_plot_dimensions <- function(width_choice, height_choice) {\n",
    "        options(repr.plot.width=width_choice, repr.plot.height=height_choice)\n",
    "        }\n",
    "\n",
    "files <- list(\"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_enhancer_calls_5Oct/z_score_processed/ASTARR/merged_peak/ASTARR_merged_enhancer_peaks_in_each_orientation.summits.random5000.bed\",\n",
    "              \"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_enhancer_calls_5Oct/z_score_processed/lentiMPRA/merged_peak/lentiMPRA_merged_enhancer_peaks_in_either_orientation.summits.random5000.bed\",\n",
    "              \"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_enhancer_calls_5Oct/z_score_processed/wgSTARR/merged_peak/wgSTARR_merged_enhancer_peaks_in_each_orientation.summits.random5000.bed\",\n",
    "              \"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/inputs/intersections/ASTARR_wgSTARR_LentiMPRA_100%_intersected.100bp_bins.logFC_filtered.with_ATAC_peaks.merged.random5000.bed\"\n",
    ")\n",
    "\n",
    "for (file in files){\n",
    "    filename <- sub('[.][^.]+$', '', basename(file))\n",
    "    print(filename)\n",
    "    peakAnnoEnrich <- annotatePeak(file, tssRegion=c(-1000, 1000),\n",
    "                                TxDb=txdb, annoDb=\"org.Hs.eg.db\")\n",
    "    print(peakAnnoEnrich)\n",
    "    foo <- as.data.frame(peakAnnoEnrich)\n",
    "    write.table(foo, \n",
    "             file=paste0('/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/enhancer_call_annotation/V2/',filename, '.annotated.txt'), \n",
    "            quote=F, sep='\\t')\n",
    "    \n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Annotate ASTARR, LentiMPRA, wgSTARR specific inactive/background peaks into promoter and distal -- V6"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_enhancer_calls_5Oct/z_score_processed/ASTARR/bin_level/ASTARR_bin_level_all_result.inactive.merged.random5000.bed`\n",
    "`/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_enhancer_calls_5Oct/z_score_processed/lentiMPRA/element_level/lentiMPRA_element_level_all_result.inactive.merged.random5000.bed` \n",
    "`/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_enhancer_calls_5Oct/z_score_processed/wgSTARR/bin_level/wgSTARR_bin_level_all_result.inactive.merged.random5000.bed`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"ASTARR_bin_level_all_result.inactive.merged.random5000\"\n",
      ">> loading peak file...\t\t\t\t 2023-06-22 03:41:03 PM \n",
      ">> preparing features information...\t\t 2023-06-22 03:41:03 PM \n",
      ">> identifying nearest features...\t\t 2023-06-22 03:41:07 PM \n",
      ">> calculating distance from peak to TSS...\t 2023-06-22 03:41:08 PM \n",
      ">> assigning genomic annotation...\t\t 2023-06-22 03:41:08 PM \n",
      ">> adding gene annotation...\t\t\t 2023-06-22 03:41:46 PM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: org.Hs.eg.db\n",
      "\n",
      "'select()' returned 1:many mapping between keys and columns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2023-06-22 03:41:46 PM \n",
      ">> done...\t\t\t\t\t 2023-06-22 03:41:46 PM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "4995/4995  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "             Feature  Frequency\n",
      "9           Promoter 34.5745746\n",
      "4             5' UTR  0.2602603\n",
      "3             3' UTR  3.3033033\n",
      "1           1st Exon  1.5015015\n",
      "7         Other Exon  4.0640641\n",
      "2         1st Intron 14.9549550\n",
      "8       Other Intron 18.1581582\n",
      "6 Downstream (<=300)  2.3623624\n",
      "5  Distal Intergenic 20.8208208\n",
      "[1] \"lentiMPRA_element_level_all_result.inactive.merged.random5000\"\n",
      ">> loading peak file...\t\t\t\t 2023-06-22 03:41:46 PM \n",
      ">> preparing features information...\t\t 2023-06-22 03:41:46 PM \n",
      ">> identifying nearest features...\t\t 2023-06-22 03:41:46 PM \n",
      ">> calculating distance from peak to TSS...\t 2023-06-22 03:41:47 PM \n",
      ">> assigning genomic annotation...\t\t 2023-06-22 03:41:47 PM \n",
      ">> adding gene annotation...\t\t\t 2023-06-22 03:41:53 PM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2023-06-22 03:41:53 PM \n",
      ">> done...\t\t\t\t\t 2023-06-22 03:41:53 PM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "4998/4998  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "             Feature  Frequency\n",
      "9           Promoter 21.6086435\n",
      "4             5' UTR  0.3601441\n",
      "3             3' UTR  2.5610244\n",
      "1           1st Exon  1.5206082\n",
      "7         Other Exon  3.9215686\n",
      "2         1st Intron 14.1056423\n",
      "8       Other Intron 25.9503802\n",
      "6 Downstream (<=300)  1.9607843\n",
      "5  Distal Intergenic 28.0112045\n",
      "[1] \"wgSTARR_bin_level_all_result.inactive.merged.random5000\"\n",
      ">> loading peak file...\t\t\t\t 2023-06-22 03:41:53 PM \n",
      ">> preparing features information...\t\t 2023-06-22 03:41:53 PM \n",
      ">> identifying nearest features...\t\t 2023-06-22 03:41:53 PM \n",
      ">> calculating distance from peak to TSS...\t 2023-06-22 03:41:54 PM \n",
      ">> assigning genomic annotation...\t\t 2023-06-22 03:41:54 PM \n",
      ">> adding gene annotation...\t\t\t 2023-06-22 03:41:58 PM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2023-06-22 03:41:59 PM \n",
      ">> done...\t\t\t\t\t 2023-06-22 03:41:59 PM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "4999/4999  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "             Feature Frequency\n",
      "9           Promoter 18.643729\n",
      "4             5' UTR  0.480096\n",
      "3             3' UTR  3.560712\n",
      "1           1st Exon  1.500300\n",
      "7         Other Exon  5.421084\n",
      "2         1st Intron 11.662332\n",
      "8       Other Intron 31.106221\n",
      "6 Downstream (<=300)  0.920184\n",
      "5  Distal Intergenic 26.705341\n"
     ]
    }
   ],
   "source": [
    "suppressMessages(suppressWarnings(library(TxDb.Hsapiens.UCSC.hg38.knownGene)))\n",
    "suppressMessages(suppressWarnings(library(ChIPseeker)))\n",
    "\n",
    "txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene\n",
    "\n",
    "\n",
    "set_plot_dimensions <- function(width_choice, height_choice) {\n",
    "        options(repr.plot.width=width_choice, repr.plot.height=height_choice)\n",
    "        }\n",
    "\n",
    "files <- list(\"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_enhancer_calls_5Oct/z_score_processed/ASTARR/bin_level/ASTARR_bin_level_all_result.inactive.merged.random5000.bed\",\n",
    "              \"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_enhancer_calls_5Oct/z_score_processed/lentiMPRA/element_level/lentiMPRA_element_level_all_result.inactive.merged.random5000.bed\",\n",
    "              \"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_enhancer_calls_5Oct/z_score_processed/wgSTARR/bin_level/wgSTARR_bin_level_all_result.inactive.merged.random5000.bed\"\n",
    ")\n",
    "\n",
    "for (file in files){\n",
    "    filename <- sub('[.][^.]+$', '', basename(file))\n",
    "    print(filename)\n",
    "    peakAnnoEnrich <- annotatePeak(file, tssRegion=c(-1000, 1000),\n",
    "                                TxDb=txdb, annoDb=\"org.Hs.eg.db\")\n",
    "    print(peakAnnoEnrich)\n",
    "    foo <- as.data.frame(peakAnnoEnrich)\n",
    "    write.table(foo, \n",
    "             file=paste0('/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/enhancer_call_annotation/V2/',filename, '.annotated.txt'), \n",
    "            quote=F, sep='\\t')\n",
    "    \n",
    "}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Annotate Repressed Peaks: \n",
    "#### ASTARR, LentiMPRA, wgSTARR specific active and inactive peaks into promoter and distal -- V2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"ASTARR_merged_repressor_peaks_in_each_orientation.random5000\"\n",
      ">> loading peak file...\t\t\t\t 2023-08-01 01:05:45 PM \n",
      ">> preparing features information...\t\t 2023-08-01 01:05:45 PM \n",
      ">> identifying nearest features...\t\t 2023-08-01 01:05:48 PM \n",
      ">> calculating distance from peak to TSS...\t 2023-08-01 01:05:50 PM \n",
      ">> assigning genomic annotation...\t\t 2023-08-01 01:05:50 PM \n",
      ">> adding gene annotation...\t\t\t 2023-08-01 01:06:31 PM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: org.Hs.eg.db\n",
      "\n",
      "'select()' returned 1:many mapping between keys and columns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2023-08-01 01:06:32 PM \n",
      ">> done...\t\t\t\t\t 2023-08-01 01:06:32 PM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "4923/5000  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "             Feature  Frequency\n",
      "9           Promoter 77.9199675\n",
      "4             5' UTR  0.1828154\n",
      "3             3' UTR  1.2390819\n",
      "1           1st Exon  1.3609588\n",
      "7         Other Exon  1.8687792\n",
      "2         1st Intron  4.0828763\n",
      "8       Other Intron  4.9360146\n",
      "6 Downstream (<=300)  0.6906358\n",
      "5  Distal Intergenic  7.7188706\n",
      "[1] \"ASTARR_bin_level_all_result.inactive_repressed_GC_content_matched.random5000\"\n",
      ">> loading peak file...\t\t\t\t 2023-08-01 01:06:32 PM \n",
      ">> preparing features information...\t\t 2023-08-01 01:06:32 PM \n",
      ">> identifying nearest features...\t\t 2023-08-01 01:06:32 PM \n",
      ">> calculating distance from peak to TSS...\t 2023-08-01 01:06:33 PM \n",
      ">> assigning genomic annotation...\t\t 2023-08-01 01:06:33 PM \n",
      ">> adding gene annotation...\t\t\t 2023-08-01 01:06:40 PM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2023-08-01 01:06:40 PM \n",
      ">> done...\t\t\t\t\t 2023-08-01 01:06:40 PM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "5000/5000  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "             Feature Frequency\n",
      "9           Promoter     65.66\n",
      "4             5' UTR      0.24\n",
      "3             3' UTR      2.74\n",
      "1           1st Exon      1.32\n",
      "7         Other Exon      3.50\n",
      "2         1st Intron      6.88\n",
      "8       Other Intron      7.90\n",
      "6 Downstream (<=300)      1.80\n",
      "5  Distal Intergenic      9.96\n",
      "[1] \"merged_repressor_peaks_in_either_orientation\"\n",
      ">> loading peak file...\t\t\t\t 2023-08-01 01:06:40 PM \n",
      ">> preparing features information...\t\t 2023-08-01 01:06:40 PM \n",
      ">> identifying nearest features...\t\t 2023-08-01 01:06:40 PM \n",
      ">> calculating distance from peak to TSS...\t 2023-08-01 01:06:41 PM \n",
      ">> assigning genomic annotation...\t\t 2023-08-01 01:06:41 PM \n",
      ">> adding gene annotation...\t\t\t 2023-08-01 01:06:46 PM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2023-08-01 01:06:46 PM \n",
      ">> done...\t\t\t\t\t 2023-08-01 01:06:46 PM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "1525/1525  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "             Feature  Frequency\n",
      "9           Promoter 46.2950820\n",
      "4             5' UTR  0.3934426\n",
      "3             3' UTR  2.2295082\n",
      "1           1st Exon  1.7049180\n",
      "7         Other Exon  2.8196721\n",
      "2         1st Intron  7.4098361\n",
      "8       Other Intron 16.7868852\n",
      "6 Downstream (<=300)  0.9836066\n",
      "5  Distal Intergenic 21.3770492\n",
      "[1] \"lentiMPRA_element_level_all_result.inactive_repressed_GC_content_matched.random1700\"\n",
      ">> loading peak file...\t\t\t\t 2023-08-01 01:06:46 PM \n",
      ">> preparing features information...\t\t 2023-08-01 01:06:46 PM \n",
      ">> identifying nearest features...\t\t 2023-08-01 01:06:46 PM \n",
      ">> calculating distance from peak to TSS...\t 2023-08-01 01:06:47 PM \n",
      ">> assigning genomic annotation...\t\t 2023-08-01 01:06:47 PM \n",
      ">> adding gene annotation...\t\t\t 2023-08-01 01:06:53 PM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2023-08-01 01:06:53 PM \n",
      ">> done...\t\t\t\t\t 2023-08-01 01:06:53 PM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "1767/1767  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "             Feature  Frequency\n",
      "9           Promoter 11.0922467\n",
      "4             5' UTR  0.2263724\n",
      "3             3' UTR  2.6598755\n",
      "1           1st Exon  0.9054895\n",
      "7         Other Exon  3.2258065\n",
      "2         1st Intron 16.5251839\n",
      "8       Other Intron 38.1437465\n",
      "6 Downstream (<=300)  1.6411998\n",
      "5  Distal Intergenic 25.5800792\n",
      "[1] \"wgSTARR_merged_repressor_peaks_in_each_orientation.random5000\"\n",
      ">> loading peak file...\t\t\t\t 2023-08-01 01:06:53 PM \n",
      ">> preparing features information...\t\t 2023-08-01 01:06:53 PM \n",
      ">> identifying nearest features...\t\t 2023-08-01 01:06:53 PM \n",
      ">> calculating distance from peak to TSS...\t 2023-08-01 01:06:54 PM \n",
      ">> assigning genomic annotation...\t\t 2023-08-01 01:06:54 PM \n",
      ">> adding gene annotation...\t\t\t 2023-08-01 01:07:00 PM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2023-08-01 01:07:00 PM \n",
      ">> done...\t\t\t\t\t 2023-08-01 01:07:00 PM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "5000/5000  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "             Feature Frequency\n",
      "9           Promoter      6.04\n",
      "4             5' UTR      0.14\n",
      "3             3' UTR      1.68\n",
      "1           1st Exon      0.56\n",
      "7         Other Exon      2.12\n",
      "2         1st Intron     16.66\n",
      "8       Other Intron     37.40\n",
      "6 Downstream (<=300)      0.88\n",
      "5  Distal Intergenic     34.52\n",
      "[1] \"wgSTARR_bin_level_all_result.inactive_repressed_GC_content_matched.random5000\"\n",
      ">> loading peak file...\t\t\t\t 2023-08-01 01:07:00 PM \n",
      ">> preparing features information...\t\t 2023-08-01 01:07:00 PM \n",
      ">> identifying nearest features...\t\t 2023-08-01 01:07:00 PM \n",
      ">> calculating distance from peak to TSS...\t 2023-08-01 01:07:01 PM \n",
      ">> assigning genomic annotation...\t\t 2023-08-01 01:07:01 PM \n",
      ">> adding gene annotation...\t\t\t 2023-08-01 01:07:07 PM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2023-08-01 01:07:07 PM \n",
      ">> done...\t\t\t\t\t 2023-08-01 01:07:07 PM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "5005/5005  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "             Feature  Frequency\n",
      "9           Promoter 10.7292707\n",
      "4             5' UTR  0.4995005\n",
      "3             3' UTR  3.3966034\n",
      "1           1st Exon  1.3586414\n",
      "7         Other Exon  4.9950050\n",
      "2         1st Intron 13.6863137\n",
      "8       Other Intron 35.2847153\n",
      "6 Downstream (<=300)  0.7592408\n",
      "5  Distal Intergenic 29.2907093\n"
     ]
    }
   ],
   "source": [
    "suppressMessages(suppressWarnings(library(TxDb.Hsapiens.UCSC.hg38.knownGene)))\n",
    "suppressMessages(suppressWarnings(library(ChIPseeker)))\n",
    "\n",
    "txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene\n",
    "\n",
    "\n",
    "set_plot_dimensions <- function(width_choice, height_choice) {\n",
    "        options(repr.plot.width=width_choice, repr.plot.height=height_choice)\n",
    "        }\n",
    "\n",
    "files <- list(\"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_enhancer_calls_5Oct/z_score_processed/ASTARR/merged_peak/ASTARR_merged_repressor_peaks_in_each_orientation.random5000.bed\",\n",
    "              \"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_enhancer_calls_5Oct/z_score_processed/ASTARR/bin_level/ASTARR_bin_level_all_result.inactive_repressed_GC_content_matched.random5000.bed\",\n",
    "              \"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_enhancer_calls_5Oct/z_score_processed/lentiMPRA/merged_peak/merged_repressor_peaks_in_either_orientation.bed\",\n",
    "              \"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_enhancer_calls_5Oct/z_score_processed/lentiMPRA/lentiMPRA_element_level_all_result.inactive_repressed_GC_content_matched.random1700.bed\",\n",
    "              \"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_enhancer_calls_5Oct/z_score_processed/wgSTARR/merged_peak/wgSTARR_merged_repressor_peaks_in_each_orientation.random5000.bed\",\n",
    "              \"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_enhancer_calls_5Oct/z_score_processed/wgSTARR/bin_level/wgSTARR_bin_level_all_result.inactive_repressed_GC_content_matched.random5000.bed\"\n",
    ")\n",
    "\n",
    "for (file in files){\n",
    "    filename <- sub('[.][^.]+$', '', basename(file))\n",
    "    print(filename)\n",
    "    peakAnnoEnrich <- annotatePeak(file, tssRegion=c(-1000, 1000),\n",
    "                                TxDb=txdb, annoDb=\"org.Hs.eg.db\")\n",
    "    print(peakAnnoEnrich)\n",
    "    foo <- as.data.frame(peakAnnoEnrich)\n",
    "    write.table(foo, \n",
    "             file=paste0('/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/enhancer_call_annotation/V3_repressed/',filename, '.annotated.txt'), \n",
    "            quote=F, sep='\\t')\n",
    "    \n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"ASTARR_LentiMPRA_wgSTARR_merged_enhancer_peak_in_each_orientation.filtered\"\n",
      ">> loading peak file...\t\t\t\t 2023-11-28 12:41:35 PM \n",
      ">> preparing features information...\t\t 2023-11-28 12:41:35 PM \n",
      ">> identifying nearest features...\t\t 2023-11-28 12:41:38 PM \n",
      ">> calculating distance from peak to TSS...\t 2023-11-28 12:41:40 PM \n",
      ">> assigning genomic annotation...\t\t 2023-11-28 12:41:40 PM \n",
      ">> adding gene annotation...\t\t\t 2023-11-28 12:42:33 PM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: org.Hs.eg.db\n",
      "\n",
      "\n",
      "\n",
      "'select()' returned 1:many mapping between keys and columns\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2023-11-28 12:42:34 PM \n",
      ">> done...\t\t\t\t\t 2023-11-28 12:42:34 PM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "2066/2066  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "             Feature  Frequency\n",
      "9           Promoter 28.3155857\n",
      "4             5' UTR  0.1452081\n",
      "3             3' UTR  1.5972894\n",
      "1           1st Exon  1.2584705\n",
      "7         Other Exon  2.3233301\n",
      "2         1st Intron 13.4075508\n",
      "8       Other Intron 25.8470474\n",
      "6 Downstream (<=300)  0.1452081\n",
      "5  Distal Intergenic 26.9603098\n",
      "[1] \"ASTARR_LentiMPRA_wgSTARR_common_background_with_ATAC.max_bw_signal.sloped_100bp.random2000\"\n",
      ">> loading peak file...\t\t\t\t 2023-11-28 12:42:34 PM \n",
      ">> preparing features information...\t\t 2023-11-28 12:42:34 PM \n",
      ">> identifying nearest features...\t\t 2023-11-28 12:42:34 PM \n",
      ">> calculating distance from peak to TSS...\t 2023-11-28 12:42:35 PM \n",
      ">> assigning genomic annotation...\t\t 2023-11-28 12:42:35 PM \n",
      ">> adding gene annotation...\t\t\t 2023-11-28 12:42:42 PM \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> assigning chromosome lengths\t\t\t 2023-11-28 12:42:42 PM \n",
      ">> done...\t\t\t\t\t 2023-11-28 12:42:42 PM \n",
      "Annotated peaks generated by ChIPseeker\n",
      "2077/2077  peaks were annotated\n",
      "Genomic Annotation Summary:\n",
      "             Feature  Frequency\n",
      "8           Promoter 16.4660568\n",
      "3             3' UTR  3.5628310\n",
      "1           1st Exon  0.9147809\n",
      "6         Other Exon  2.8887819\n",
      "2         1st Intron 17.0438132\n",
      "7       Other Intron 28.6470871\n",
      "5 Downstream (<=300)  0.2888782\n",
      "4  Distal Intergenic 30.1877708\n"
     ]
    }
   ],
   "source": [
    "suppressMessages(suppressWarnings(library(TxDb.Hsapiens.UCSC.hg38.knownGene)))\n",
    "suppressMessages(suppressWarnings(library(ChIPseeker)))\n",
    "\n",
    "txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene\n",
    "\n",
    "\n",
    "set_plot_dimensions <- function(width_choice, height_choice) {\n",
    "        options(repr.plot.width=width_choice, repr.plot.height=height_choice)\n",
    "        }\n",
    "\n",
    "files <- list(\"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_enhancer_calls_5Oct/z_score_processed/intersected/ASTARR_LentiMPRA_wgSTARR_merged_enhancer_peak_in_each_orientation.filtered.bed\",\n",
    "              \"/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/Junke_enhancer_calls_5Oct/z_score_processed/intersected/ASTARR_LentiMPRA_wgSTARR_common_background_with_ATAC.max_bw_signal.sloped_100bp.random2000.bed\"\n",
    ")\n",
    "\n",
    "for (file in files){\n",
    "    filename <- sub('[.][^.]+$', '', basename(file))\n",
    "    print(filename)\n",
    "    peakAnnoEnrich <- annotatePeak(file, tssRegion=c(-1000, 1000),\n",
    "                                TxDb=txdb, annoDb=\"org.Hs.eg.db\")\n",
    "    print(peakAnnoEnrich)\n",
    "    foo <- as.data.frame(peakAnnoEnrich)\n",
    "    write.table(foo, \n",
    "             file=paste0('/data/reddylab/Revathy/collabs/Jamborees/fcc_capstone/data/enhancer_call_annotation/V2/',filename, '.annotated.txt'), \n",
    "            quote=F, sep='\\t')\n",
    "    \n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n",
      "'help(\"repositories\", package = \"BiocManager\")' for details.\n",
      "Replacement repositories:\n",
      "    CRAN: https://cran.r-project.org\n",
      "\n",
      "Bioconductor version 3.18 (BiocManager 1.30.22), R 4.3.1 (2023-06-16)\n",
      "\n",
      "Installing package(s) 'AnnotationDbi'\n",
      "\n",
      "Warning message in install.packages(...):\n",
      "installation of package AnnotationDbi had non-zero exit status\n",
      "Old packages: 'AnnotationDbi', 'chromVAR', 'CNEr', 'DelayedArray', 'DT',\n",
      "  'GenomicAlignments', 'GenomicRanges', 'gert', 'ggrepel', 'GO.db',\n",
      "  'htmlwidgets', 'KEGGREST', 'motifmatchr', 'plotly', 'prodlim', 'ragg',\n",
      "  'recipes', 'rmarkdown', 'Rsamtools', 'rtracklayer', 'sctransform', 'Seurat',\n",
      "  'spatstat.explore', 'spatstat.random', 'SummarizedExperiment', 'systemfonts',\n",
      "  'textshaping', 'TFBSTools', 'XML'\n",
      "\n"
     ]
    }
   ],
   "source": [
    "if (!require(\"BiocManager\", quietly = TRUE))\n",
    "    install.packages(\"BiocManager\")\n",
    "\n",
    "BiocManager::install(\"AnnotationDbi\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "if (!require(\"BiocManager\", quietly = TRUE))\n",
    "    install.packages(\"BiocManager\")\n",
    "\n",
    "BiocManager::install([\"BiocIO\", \"Biostrings\", \"BSgenome\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<strong>CRAN:</strong> 'https://cran.r-project.org'"
      ],
      "text/latex": [
       "\\textbf{CRAN:} 'https://cran.r-project.org'"
      ],
      "text/markdown": [
       "**CRAN:** 'https://cran.r-project.org'"
      ],
      "text/plain": [
       "                        CRAN \n",
       "\"https://cran.r-project.org\" \n",
       "attr(,\"irkernel\")\n",
       "[1] TRUE"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "getOption(\"repos\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n",
      "'help(\"repositories\", package = \"BiocManager\")' for details.\n",
      "Replacement repositories:\n",
      "    CRAN: https://cran.r-project.org\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>BioCsoft</dt><dd>'https://bioconductor.org/packages/3.18/bioc'</dd><dt>BioCann</dt><dd>'https://bioconductor.org/packages/3.18/data/annotation'</dd><dt>BioCexp</dt><dd>'https://bioconductor.org/packages/3.18/data/experiment'</dd><dt>BioCworkflows</dt><dd>'https://bioconductor.org/packages/3.18/workflows'</dd><dt>BioCbooks</dt><dd>'https://bioconductor.org/packages/3.18/books'</dd><dt>CRAN</dt><dd>'https://cran.r-project.org'</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[BioCsoft] 'https://bioconductor.org/packages/3.18/bioc'\n",
       "\\item[BioCann] 'https://bioconductor.org/packages/3.18/data/annotation'\n",
       "\\item[BioCexp] 'https://bioconductor.org/packages/3.18/data/experiment'\n",
       "\\item[BioCworkflows] 'https://bioconductor.org/packages/3.18/workflows'\n",
       "\\item[BioCbooks] 'https://bioconductor.org/packages/3.18/books'\n",
       "\\item[CRAN] 'https://cran.r-project.org'\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "BioCsoft\n",
       ":   'https://bioconductor.org/packages/3.18/bioc'BioCann\n",
       ":   'https://bioconductor.org/packages/3.18/data/annotation'BioCexp\n",
       ":   'https://bioconductor.org/packages/3.18/data/experiment'BioCworkflows\n",
       ":   'https://bioconductor.org/packages/3.18/workflows'BioCbooks\n",
       ":   'https://bioconductor.org/packages/3.18/books'CRAN\n",
       ":   'https://cran.r-project.org'\n",
       "\n"
      ],
      "text/plain": [
       "                                                BioCsoft \n",
       "           \"https://bioconductor.org/packages/3.18/bioc\" \n",
       "                                                 BioCann \n",
       "\"https://bioconductor.org/packages/3.18/data/annotation\" \n",
       "                                                 BioCexp \n",
       "\"https://bioconductor.org/packages/3.18/data/experiment\" \n",
       "                                           BioCworkflows \n",
       "      \"https://bioconductor.org/packages/3.18/workflows\" \n",
       "                                               BioCbooks \n",
       "          \"https://bioconductor.org/packages/3.18/books\" \n",
       "                                                    CRAN \n",
       "                            \"https://cran.r-project.org\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "BiocManager::repositories(version=\"3.18\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "R version 4.3.1 (2023-06-16)\n",
       "Platform: x86_64-conda-linux-gnu (64-bit)\n",
       "Running under: CentOS Linux 7 (Core)\n",
       "\n",
       "Matrix products: default\n",
       "BLAS/LAPACK: /gpfs/fs1/home/rv103/miniconda3/envs/r_env/lib/libopenblasp-r0.3.23.so;  LAPACK version 3.11.0\n",
       "\n",
       "locale:\n",
       " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
       " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
       " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
       " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
       " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
       "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
       "\n",
       "time zone: America/New_York\n",
       "tzcode source: system (glibc)\n",
       "\n",
       "attached base packages:\n",
       "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
       "\n",
       "other attached packages:\n",
       "[1] BiocManager_1.30.21.1\n",
       "\n",
       "loaded via a namespace (and not attached):\n",
       " [1] digest_0.6.33       IRdisplay_1.1       utf8_1.2.3         \n",
       " [4] base64enc_0.1-3     fastmap_1.1.1       glue_1.6.2         \n",
       " [7] htmltools_0.5.5     repr_1.1.6          lifecycle_1.0.3    \n",
       "[10] cli_3.6.1           fansi_1.0.4         vctrs_0.6.3        \n",
       "[13] pbdZMQ_0.3-9        compiler_4.3.1      tools_4.3.1        \n",
       "[16] evaluate_0.21       pillar_1.9.0        crayon_1.5.2       \n",
       "[19] rlang_1.1.1         jsonlite_1.8.7      IRkernel_1.3.2.9000\n",
       "[22] uuid_1.1-0         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sessionInfo()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in library(ChIPseeker): there is no package called ChIPseeker\n",
     "output_type": "error",
     "traceback": [
      "Error in library(ChIPseeker): there is no package called ChIPseeker\nTraceback:\n",
      "1. library(ChIPseeker)"
     ]
    }
   ],
   "source": [
    "library(ChIPseeker)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R(base_py3)",
   "language": "R",
   "name": "r_env"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
