{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Define candidate enhancer regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running: awk 'FNR==NR {x2[$1] = $0; next} $1 in x2 {print x2[$1]}' /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes <(samtools view -H /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_1.masked.dups_marked.bam | grep SQ | cut -f 2 | cut -c 4- )  > /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_1.masked.dedup.sorted_peaks.narrowPeak.Th01_1.masked.dups_marked.bam.Counts.bed.temp_sort_order\n",
      "Running: bedtools sort -faidx /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_1.masked.dedup.sorted_peaks.narrowPeak.Th01_1.masked.dups_marked.bam.Counts.bed.temp_sort_order -i /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_1.masked.dedup.sorted_peaks.narrowPeak | bedtools coverage -g /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_1.masked.dedup.sorted_peaks.narrowPeak.Th01_1.masked.dups_marked.bam.Counts.bed.temp_sort_order -counts -sorted -a stdin -b /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_1.masked.dups_marked.bam | awk '{print $1 \"\\t\" $2 \"\\t\" $3 \"\\t\" $NF}'  | bedtools sort -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes -i stdin > /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_1.masked.dedup.sorted_peaks.narrowPeak.Th01_1.masked.dups_marked.bam.Counts.bed; rm /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_1.masked.dedup.sorted_peaks.narrowPeak.Th01_1.masked.dups_marked.bam.Counts.bed.temp_sort_order\n",
      "Fast count method failed to count: /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_1.masked.dups_marked.bam\n",
      "\n",
      "Error: The genome file /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_1.masked.dedup.sorted_peaks.narrowPeak.Th01_1.masked.dups_marked.bam.Counts.bed.temp_sort_order has no valid entries. Exiting.\n",
      "Chromosome \"chr1\" undefined in /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_1.masked.dedup.sorted_peaks.narrowPeak.Th01_1.masked.dups_marked.bam.Counts.bed.temp_sort_order\n",
      "\n",
      "Trying bamtobed method ...\n",
      "\n",
      "Running: bedtools bamtobed -i /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_1.masked.dups_marked.bam | cut -f 1-3 | bedtools intersect -wa -a stdin -b /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes.bed | bedtools sort -i stdin -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes | bedtools coverage -g /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes -counts -sorted -a /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_1.masked.dedup.sorted_peaks.narrowPeak -b stdin | awk '{print $1 \"\\t\" $2 \"\\t\" $3 \"\\t\" $NF}' > /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_1.masked.dedup.sorted_peaks.narrowPeak.Th01_1.masked.dups_marked.bam.Counts.bed\n",
      "BEDTools failed to count file: /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_1.masked.dups_marked.bam\n",
      "\n",
      "Error: requested chromosome chr14_GL000225v1_random does not exist in the genome file /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes. Exiting.\n",
      "\n",
      "Running: bedtools sort -i /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_1.masked.dedup.sorted_peaks.narrowPeak.Th01_1.masked.dups_marked.bam.Counts.bed -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes | bedtools merge -i stdin -c 4 -o max | sort -nr -k 4 | head -n 150000 |bedtools intersect -b stdin -a /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_1.masked.dedup.sorted_peaks.narrowPeak -wa |awk '{print $1 \"\\t\" $2 + $10 \"\\t\" $2 + $10}' |bedtools slop -i stdin -b 250 -g /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes |bedtools sort -i stdin -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes |bedtools merge -i stdin | bedtools intersect -v -wa -a stdin -b /data/reddylab/Revathy/dev/Maria/hg38_wgEncodeHg19ConsensusSignalArtifactRegions.bed | cut -f 1-3 | (bedtools intersect -a /data/reddylab/Revathy/dev/Maria/hg38_RefSeqCurated.170308.bed.CollapsedGeneBounds.TSS500bp.bed -b /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes.bed -wa | cut -f 1-3 && cat) |bedtools sort -i stdin -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes | bedtools merge -i stdin > /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_1.masked.dedup.sorted_peaks.narrowPeak.candidateRegions.bed\n",
      "Running: awk 'FNR==NR {x2[$1] = $0; next} $1 in x2 {print x2[$1]}' /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes <(samtools view -H /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_2.masked.dups_marked.bam | grep SQ | cut -f 2 | cut -c 4- )  > /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_2.masked.dedup.sorted_peaks.narrowPeak.Th01_2.masked.dups_marked.bam.Counts.bed.temp_sort_order\n",
      "Running: bedtools sort -faidx /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_2.masked.dedup.sorted_peaks.narrowPeak.Th01_2.masked.dups_marked.bam.Counts.bed.temp_sort_order -i /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_2.masked.dedup.sorted_peaks.narrowPeak | bedtools coverage -g /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_2.masked.dedup.sorted_peaks.narrowPeak.Th01_2.masked.dups_marked.bam.Counts.bed.temp_sort_order -counts -sorted -a stdin -b /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_2.masked.dups_marked.bam | awk '{print $1 \"\\t\" $2 \"\\t\" $3 \"\\t\" $NF}'  | bedtools sort -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes -i stdin > /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_2.masked.dedup.sorted_peaks.narrowPeak.Th01_2.masked.dups_marked.bam.Counts.bed; rm /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_2.masked.dedup.sorted_peaks.narrowPeak.Th01_2.masked.dups_marked.bam.Counts.bed.temp_sort_order\n",
      "Fast count method failed to count: /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_2.masked.dups_marked.bam\n",
      "\n",
      "Error: The genome file /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_2.masked.dedup.sorted_peaks.narrowPeak.Th01_2.masked.dups_marked.bam.Counts.bed.temp_sort_order has no valid entries. Exiting.\n",
      "Chromosome \"chr1\" undefined in /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_2.masked.dedup.sorted_peaks.narrowPeak.Th01_2.masked.dups_marked.bam.Counts.bed.temp_sort_order\n",
      "\n",
      "Trying bamtobed method ...\n",
      "\n",
      "Running: bedtools bamtobed -i /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_2.masked.dups_marked.bam | cut -f 1-3 | bedtools intersect -wa -a stdin -b /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes.bed | bedtools sort -i stdin -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes | bedtools coverage -g /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes -counts -sorted -a /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_2.masked.dedup.sorted_peaks.narrowPeak -b stdin | awk '{print $1 \"\\t\" $2 \"\\t\" $3 \"\\t\" $NF}' > /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_2.masked.dedup.sorted_peaks.narrowPeak.Th01_2.masked.dups_marked.bam.Counts.bed\n",
      "BEDTools failed to count file: /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_2.masked.dups_marked.bam\n",
      "\n",
      "Error: requested chromosome chr14_GL000225v1_random does not exist in the genome file /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes. Exiting.\n",
      "\n",
      "Running: bedtools sort -i /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_2.masked.dedup.sorted_peaks.narrowPeak.Th01_2.masked.dups_marked.bam.Counts.bed -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes | bedtools merge -i stdin -c 4 -o max | sort -nr -k 4 | head -n 150000 |bedtools intersect -b stdin -a /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_2.masked.dedup.sorted_peaks.narrowPeak -wa |awk '{print $1 \"\\t\" $2 + $10 \"\\t\" $2 + $10}' |bedtools slop -i stdin -b 250 -g /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes |bedtools sort -i stdin -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes |bedtools merge -i stdin | bedtools intersect -v -wa -a stdin -b /data/reddylab/Revathy/dev/Maria/hg38_wgEncodeHg19ConsensusSignalArtifactRegions.bed | cut -f 1-3 | (bedtools intersect -a /data/reddylab/Revathy/dev/Maria/hg38_RefSeqCurated.170308.bed.CollapsedGeneBounds.TSS500bp.bed -b /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes.bed -wa | cut -f 1-3 && cat) |bedtools sort -i stdin -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes | bedtools merge -i stdin > /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_2.masked.dedup.sorted_peaks.narrowPeak.candidateRegions.bed\n",
      "Running: awk 'FNR==NR {x2[$1] = $0; next} $1 in x2 {print x2[$1]}' /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes <(samtools view -H /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_cc_1.masked.dups_marked.bam | grep SQ | cut -f 2 | cut -c 4- )  > /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_cc_1.masked.dedup.sorted_peaks.narrowPeak.Th01_cc_1.masked.dups_marked.bam.Counts.bed.temp_sort_order\n",
      "Running: bedtools sort -faidx /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_cc_1.masked.dedup.sorted_peaks.narrowPeak.Th01_cc_1.masked.dups_marked.bam.Counts.bed.temp_sort_order -i /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_cc_1.masked.dedup.sorted_peaks.narrowPeak | bedtools coverage -g /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_cc_1.masked.dedup.sorted_peaks.narrowPeak.Th01_cc_1.masked.dups_marked.bam.Counts.bed.temp_sort_order -counts -sorted -a stdin -b /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_cc_1.masked.dups_marked.bam | awk '{print $1 \"\\t\" $2 \"\\t\" $3 \"\\t\" $NF}'  | bedtools sort -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes -i stdin > /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_cc_1.masked.dedup.sorted_peaks.narrowPeak.Th01_cc_1.masked.dups_marked.bam.Counts.bed; rm /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_cc_1.masked.dedup.sorted_peaks.narrowPeak.Th01_cc_1.masked.dups_marked.bam.Counts.bed.temp_sort_order\n",
      "Fast count method failed to count: /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_cc_1.masked.dups_marked.bam\n",
      "\n",
      "Error: The genome file /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_cc_1.masked.dedup.sorted_peaks.narrowPeak.Th01_cc_1.masked.dups_marked.bam.Counts.bed.temp_sort_order has no valid entries. Exiting.\n",
      "Chromosome \"chr1\" undefined in /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_cc_1.masked.dedup.sorted_peaks.narrowPeak.Th01_cc_1.masked.dups_marked.bam.Counts.bed.temp_sort_order\n",
      "\n",
      "Trying bamtobed method ...\n",
      "\n",
      "Running: bedtools bamtobed -i /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_cc_1.masked.dups_marked.bam | cut -f 1-3 | bedtools intersect -wa -a stdin -b /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes.bed | bedtools sort -i stdin -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes | bedtools coverage -g /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes -counts -sorted -a /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_cc_1.masked.dedup.sorted_peaks.narrowPeak -b stdin | awk '{print $1 \"\\t\" $2 \"\\t\" $3 \"\\t\" $NF}' > /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_cc_1.masked.dedup.sorted_peaks.narrowPeak.Th01_cc_1.masked.dups_marked.bam.Counts.bed\n",
      "BEDTools failed to count file: /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_cc_1.masked.dups_marked.bam\n",
      "\n",
      "Error: requested chromosome chr14_GL000194v1_random does not exist in the genome file /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes. Exiting.\n",
      "\n",
      "Running: bedtools sort -i /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_cc_1.masked.dedup.sorted_peaks.narrowPeak.Th01_cc_1.masked.dups_marked.bam.Counts.bed -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes | bedtools merge -i stdin -c 4 -o max | sort -nr -k 4 | head -n 150000 |bedtools intersect -b stdin -a /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_cc_1.masked.dedup.sorted_peaks.narrowPeak -wa |awk '{print $1 \"\\t\" $2 + $10 \"\\t\" $2 + $10}' |bedtools slop -i stdin -b 250 -g /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes |bedtools sort -i stdin -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes |bedtools merge -i stdin | bedtools intersect -v -wa -a stdin -b /data/reddylab/Revathy/dev/Maria/hg38_wgEncodeHg19ConsensusSignalArtifactRegions.bed | cut -f 1-3 | (bedtools intersect -a /data/reddylab/Revathy/dev/Maria/hg38_RefSeqCurated.170308.bed.CollapsedGeneBounds.TSS500bp.bed -b /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes.bed -wa | cut -f 1-3 && cat) |bedtools sort -i stdin -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes | bedtools merge -i stdin > /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_cc_1.masked.dedup.sorted_peaks.narrowPeak.candidateRegions.bed\n",
      "Running: awk 'FNR==NR {x2[$1] = $0; next} $1 in x2 {print x2[$1]}' /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes <(samtools view -H /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_cc_2.masked.dups_marked.bam | grep SQ | cut -f 2 | cut -c 4- )  > /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_cc_2.masked.dedup.sorted_peaks.narrowPeak.Th01_cc_2.masked.dups_marked.bam.Counts.bed.temp_sort_order\n",
      "Running: bedtools sort -faidx /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_cc_2.masked.dedup.sorted_peaks.narrowPeak.Th01_cc_2.masked.dups_marked.bam.Counts.bed.temp_sort_order -i /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_cc_2.masked.dedup.sorted_peaks.narrowPeak | bedtools coverage -g /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_cc_2.masked.dedup.sorted_peaks.narrowPeak.Th01_cc_2.masked.dups_marked.bam.Counts.bed.temp_sort_order -counts -sorted -a stdin -b /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_cc_2.masked.dups_marked.bam | awk '{print $1 \"\\t\" $2 \"\\t\" $3 \"\\t\" $NF}'  | bedtools sort -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes -i stdin > /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_cc_2.masked.dedup.sorted_peaks.narrowPeak.Th01_cc_2.masked.dups_marked.bam.Counts.bed; rm /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_cc_2.masked.dedup.sorted_peaks.narrowPeak.Th01_cc_2.masked.dups_marked.bam.Counts.bed.temp_sort_order\n",
      "Fast count method failed to count: /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_cc_2.masked.dups_marked.bam\n",
      "\n",
      "Error: The genome file /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_cc_2.masked.dedup.sorted_peaks.narrowPeak.Th01_cc_2.masked.dups_marked.bam.Counts.bed.temp_sort_order has no valid entries. Exiting.\n",
      "Chromosome \"chr1\" undefined in /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_cc_2.masked.dedup.sorted_peaks.narrowPeak.Th01_cc_2.masked.dups_marked.bam.Counts.bed.temp_sort_order\n",
      "\n",
      "Trying bamtobed method ...\n",
      "\n",
      "Running: bedtools bamtobed -i /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_cc_2.masked.dups_marked.bam | cut -f 1-3 | bedtools intersect -wa -a stdin -b /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes.bed | bedtools sort -i stdin -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes | bedtools coverage -g /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes -counts -sorted -a /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_cc_2.masked.dedup.sorted_peaks.narrowPeak -b stdin | awk '{print $1 \"\\t\" $2 \"\\t\" $3 \"\\t\" $NF}' > /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_cc_2.masked.dedup.sorted_peaks.narrowPeak.Th01_cc_2.masked.dups_marked.bam.Counts.bed\n",
      "BEDTools failed to count file: /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_cc_2.masked.dups_marked.bam\n",
      "\n",
      "Error: requested chromosome chr14_GL000194v1_random does not exist in the genome file /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes. Exiting.\n",
      "\n",
      "Running: bedtools sort -i /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_cc_2.masked.dedup.sorted_peaks.narrowPeak.Th01_cc_2.masked.dups_marked.bam.Counts.bed -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes | bedtools merge -i stdin -c 4 -o max | sort -nr -k 4 | head -n 150000 |bedtools intersect -b stdin -a /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_cc_2.masked.dedup.sorted_peaks.narrowPeak -wa |awk '{print $1 \"\\t\" $2 + $10 \"\\t\" $2 + $10}' |bedtools slop -i stdin -b 250 -g /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes |bedtools sort -i stdin -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes |bedtools merge -i stdin | bedtools intersect -v -wa -a stdin -b /data/reddylab/Revathy/dev/Maria/hg38_wgEncodeHg19ConsensusSignalArtifactRegions.bed | cut -f 1-3 | (bedtools intersect -a /data/reddylab/Revathy/dev/Maria/hg38_RefSeqCurated.170308.bed.CollapsedGeneBounds.TSS500bp.bed -b /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes.bed -wa | cut -f 1-3 && cat) |bedtools sort -i stdin -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes | bedtools merge -i stdin > /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_cc_2.masked.dedup.sorted_peaks.narrowPeak.candidateRegions.bed\n",
      "Running: awk 'FNR==NR {x2[$1] = $0; next} $1 in x2 {print x2[$1]}' /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes <(samtools view -H /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th17_1.masked.dups_marked.bam | grep SQ | cut -f 2 | cut -c 4- )  > /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_1.masked.dedup.sorted_peaks.narrowPeak.Th17_1.masked.dups_marked.bam.Counts.bed.temp_sort_order\n",
      "Running: bedtools sort -faidx /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_1.masked.dedup.sorted_peaks.narrowPeak.Th17_1.masked.dups_marked.bam.Counts.bed.temp_sort_order -i /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th17_1.masked.dedup.sorted_peaks.narrowPeak | bedtools coverage -g /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_1.masked.dedup.sorted_peaks.narrowPeak.Th17_1.masked.dups_marked.bam.Counts.bed.temp_sort_order -counts -sorted -a stdin -b /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th17_1.masked.dups_marked.bam | awk '{print $1 \"\\t\" $2 \"\\t\" $3 \"\\t\" $NF}'  | bedtools sort -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes -i stdin > /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_1.masked.dedup.sorted_peaks.narrowPeak.Th17_1.masked.dups_marked.bam.Counts.bed; rm /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_1.masked.dedup.sorted_peaks.narrowPeak.Th17_1.masked.dups_marked.bam.Counts.bed.temp_sort_order\n",
      "Fast count method failed to count: /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th17_1.masked.dups_marked.bam\n",
      "\n",
      "Error: The genome file /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_1.masked.dedup.sorted_peaks.narrowPeak.Th17_1.masked.dups_marked.bam.Counts.bed.temp_sort_order has no valid entries. Exiting.\n",
      "Chromosome \"chr1\" undefined in /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_1.masked.dedup.sorted_peaks.narrowPeak.Th17_1.masked.dups_marked.bam.Counts.bed.temp_sort_order\n",
      "\n",
      "Trying bamtobed method ...\n",
      "\n",
      "Running: bedtools bamtobed -i /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th17_1.masked.dups_marked.bam | cut -f 1-3 | bedtools intersect -wa -a stdin -b /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes.bed | bedtools sort -i stdin -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes | bedtools coverage -g /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes -counts -sorted -a /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th17_1.masked.dedup.sorted_peaks.narrowPeak -b stdin | awk '{print $1 \"\\t\" $2 \"\\t\" $3 \"\\t\" $NF}' > /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_1.masked.dedup.sorted_peaks.narrowPeak.Th17_1.masked.dups_marked.bam.Counts.bed\n",
      "BEDTools failed to count file: /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th17_1.masked.dups_marked.bam\n",
      "\n",
      "Error: requested chromosome chr14_GL000194v1_random does not exist in the genome file /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes. Exiting.\n",
      "\n",
      "Running: bedtools sort -i /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_1.masked.dedup.sorted_peaks.narrowPeak.Th17_1.masked.dups_marked.bam.Counts.bed -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes | bedtools merge -i stdin -c 4 -o max | sort -nr -k 4 | head -n 150000 |bedtools intersect -b stdin -a /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th17_1.masked.dedup.sorted_peaks.narrowPeak -wa |awk '{print $1 \"\\t\" $2 + $10 \"\\t\" $2 + $10}' |bedtools slop -i stdin -b 250 -g /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes |bedtools sort -i stdin -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes |bedtools merge -i stdin | bedtools intersect -v -wa -a stdin -b /data/reddylab/Revathy/dev/Maria/hg38_wgEncodeHg19ConsensusSignalArtifactRegions.bed | cut -f 1-3 | (bedtools intersect -a /data/reddylab/Revathy/dev/Maria/hg38_RefSeqCurated.170308.bed.CollapsedGeneBounds.TSS500bp.bed -b /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes.bed -wa | cut -f 1-3 && cat) |bedtools sort -i stdin -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes | bedtools merge -i stdin > /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_1.masked.dedup.sorted_peaks.narrowPeak.candidateRegions.bed\n",
      "Running: awk 'FNR==NR {x2[$1] = $0; next} $1 in x2 {print x2[$1]}' /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes <(samtools view -H /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th17_2.masked.dups_marked.bam | grep SQ | cut -f 2 | cut -c 4- )  > /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_2.masked.dedup.sorted_peaks.narrowPeak.Th17_2.masked.dups_marked.bam.Counts.bed.temp_sort_order\n",
      "Running: bedtools sort -faidx /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_2.masked.dedup.sorted_peaks.narrowPeak.Th17_2.masked.dups_marked.bam.Counts.bed.temp_sort_order -i /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th17_2.masked.dedup.sorted_peaks.narrowPeak | bedtools coverage -g /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_2.masked.dedup.sorted_peaks.narrowPeak.Th17_2.masked.dups_marked.bam.Counts.bed.temp_sort_order -counts -sorted -a stdin -b /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th17_2.masked.dups_marked.bam | awk '{print $1 \"\\t\" $2 \"\\t\" $3 \"\\t\" $NF}'  | bedtools sort -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes -i stdin > /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_2.masked.dedup.sorted_peaks.narrowPeak.Th17_2.masked.dups_marked.bam.Counts.bed; rm /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_2.masked.dedup.sorted_peaks.narrowPeak.Th17_2.masked.dups_marked.bam.Counts.bed.temp_sort_order\n",
      "Fast count method failed to count: /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th17_2.masked.dups_marked.bam\n",
      "\n",
      "Error: The genome file /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_2.masked.dedup.sorted_peaks.narrowPeak.Th17_2.masked.dups_marked.bam.Counts.bed.temp_sort_order has no valid entries. Exiting.\n",
      "Chromosome \"chr1\" undefined in /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_2.masked.dedup.sorted_peaks.narrowPeak.Th17_2.masked.dups_marked.bam.Counts.bed.temp_sort_order\n",
      "\n",
      "Trying bamtobed method ...\n",
      "\n",
      "Running: bedtools bamtobed -i /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th17_2.masked.dups_marked.bam | cut -f 1-3 | bedtools intersect -wa -a stdin -b /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes.bed | bedtools sort -i stdin -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes | bedtools coverage -g /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes -counts -sorted -a /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th17_2.masked.dedup.sorted_peaks.narrowPeak -b stdin | awk '{print $1 \"\\t\" $2 \"\\t\" $3 \"\\t\" $NF}' > /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_2.masked.dedup.sorted_peaks.narrowPeak.Th17_2.masked.dups_marked.bam.Counts.bed\n",
      "BEDTools failed to count file: /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th17_2.masked.dups_marked.bam\n",
      "\n",
      "Error: requested chromosome chr14_GL000225v1_random does not exist in the genome file /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes. Exiting.\n",
      "\n",
      "Running: bedtools sort -i /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_2.masked.dedup.sorted_peaks.narrowPeak.Th17_2.masked.dups_marked.bam.Counts.bed -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes | bedtools merge -i stdin -c 4 -o max | sort -nr -k 4 | head -n 150000 |bedtools intersect -b stdin -a /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th17_2.masked.dedup.sorted_peaks.narrowPeak -wa |awk '{print $1 \"\\t\" $2 + $10 \"\\t\" $2 + $10}' |bedtools slop -i stdin -b 250 -g /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes |bedtools sort -i stdin -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes |bedtools merge -i stdin | bedtools intersect -v -wa -a stdin -b /data/reddylab/Revathy/dev/Maria/hg38_wgEncodeHg19ConsensusSignalArtifactRegions.bed | cut -f 1-3 | (bedtools intersect -a /data/reddylab/Revathy/dev/Maria/hg38_RefSeqCurated.170308.bed.CollapsedGeneBounds.TSS500bp.bed -b /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes.bed -wa | cut -f 1-3 && cat) |bedtools sort -i stdin -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes | bedtools merge -i stdin > /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_2.masked.dedup.sorted_peaks.narrowPeak.candidateRegions.bed\n",
      "Running: awk 'FNR==NR {x2[$1] = $0; next} $1 in x2 {print x2[$1]}' /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes <(samtools view -H /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th17_cc_1.masked.dups_marked.bam | grep SQ | cut -f 2 | cut -c 4- )  > /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_cc_1.masked.dedup.sorted_peaks.narrowPeak.Th17_cc_1.masked.dups_marked.bam.Counts.bed.temp_sort_order\n",
      "Running: bedtools sort -faidx /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_cc_1.masked.dedup.sorted_peaks.narrowPeak.Th17_cc_1.masked.dups_marked.bam.Counts.bed.temp_sort_order -i /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th17_cc_1.masked.dedup.sorted_peaks.narrowPeak | bedtools coverage -g /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_cc_1.masked.dedup.sorted_peaks.narrowPeak.Th17_cc_1.masked.dups_marked.bam.Counts.bed.temp_sort_order -counts -sorted -a stdin -b /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th17_cc_1.masked.dups_marked.bam | awk '{print $1 \"\\t\" $2 \"\\t\" $3 \"\\t\" $NF}'  | bedtools sort -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes -i stdin > /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_cc_1.masked.dedup.sorted_peaks.narrowPeak.Th17_cc_1.masked.dups_marked.bam.Counts.bed; rm /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_cc_1.masked.dedup.sorted_peaks.narrowPeak.Th17_cc_1.masked.dups_marked.bam.Counts.bed.temp_sort_order\n",
      "Fast count method failed to count: /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th17_cc_1.masked.dups_marked.bam\n",
      "\n",
      "Error: The genome file /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_cc_1.masked.dedup.sorted_peaks.narrowPeak.Th17_cc_1.masked.dups_marked.bam.Counts.bed.temp_sort_order has no valid entries. Exiting.\n",
      "Chromosome \"chr1\" undefined in /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_cc_1.masked.dedup.sorted_peaks.narrowPeak.Th17_cc_1.masked.dups_marked.bam.Counts.bed.temp_sort_order\n",
      "\n",
      "Trying bamtobed method ...\n",
      "\n",
      "Running: bedtools bamtobed -i /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th17_cc_1.masked.dups_marked.bam | cut -f 1-3 | bedtools intersect -wa -a stdin -b /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes.bed | bedtools sort -i stdin -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes | bedtools coverage -g /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes -counts -sorted -a /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th17_cc_1.masked.dedup.sorted_peaks.narrowPeak -b stdin | awk '{print $1 \"\\t\" $2 \"\\t\" $3 \"\\t\" $NF}' > /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_cc_1.masked.dedup.sorted_peaks.narrowPeak.Th17_cc_1.masked.dups_marked.bam.Counts.bed\n",
      "BEDTools failed to count file: /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th17_cc_1.masked.dups_marked.bam\n",
      "\n",
      "Error: requested chromosome chr14_GL000194v1_random does not exist in the genome file /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes. Exiting.\n",
      "\n",
      "Running: bedtools sort -i /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_cc_1.masked.dedup.sorted_peaks.narrowPeak.Th17_cc_1.masked.dups_marked.bam.Counts.bed -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes | bedtools merge -i stdin -c 4 -o max | sort -nr -k 4 | head -n 150000 |bedtools intersect -b stdin -a /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th17_cc_1.masked.dedup.sorted_peaks.narrowPeak -wa |awk '{print $1 \"\\t\" $2 + $10 \"\\t\" $2 + $10}' |bedtools slop -i stdin -b 250 -g /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes |bedtools sort -i stdin -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes |bedtools merge -i stdin | bedtools intersect -v -wa -a stdin -b /data/reddylab/Revathy/dev/Maria/hg38_wgEncodeHg19ConsensusSignalArtifactRegions.bed | cut -f 1-3 | (bedtools intersect -a /data/reddylab/Revathy/dev/Maria/hg38_RefSeqCurated.170308.bed.CollapsedGeneBounds.TSS500bp.bed -b /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes.bed -wa | cut -f 1-3 && cat) |bedtools sort -i stdin -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes | bedtools merge -i stdin > /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_cc_1.masked.dedup.sorted_peaks.narrowPeak.candidateRegions.bed\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "mkdir -p /data/reddylab/Revathy/dev/Maria/ABC\n",
    "source /data/reddylab/software/miniconda3/bin/activate jung_py3\n",
    "module load bedtools2\n",
    "abc=/data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction\n",
    "abc_dir=/data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction\n",
    "in_dir=/data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal\n",
    "out_dir=/data/reddylab/Revathy/dev/Maria/ABC\n",
    "list=list_of_samples.txt\n",
    "samples=$(cat /data/reddylab/Revathy/dev/Maria/ABC/$list|tr '\\n' ' ')\n",
    "# Need hg38 version of whitelist and gene list in bed format\n",
    "\n",
    "#hg38_RefSeqCurated.170308.bed.CollapsedGeneBounds.bed           \n",
    "#hg38_wgEncodeHg19ConsensusSignalArtifactRegions.bed\n",
    "#hg38_RefSeqCurated.170308.bed.CollapsedGeneBounds.TSS500bp.bed\n",
    "################################################################################################################\n",
    "# 1. Count DNase-seq reads in each peak and retain the top N peaks with the most read counts\n",
    "################################################################################################################\n",
    "\n",
    "task=candidate_region_by_dnase\n",
    "mkdir -p /data/reddylab/Revathy/dev/Maria/ABC/${task}\n",
    "mkdir -p /data/reddylab/Revathy/dev/Maria/ABC/logs\n",
    "for dnase in ${samples} ;\n",
    "do\n",
    "python ${abc}/src/makeCandidateRegions.py \\\n",
    "--narrowPeak ${in_dir}/${dnase}.masked.dedup.sorted_peaks.narrowPeak \\\n",
    "--bam ${in_dir}/${dnase}.masked.dups_marked.bam \\\n",
    "--outDir /data/reddylab/Revathy/dev/Maria/ABC/${task}/ \\\n",
    "--chrom_sizes ${abc_dir}/reference/chr_sizes \\\n",
    "--regions_blacklist /data/reddylab/Revathy/dev/Maria/hg38_wgEncodeHg19ConsensusSignalArtifactRegions.bed \\\n",
    "--regions_whitelist /data/reddylab/Revathy/dev/Maria/hg38_RefSeqCurated.170308.bed.CollapsedGeneBounds.TSS500bp.bed \\\n",
    "--peakExtendFromSummit 250 \\\n",
    "--nStrongestPeaks 150000\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /data/reddylab/Revathy/dev/Maria/ABC/data\n",
    "cp *.bam *.bam.bai /data/reddylab/Jungkyun/MS_maria/results/processing/chip_seq/CIOFANI_5481_190305B1_MCS234-249-se"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /data/reddylab/Revathy/dev/Maria/ABC/data\n",
    "for file in *.bam ; \n",
    "do \n",
    "mv \"$file\" \"${file//_input_1/_1}\"\n",
    "mv \"$file\" \"${file//_input_2/_2}\"\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Quantify enhancer activity"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Th01\n",
      "Namespace(ATAC='', DHS='/data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_1.masked.dups_marked.bam', H3K27ac='/data/reddylab/Revathy/dev/Maria/ABC/data/Th01_1.masked.dedup.sorted.bam', candidate_enhancer_regions='/data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_1.masked.dedup.sorted_peaks.narrowPeak.candidateRegions.bed', cellType='Th01_abc', chrom_sizes='/data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes', default_accessibility_feature=None, enhancer_class_override=None, expression_table='/data/reddylab/Revathy/dev/Maria/ABC/data/rna_seq/Th01.TPM_Expr.txt', gene_name_annotations='symbol', genes='/data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/RefSeqCurated.170308.bed.CollapsedGeneBounds.bed', genes_for_class_assignment=None, outdir='/gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc', primary_gene_identifier='symbol', qnorm=None, skip_gene_counts=False, skip_rpkm_quantile=False, supplementary_features=None, tss_slop_for_class_assignment=500, ubiquitously_expressed_genes='/data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/UbiquitouslyExpressedGenesHG19.txt', use_secondary_counting_method=False)\n",
      "Using gene expression from files: ['/data/reddylab/Revathy/dev/Maria/ABC/data/rna_seq/Th01.TPM_Expr.txt'] \n",
      "\n",
      "Running command: bedtools sort -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes -i /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/GeneList.TSS1kb.bed > /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/GeneList.TSS1kb.bed.sorted; mv /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/GeneList.TSS1kb.bed.sorted /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/GeneList.TSS1kb.bed\n",
      "Loading coverage from pre-calculated file for Genes.H3K27ac.Th01_1.masked.dedup.sorted.bam\n",
      "Feature H3K27ac completed in 19.764678716659546\n",
      "Loading coverage from pre-calculated file for Genes.DHS.Th01_1.masked.dups_marked.bam\n",
      "Feature DHS completed in 0.3242614269256592\n",
      "Loading coverage from pre-calculated file for Genes.TSS1kb.H3K27ac.Th01_1.masked.dedup.sorted.bam\n",
      "Feature H3K27ac completed in 19.56548047065735\n",
      "Loading coverage from pre-calculated file for Genes.TSS1kb.DHS.Th01_1.masked.dups_marked.bam\n",
      "Feature DHS completed in 0.21199703216552734\n",
      "Regenerating /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/Enhancers.H3K27ac.Th01_1.masked.dedup.sorted.bam.CountReads.bedgraph\n",
      "Counting coverage for Enhancers.H3K27ac.Th01_1.masked.dedup.sorted.bam\n",
      "Running: awk 'FNR==NR {x2[$1] = $0; next} $1 in x2 {print x2[$1]}' /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes <(samtools view -H /data/reddylab/Revathy/dev/Maria/ABC/data/Th01_1.masked.dedup.sorted.bam | grep SQ | cut -f 2 | cut -c 4- )  > /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/Enhancers.H3K27ac.Th01_1.masked.dedup.sorted.bam.CountReads.bedgraph.temp_sort_order\n",
      "Running: bedtools sort -faidx /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/Enhancers.H3K27ac.Th01_1.masked.dedup.sorted.bam.CountReads.bedgraph.temp_sort_order -i /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_1.masked.dedup.sorted_peaks.narrowPeak.candidateRegions.bed | bedtools coverage -g /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/Enhancers.H3K27ac.Th01_1.masked.dedup.sorted.bam.CountReads.bedgraph.temp_sort_order -counts -sorted -a stdin -b /data/reddylab/Revathy/dev/Maria/ABC/data/Th01_1.masked.dedup.sorted.bam | awk '{print $1 \"\\t\" $2 \"\\t\" $3 \"\\t\" $NF}'  | bedtools sort -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes -i stdin > /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/Enhancers.H3K27ac.Th01_1.masked.dedup.sorted.bam.CountReads.bedgraph; rm /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/Enhancers.H3K27ac.Th01_1.masked.dedup.sorted.bam.CountReads.bedgraph.temp_sort_order\n",
      "BEDTools completed successfully. \n",
      "\n",
      "Feature H3K27ac completed in 58.66090679168701\n",
      "Regenerating /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/Enhancers.DHS.Th01_1.masked.dups_marked.bam.CountReads.bedgraph\n",
      "Counting coverage for Enhancers.DHS.Th01_1.masked.dups_marked.bam\n",
      "Running: awk 'FNR==NR {x2[$1] = $0; next} $1 in x2 {print x2[$1]}' /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes <(samtools view -H /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_1.masked.dups_marked.bam | grep SQ | cut -f 2 | cut -c 4- )  > /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/Enhancers.DHS.Th01_1.masked.dups_marked.bam.CountReads.bedgraph.temp_sort_order\n",
      "Running: bedtools sort -faidx /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/Enhancers.DHS.Th01_1.masked.dups_marked.bam.CountReads.bedgraph.temp_sort_order -i /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_1.masked.dedup.sorted_peaks.narrowPeak.candidateRegions.bed | bedtools coverage -g /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/Enhancers.DHS.Th01_1.masked.dups_marked.bam.CountReads.bedgraph.temp_sort_order -counts -sorted -a stdin -b /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_1.masked.dups_marked.bam | awk '{print $1 \"\\t\" $2 \"\\t\" $3 \"\\t\" $NF}'  | bedtools sort -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes -i stdin > /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/Enhancers.DHS.Th01_1.masked.dups_marked.bam.CountReads.bedgraph; rm /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/Enhancers.DHS.Th01_1.masked.dups_marked.bam.CountReads.bedgraph.temp_sort_order\n",
      "BEDTools completed successfully. \n",
      "\n",
      "Feature DHS completed in 95.8973798751831\n",
      "Assigning classes to enhancers\n",
      "Total enhancers: 50650\n",
      "         Promoters: 1678\n",
      "         Genic: 23498\n",
      "         Intergenic: 25474\n",
      "Neighborhoods Complete! \n",
      "\n",
      "Th01\n",
      "Namespace(ATAC='', DHS='/data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_2.masked.dups_marked.bam', H3K27ac='/data/reddylab/Revathy/dev/Maria/ABC/data/Th01_2.masked.dedup.sorted.bam', candidate_enhancer_regions='/data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_2.masked.dedup.sorted_peaks.narrowPeak.candidateRegions.bed', cellType='Th01_abc', chrom_sizes='/data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes', default_accessibility_feature=None, enhancer_class_override=None, expression_table='/data/reddylab/Revathy/dev/Maria/ABC/data/rna_seq/Th01.TPM_Expr.txt', gene_name_annotations='symbol', genes='/data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/RefSeqCurated.170308.bed.CollapsedGeneBounds.bed', genes_for_class_assignment=None, outdir='/gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc', primary_gene_identifier='symbol', qnorm=None, skip_gene_counts=False, skip_rpkm_quantile=False, supplementary_features=None, tss_slop_for_class_assignment=500, ubiquitously_expressed_genes='/data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/UbiquitouslyExpressedGenesHG19.txt', use_secondary_counting_method=False)\n",
      "Using gene expression from files: ['/data/reddylab/Revathy/dev/Maria/ABC/data/rna_seq/Th01.TPM_Expr.txt'] \n",
      "\n",
      "Running command: bedtools sort -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes -i /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/GeneList.TSS1kb.bed > /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/GeneList.TSS1kb.bed.sorted; mv /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/GeneList.TSS1kb.bed.sorted /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/GeneList.TSS1kb.bed\n",
      "Loading coverage from pre-calculated file for Genes.H3K27ac.Th01_2.masked.dedup.sorted.bam\n",
      "Feature H3K27ac completed in 23.909016847610474\n",
      "Loading coverage from pre-calculated file for Genes.DHS.Th01_2.masked.dups_marked.bam\n",
      "Feature DHS completed in 0.3666038513183594\n",
      "Loading coverage from pre-calculated file for Genes.TSS1kb.H3K27ac.Th01_2.masked.dedup.sorted.bam\n",
      "Feature H3K27ac completed in 23.755717277526855\n",
      "Loading coverage from pre-calculated file for Genes.TSS1kb.DHS.Th01_2.masked.dups_marked.bam\n",
      "Feature DHS completed in 0.2169022560119629\n",
      "Regenerating /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/Enhancers.H3K27ac.Th01_2.masked.dedup.sorted.bam.CountReads.bedgraph\n",
      "Counting coverage for Enhancers.H3K27ac.Th01_2.masked.dedup.sorted.bam\n",
      "Running: awk 'FNR==NR {x2[$1] = $0; next} $1 in x2 {print x2[$1]}' /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes <(samtools view -H /data/reddylab/Revathy/dev/Maria/ABC/data/Th01_2.masked.dedup.sorted.bam | grep SQ | cut -f 2 | cut -c 4- )  > /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/Enhancers.H3K27ac.Th01_2.masked.dedup.sorted.bam.CountReads.bedgraph.temp_sort_order\n",
      "Running: bedtools sort -faidx /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/Enhancers.H3K27ac.Th01_2.masked.dedup.sorted.bam.CountReads.bedgraph.temp_sort_order -i /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_2.masked.dedup.sorted_peaks.narrowPeak.candidateRegions.bed | bedtools coverage -g /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/Enhancers.H3K27ac.Th01_2.masked.dedup.sorted.bam.CountReads.bedgraph.temp_sort_order -counts -sorted -a stdin -b /data/reddylab/Revathy/dev/Maria/ABC/data/Th01_2.masked.dedup.sorted.bam | awk '{print $1 \"\\t\" $2 \"\\t\" $3 \"\\t\" $NF}'  | bedtools sort -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes -i stdin > /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/Enhancers.H3K27ac.Th01_2.masked.dedup.sorted.bam.CountReads.bedgraph; rm /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/Enhancers.H3K27ac.Th01_2.masked.dedup.sorted.bam.CountReads.bedgraph.temp_sort_order\n",
      "BEDTools completed successfully. \n",
      "\n",
      "Feature H3K27ac completed in 70.87147355079651\n",
      "Regenerating /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/Enhancers.DHS.Th01_2.masked.dups_marked.bam.CountReads.bedgraph\n",
      "Counting coverage for Enhancers.DHS.Th01_2.masked.dups_marked.bam\n",
      "Running: awk 'FNR==NR {x2[$1] = $0; next} $1 in x2 {print x2[$1]}' /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes <(samtools view -H /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_2.masked.dups_marked.bam | grep SQ | cut -f 2 | cut -c 4- )  > /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/Enhancers.DHS.Th01_2.masked.dups_marked.bam.CountReads.bedgraph.temp_sort_order\n",
      "Running: bedtools sort -faidx /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/Enhancers.DHS.Th01_2.masked.dups_marked.bam.CountReads.bedgraph.temp_sort_order -i /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th01_2.masked.dedup.sorted_peaks.narrowPeak.candidateRegions.bed | bedtools coverage -g /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/Enhancers.DHS.Th01_2.masked.dups_marked.bam.CountReads.bedgraph.temp_sort_order -counts -sorted -a stdin -b /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th01_2.masked.dups_marked.bam | awk '{print $1 \"\\t\" $2 \"\\t\" $3 \"\\t\" $NF}'  | bedtools sort -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes -i stdin > /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/Enhancers.DHS.Th01_2.masked.dups_marked.bam.CountReads.bedgraph; rm /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th01_abc/Enhancers.DHS.Th01_2.masked.dups_marked.bam.CountReads.bedgraph.temp_sort_order\n",
      "BEDTools completed successfully. \n",
      "\n",
      "Feature DHS completed in 93.64930939674377\n",
      "Assigning classes to enhancers\n",
      "Total enhancers: 48454\n",
      "         Promoters: 1647\n",
      "         Genic: 22387\n",
      "         Intergenic: 24420\n",
      "Neighborhoods Complete! \n",
      "\n",
      "Th17\n",
      "Namespace(ATAC='', DHS='/data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th17_1.masked.dups_marked.bam', H3K27ac='/data/reddylab/Revathy/dev/Maria/ABC/data/Th17_1.masked.dedup.sorted.bam', candidate_enhancer_regions='/data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_1.masked.dedup.sorted_peaks.narrowPeak.candidateRegions.bed', cellType='Th17_abc', chrom_sizes='/data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes', default_accessibility_feature=None, enhancer_class_override=None, expression_table='/data/reddylab/Revathy/dev/Maria/ABC/data/rna_seq/Th17.TPM_Expr.txt', gene_name_annotations='symbol', genes='/data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/RefSeqCurated.170308.bed.CollapsedGeneBounds.bed', genes_for_class_assignment=None, outdir='/gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc', primary_gene_identifier='symbol', qnorm=None, skip_gene_counts=False, skip_rpkm_quantile=False, supplementary_features=None, tss_slop_for_class_assignment=500, ubiquitously_expressed_genes='/data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/UbiquitouslyExpressedGenesHG19.txt', use_secondary_counting_method=False)\n",
      "Using gene expression from files: ['/data/reddylab/Revathy/dev/Maria/ABC/data/rna_seq/Th17.TPM_Expr.txt'] \n",
      "\n",
      "Running command: bedtools sort -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes -i /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/GeneList.TSS1kb.bed > /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/GeneList.TSS1kb.bed.sorted; mv /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/GeneList.TSS1kb.bed.sorted /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/GeneList.TSS1kb.bed\n",
      "Loading coverage from pre-calculated file for Genes.H3K27ac.Th17_1.masked.dedup.sorted.bam\n",
      "Feature H3K27ac completed in 25.734113931655884\n",
      "Loading coverage from pre-calculated file for Genes.DHS.Th17_1.masked.dups_marked.bam\n",
      "Feature DHS completed in 0.33610057830810547\n",
      "Loading coverage from pre-calculated file for Genes.TSS1kb.H3K27ac.Th17_1.masked.dedup.sorted.bam\n",
      "Feature H3K27ac completed in 24.974273443222046\n",
      "Loading coverage from pre-calculated file for Genes.TSS1kb.DHS.Th17_1.masked.dups_marked.bam\n",
      "Feature DHS completed in 0.20738577842712402\n",
      "Regenerating /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/Enhancers.H3K27ac.Th17_1.masked.dedup.sorted.bam.CountReads.bedgraph\n",
      "Counting coverage for Enhancers.H3K27ac.Th17_1.masked.dedup.sorted.bam\n",
      "Running: awk 'FNR==NR {x2[$1] = $0; next} $1 in x2 {print x2[$1]}' /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes <(samtools view -H /data/reddylab/Revathy/dev/Maria/ABC/data/Th17_1.masked.dedup.sorted.bam | grep SQ | cut -f 2 | cut -c 4- )  > /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/Enhancers.H3K27ac.Th17_1.masked.dedup.sorted.bam.CountReads.bedgraph.temp_sort_order\n",
      "Running: bedtools sort -faidx /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/Enhancers.H3K27ac.Th17_1.masked.dedup.sorted.bam.CountReads.bedgraph.temp_sort_order -i /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_1.masked.dedup.sorted_peaks.narrowPeak.candidateRegions.bed | bedtools coverage -g /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/Enhancers.H3K27ac.Th17_1.masked.dedup.sorted.bam.CountReads.bedgraph.temp_sort_order -counts -sorted -a stdin -b /data/reddylab/Revathy/dev/Maria/ABC/data/Th17_1.masked.dedup.sorted.bam | awk '{print $1 \"\\t\" $2 \"\\t\" $3 \"\\t\" $NF}'  | bedtools sort -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes -i stdin > /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/Enhancers.H3K27ac.Th17_1.masked.dedup.sorted.bam.CountReads.bedgraph; rm /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/Enhancers.H3K27ac.Th17_1.masked.dedup.sorted.bam.CountReads.bedgraph.temp_sort_order\n",
      "BEDTools completed successfully. \n",
      "\n",
      "Feature H3K27ac completed in 73.95325016975403\n",
      "Regenerating /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/Enhancers.DHS.Th17_1.masked.dups_marked.bam.CountReads.bedgraph\n",
      "Counting coverage for Enhancers.DHS.Th17_1.masked.dups_marked.bam\n",
      "Running: awk 'FNR==NR {x2[$1] = $0; next} $1 in x2 {print x2[$1]}' /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes <(samtools view -H /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th17_1.masked.dups_marked.bam | grep SQ | cut -f 2 | cut -c 4- )  > /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/Enhancers.DHS.Th17_1.masked.dups_marked.bam.CountReads.bedgraph.temp_sort_order\n",
      "Running: bedtools sort -faidx /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/Enhancers.DHS.Th17_1.masked.dups_marked.bam.CountReads.bedgraph.temp_sort_order -i /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_1.masked.dedup.sorted_peaks.narrowPeak.candidateRegions.bed | bedtools coverage -g /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/Enhancers.DHS.Th17_1.masked.dups_marked.bam.CountReads.bedgraph.temp_sort_order -counts -sorted -a stdin -b /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th17_1.masked.dups_marked.bam | awk '{print $1 \"\\t\" $2 \"\\t\" $3 \"\\t\" $NF}'  | bedtools sort -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes -i stdin > /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/Enhancers.DHS.Th17_1.masked.dups_marked.bam.CountReads.bedgraph; rm /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/Enhancers.DHS.Th17_1.masked.dups_marked.bam.CountReads.bedgraph.temp_sort_order\n",
      "BEDTools completed successfully. \n",
      "\n",
      "Feature DHS completed in 83.54709267616272\n",
      "Assigning classes to enhancers\n",
      "Total enhancers: 36043\n",
      "         Promoters: 1383\n",
      "         Genic: 16087\n",
      "         Intergenic: 18573\n",
      "Neighborhoods Complete! \n",
      "\n",
      "Th17\n",
      "Namespace(ATAC='', DHS='/data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th17_2.masked.dups_marked.bam', H3K27ac='/data/reddylab/Revathy/dev/Maria/ABC/data/Th17_2.masked.dedup.sorted.bam', candidate_enhancer_regions='/data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_2.masked.dedup.sorted_peaks.narrowPeak.candidateRegions.bed', cellType='Th17_abc', chrom_sizes='/data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes', default_accessibility_feature=None, enhancer_class_override=None, expression_table='/data/reddylab/Revathy/dev/Maria/ABC/data/rna_seq/Th17.TPM_Expr.txt', gene_name_annotations='symbol', genes='/data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/RefSeqCurated.170308.bed.CollapsedGeneBounds.bed', genes_for_class_assignment=None, outdir='/gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc', primary_gene_identifier='symbol', qnorm=None, skip_gene_counts=False, skip_rpkm_quantile=False, supplementary_features=None, tss_slop_for_class_assignment=500, ubiquitously_expressed_genes='/data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/UbiquitouslyExpressedGenesHG19.txt', use_secondary_counting_method=False)\n",
      "Using gene expression from files: ['/data/reddylab/Revathy/dev/Maria/ABC/data/rna_seq/Th17.TPM_Expr.txt'] \n",
      "\n",
      "Running command: bedtools sort -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes -i /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/GeneList.TSS1kb.bed > /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/GeneList.TSS1kb.bed.sorted; mv /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/GeneList.TSS1kb.bed.sorted /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/GeneList.TSS1kb.bed\n",
      "Loading coverage from pre-calculated file for Genes.H3K27ac.Th17_2.masked.dedup.sorted.bam\n",
      "Feature H3K27ac completed in 26.15342402458191\n",
      "Loading coverage from pre-calculated file for Genes.DHS.Th17_2.masked.dups_marked.bam\n",
      "Feature DHS completed in 0.32222795486450195\n",
      "Loading coverage from pre-calculated file for Genes.TSS1kb.H3K27ac.Th17_2.masked.dedup.sorted.bam\n",
      "Feature H3K27ac completed in 25.869748830795288\n",
      "Loading coverage from pre-calculated file for Genes.TSS1kb.DHS.Th17_2.masked.dups_marked.bam\n",
      "Feature DHS completed in 0.20093822479248047\n",
      "Regenerating /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/Enhancers.H3K27ac.Th17_2.masked.dedup.sorted.bam.CountReads.bedgraph\n",
      "Counting coverage for Enhancers.H3K27ac.Th17_2.masked.dedup.sorted.bam\n",
      "Running: awk 'FNR==NR {x2[$1] = $0; next} $1 in x2 {print x2[$1]}' /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes <(samtools view -H /data/reddylab/Revathy/dev/Maria/ABC/data/Th17_2.masked.dedup.sorted.bam | grep SQ | cut -f 2 | cut -c 4- )  > /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/Enhancers.H3K27ac.Th17_2.masked.dedup.sorted.bam.CountReads.bedgraph.temp_sort_order\n",
      "Running: bedtools sort -faidx /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/Enhancers.H3K27ac.Th17_2.masked.dedup.sorted.bam.CountReads.bedgraph.temp_sort_order -i /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_2.masked.dedup.sorted_peaks.narrowPeak.candidateRegions.bed | bedtools coverage -g /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/Enhancers.H3K27ac.Th17_2.masked.dedup.sorted.bam.CountReads.bedgraph.temp_sort_order -counts -sorted -a stdin -b /data/reddylab/Revathy/dev/Maria/ABC/data/Th17_2.masked.dedup.sorted.bam | awk '{print $1 \"\\t\" $2 \"\\t\" $3 \"\\t\" $NF}'  | bedtools sort -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes -i stdin > /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/Enhancers.H3K27ac.Th17_2.masked.dedup.sorted.bam.CountReads.bedgraph; rm /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/Enhancers.H3K27ac.Th17_2.masked.dedup.sorted.bam.CountReads.bedgraph.temp_sort_order\n",
      "BEDTools completed successfully. \n",
      "\n",
      "Feature H3K27ac completed in 76.94649028778076\n",
      "Regenerating /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/Enhancers.DHS.Th17_2.masked.dups_marked.bam.CountReads.bedgraph\n",
      "Counting coverage for Enhancers.DHS.Th17_2.masked.dups_marked.bam\n",
      "Running: awk 'FNR==NR {x2[$1] = $0; next} $1 in x2 {print x2[$1]}' /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes <(samtools view -H /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th17_2.masked.dups_marked.bam | grep SQ | cut -f 2 | cut -c 4- )  > /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/Enhancers.DHS.Th17_2.masked.dups_marked.bam.CountReads.bedgraph.temp_sort_order\n",
      "Running: bedtools sort -faidx /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/Enhancers.DHS.Th17_2.masked.dups_marked.bam.CountReads.bedgraph.temp_sort_order -i /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/Th17_2.masked.dedup.sorted_peaks.narrowPeak.candidateRegions.bed | bedtools coverage -g /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/Enhancers.DHS.Th17_2.masked.dups_marked.bam.CountReads.bedgraph.temp_sort_order -counts -sorted -a stdin -b /data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal/Th17_2.masked.dups_marked.bam | awk '{print $1 \"\\t\" $2 \"\\t\" $3 \"\\t\" $NF}'  | bedtools sort -faidx /data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction/reference/chr_sizes -i stdin > /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/Enhancers.DHS.Th17_2.masked.dups_marked.bam.CountReads.bedgraph; rm /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/enhancer_acitivity/Th17_abc/Enhancers.DHS.Th17_2.masked.dups_marked.bam.CountReads.bedgraph.temp_sort_order\n",
      "BEDTools completed successfully. \n",
      "\n",
      "Feature DHS completed in 98.4563398361206\n",
      "Assigning classes to enhancers\n",
      "Total enhancers: 48972\n",
      "         Promoters: 1653\n",
      "         Genic: 22662\n",
      "         Intergenic: 24657\n",
      "Neighborhoods Complete! \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[E::idx_find_and_load] Could not retrieve index file for '/data/reddylab/Revathy/dev/Maria/ABC/data/Th01_1.masked.dedup.sorted.bam'\n",
      "samtools idxstats: fail to load index for \"/data/reddylab/Revathy/dev/Maria/ABC/data/Th01_1.masked.dedup.sorted.bam\", reverting to slow method\n",
      "[E::idx_find_and_load] Could not retrieve index file for '/data/reddylab/Revathy/dev/Maria/ABC/data/Th01_1.masked.dedup.sorted.bam'\n",
      "samtools idxstats: fail to load index for \"/data/reddylab/Revathy/dev/Maria/ABC/data/Th01_1.masked.dedup.sorted.bam\", reverting to slow method\n",
      "[E::idx_find_and_load] Could not retrieve index file for '/data/reddylab/Revathy/dev/Maria/ABC/data/Th01_1.masked.dedup.sorted.bam'\n",
      "samtools idxstats: fail to load index for \"/data/reddylab/Revathy/dev/Maria/ABC/data/Th01_1.masked.dedup.sorted.bam\", reverting to slow method\n",
      "[E::idx_find_and_load] Could not retrieve index file for '/data/reddylab/Revathy/dev/Maria/ABC/data/Th01_2.masked.dedup.sorted.bam'\n",
      "samtools idxstats: fail to load index for \"/data/reddylab/Revathy/dev/Maria/ABC/data/Th01_2.masked.dedup.sorted.bam\", reverting to slow method\n",
      "[E::idx_find_and_load] Could not retrieve index file for '/data/reddylab/Revathy/dev/Maria/ABC/data/Th01_2.masked.dedup.sorted.bam'\n",
      "samtools idxstats: fail to load index for \"/data/reddylab/Revathy/dev/Maria/ABC/data/Th01_2.masked.dedup.sorted.bam\", reverting to slow method\n",
      "[E::idx_find_and_load] Could not retrieve index file for '/data/reddylab/Revathy/dev/Maria/ABC/data/Th01_2.masked.dedup.sorted.bam'\n",
      "samtools idxstats: fail to load index for \"/data/reddylab/Revathy/dev/Maria/ABC/data/Th01_2.masked.dedup.sorted.bam\", reverting to slow method\n",
      "[E::idx_find_and_load] Could not retrieve index file for '/data/reddylab/Revathy/dev/Maria/ABC/data/Th17_1.masked.dedup.sorted.bam'\n",
      "samtools idxstats: fail to load index for \"/data/reddylab/Revathy/dev/Maria/ABC/data/Th17_1.masked.dedup.sorted.bam\", reverting to slow method\n",
      "[E::idx_find_and_load] Could not retrieve index file for '/data/reddylab/Revathy/dev/Maria/ABC/data/Th17_1.masked.dedup.sorted.bam'\n",
      "samtools idxstats: fail to load index for \"/data/reddylab/Revathy/dev/Maria/ABC/data/Th17_1.masked.dedup.sorted.bam\", reverting to slow method\n",
      "[E::idx_find_and_load] Could not retrieve index file for '/data/reddylab/Revathy/dev/Maria/ABC/data/Th17_1.masked.dedup.sorted.bam'\n",
      "samtools idxstats: fail to load index for \"/data/reddylab/Revathy/dev/Maria/ABC/data/Th17_1.masked.dedup.sorted.bam\", reverting to slow method\n",
      "[E::idx_find_and_load] Could not retrieve index file for '/data/reddylab/Revathy/dev/Maria/ABC/data/Th17_2.masked.dedup.sorted.bam'\n",
      "samtools idxstats: fail to load index for \"/data/reddylab/Revathy/dev/Maria/ABC/data/Th17_2.masked.dedup.sorted.bam\", reverting to slow method\n",
      "[E::idx_find_and_load] Could not retrieve index file for '/data/reddylab/Revathy/dev/Maria/ABC/data/Th17_2.masked.dedup.sorted.bam'\n",
      "samtools idxstats: fail to load index for \"/data/reddylab/Revathy/dev/Maria/ABC/data/Th17_2.masked.dedup.sorted.bam\", reverting to slow method\n",
      "[E::idx_find_and_load] Could not retrieve index file for '/data/reddylab/Revathy/dev/Maria/ABC/data/Th17_2.masked.dedup.sorted.bam'\n",
      "samtools idxstats: fail to load index for \"/data/reddylab/Revathy/dev/Maria/ABC/data/Th17_2.masked.dedup.sorted.bam\", reverting to slow method\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "task=enhancer_acitivity\n",
    "mkdir -p /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/${task}\n",
    "source /data/reddylab/software/miniconda3/bin/activate jung_py3\n",
    "module load bedtools2\n",
    "module load samtools\n",
    "H3K27ac_dir=/data/reddylab/Revathy/dev/Maria/ABC/data\n",
    "dnase_dir=/data/reddylab/Jungkyun/MS_maria/results/processing/atac_seq/Ciofani_5484_190307A5_MCS226-233_258_259-pe-blacklist-removal\n",
    "abc=/data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction\n",
    "abc_dir=/data/reddylab/Revathy/dev/Maria/ABC/ABC-Enhancer-Gene-Prediction\n",
    "gene_ex_dir=/data/reddylab/Revathy/dev/Maria/ABC/data/rna_seq\n",
    "LIST=sample_names.txt\n",
    "SAMPLES=$(cat /data/reddylab/Revathy/dev/Maria/ABC/$LIST|tr '\\n' ' ')\n",
    "\n",
    "for sample in ${SAMPLES};\n",
    "do\n",
    "\n",
    "idx=$(echo ${sample}|cut -d \"_\" -f1)\n",
    "echo ${idx}\n",
    "mkdir -p /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/${task}/${idx}_abc\n",
    "\n",
    "python ${abc}/src/run.neighborhoods.py \\\n",
    "--candidate_enhancer_regions /data/reddylab/Revathy/dev/Maria/ABC/candidate_region_by_dnase/${sample}.masked.dedup.sorted_peaks.narrowPeak.candidateRegions.bed \\\n",
    "--genes ${abc_dir}/reference/RefSeqCurated.170308.bed.CollapsedGeneBounds.bed \\\n",
    "--H3K27ac ${H3K27ac_dir}/${sample}.masked.dedup.sorted.bam \\\n",
    "--DHS ${dnase_dir}/${sample}.masked.dups_marked.bam \\\n",
    "--expression_table /data/reddylab/Revathy/dev/Maria/ABC/data/rna_seq/${idx}.TPM_Expr.txt \\\n",
    "--chrom_sizes ${abc_dir}/reference/chr_sizes \\\n",
    "--ubiquitously_expressed_genes ${abc_dir}/reference/UbiquitouslyExpressedGenesHG19.txt \\\n",
    "--cellType ${idx}_abc \\\n",
    "--outdir /gpfs/fs1/data/reddylab/Revathy/dev/Maria/ABC/${task}/${idx}_abc \n",
    "\n",
    "done\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "HiC files are pre-processed using Juicer, can be found in /data/reddylab/Revathy/dev/Maria/HiC_preprocessing.ipynb"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Compute ABC scores"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "reading genes\n",
      "reading enhancers\n",
      "Making predictions for chromosome: chr22\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th01/chr22/chr22.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 0.9547169208526611\n",
      "HiC Matrix has row sums of 1669.5396348412314, making doubly stochastic...\n",
      "HiC has 1399603 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 2.2869439125061035\n",
      "HiC added to predictions table. Elapsed time: 0.37697601318359375\n",
      "HiC Complete\n",
      "Completed chromosome: chr22. Elapsed time: 4.167795658111572 \n",
      "\n",
      "Making predictions for chromosome: chr1\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th01/chr1/chr1.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 8.680987119674683\n",
      "HiC Matrix has row sums of 1821.457072176637, making doubly stochastic...\n",
      "HiC has 9718374 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 18.089518070220947\n",
      "HiC added to predictions table. Elapsed time: 4.740597248077393\n",
      "HiC Complete\n",
      "Completed chromosome: chr1. Elapsed time: 34.64890098571777 \n",
      "\n",
      "Making predictions for chromosome: chr15\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th01/chr15/chr15.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 2.550659656524658\n",
      "HiC Matrix has row sums of 1700.5841866727549, making doubly stochastic...\n",
      "HiC has 3350294 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 5.8311076164245605\n",
      "HiC added to predictions table. Elapsed time: 1.1859140396118164\n",
      "HiC Complete\n",
      "Completed chromosome: chr15. Elapsed time: 10.259731531143188 \n",
      "\n",
      "Making predictions for chromosome: chr20\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th01/chr20/chr20.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 1.994340181350708\n",
      "HiC Matrix has row sums of 1711.636629811376, making doubly stochastic...\n",
      "HiC has 2710341 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 4.477664470672607\n",
      "HiC added to predictions table. Elapsed time: 0.7996923923492432\n",
      "HiC Complete\n",
      "Completed chromosome: chr20. Elapsed time: 7.840063571929932 \n",
      "\n",
      "Making predictions for chromosome: chr11\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th01/chr11/chr11.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 5.616123199462891\n",
      "HiC Matrix has row sums of 1718.4666921751407, making doubly stochastic...\n",
      "HiC has 5644661 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 10.28699016571045\n",
      "HiC added to predictions table. Elapsed time: 2.402630567550659\n",
      "HiC Complete\n",
      "Completed chromosome: chr11. Elapsed time: 20.185552835464478 \n",
      "\n",
      "Making predictions for chromosome: chr3\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th01/chr3/chr3.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 8.328439235687256\n",
      "HiC Matrix has row sums of 1777.7829211558487, making doubly stochastic...\n",
      "HiC has 8765088 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 16.04148006439209\n",
      "HiC added to predictions table. Elapsed time: 3.574261426925659\n",
      "HiC Complete\n",
      "Completed chromosome: chr3. Elapsed time: 30.013935804367065 \n",
      "\n",
      "Making predictions for chromosome: chr18\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th01/chr18/chr18.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 2.6987690925598145\n",
      "HiC Matrix has row sums of 1586.2914158939402, making doubly stochastic...\n",
      "HiC has 3201448 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 5.680152654647827\n",
      "HiC added to predictions table. Elapsed time: 0.8589756488800049\n",
      "HiC Complete\n",
      "Completed chromosome: chr18. Elapsed time: 9.909788608551025 \n",
      "\n",
      "Making predictions for chromosome: chr6\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th01/chr6/chr6.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 6.887752532958984\n",
      "HiC Matrix has row sums of 1756.3692637008692, making doubly stochastic...\n",
      "HiC has 7461927 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 13.63762378692627\n",
      "HiC added to predictions table. Elapsed time: 3.2331974506378174\n",
      "HiC Complete\n",
      "Completed chromosome: chr6. Elapsed time: 25.591208934783936 \n",
      "\n",
      "Making predictions for chromosome: chr2\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th01/chr2/chr2.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 9.966347455978394\n",
      "HiC Matrix has row sums of 1784.1953619122014, making doubly stochastic...\n",
      "HiC has 10239912 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 19.78471612930298\n",
      "HiC added to predictions table. Elapsed time: 4.043871641159058\n",
      "HiC Complete\n",
      "Completed chromosome: chr2. Elapsed time: 36.24546480178833 \n",
      "\n",
      "Making predictions for chromosome: chr17\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th01/chr17/chr17.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 2.51920747756958\n",
      "HiC Matrix has row sums of 1742.3018261408577, making doubly stochastic...\n",
      "HiC has 3166013 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 5.63075590133667\n",
      "HiC added to predictions table. Elapsed time: 1.332564353942871\n",
      "HiC Complete\n",
      "Completed chromosome: chr17. Elapsed time: 10.34425139427185 \n",
      "\n",
      "Making predictions for chromosome: chr14\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th01/chr14/chr14.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 3.2229676246643066\n",
      "HiC Matrix has row sums of 1707.215499012547, making doubly stochastic...\n",
      "HiC has 3911041 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 6.3689539432525635\n",
      "HiC added to predictions table. Elapsed time: 1.5917332172393799\n",
      "HiC Complete\n",
      "Completed chromosome: chr14. Elapsed time: 12.225194692611694 \n",
      "\n",
      "Making predictions for chromosome: chr12\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th01/chr12/chr12.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 5.179730176925659\n",
      "HiC Matrix has row sums of 1760.9094632563801, making doubly stochastic...\n",
      "HiC has 5851233 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 10.293895483016968\n",
      "HiC added to predictions table. Elapsed time: 2.3303587436676025\n",
      "HiC Complete\n",
      "Completed chromosome: chr12. Elapsed time: 19.486316204071045 \n",
      "\n",
      "Making predictions for chromosome: chr19\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th01/chr19/chr19.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 1.696004867553711\n",
      "HiC Matrix has row sums of 1771.123374264509, making doubly stochastic...\n",
      "HiC has 2413972 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 3.8413403034210205\n",
      "HiC added to predictions table. Elapsed time: 1.1094412803649902\n",
      "HiC Complete\n",
      "Completed chromosome: chr19. Elapsed time: 7.663561582565308 \n",
      "\n",
      "Making predictions for chromosome: chr13\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th01/chr13/chr13.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 3.7369606494903564\n",
      "HiC Matrix has row sums of 1587.994570594416, making doubly stochastic...\n",
      "HiC has 4202560 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 7.4319727420806885\n",
      "HiC added to predictions table. Elapsed time: 1.5038387775421143\n",
      "HiC Complete\n",
      "Completed chromosome: chr13. Elapsed time: 13.65760087966919 \n",
      "\n",
      "Making predictions for chromosome: chr9\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th01/chr9/chr9.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 3.830440044403076\n",
      "HiC Matrix has row sums of 1626.9624805381495, making doubly stochastic...\n",
      "HiC has 4888634 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 8.45158863067627\n",
      "HiC added to predictions table. Elapsed time: 1.710240364074707\n",
      "HiC Complete\n",
      "Completed chromosome: chr9. Elapsed time: 15.02242374420166 \n",
      "\n",
      "Making predictions for chromosome: chr4\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th01/chr4/chr4.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 7.617102384567261\n",
      "HiC Matrix has row sums of 1623.1186781733436, making doubly stochastic...\n",
      "HiC has 7967834 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 15.215095281600952\n",
      "HiC added to predictions table. Elapsed time: 3.018317461013794\n",
      "HiC Complete\n",
      "Completed chromosome: chr4. Elapsed time: 27.680612087249756 \n",
      "\n",
      "Making predictions for chromosome: chr8\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th01/chr8/chr8.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 5.96468710899353\n",
      "HiC Matrix has row sums of 1697.2077205693502, making doubly stochastic...\n",
      "HiC has 6101265 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 11.39745831489563\n",
      "HiC added to predictions table. Elapsed time: 2.009514093399048\n",
      "HiC Complete\n",
      "Completed chromosome: chr8. Elapsed time: 20.76465082168579 \n",
      "\n",
      "Making predictions for chromosome: chrX\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th01/chrX/chrX.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 4.378861904144287\n",
      "HiC Matrix has row sums of 1107.965440437105, making doubly stochastic...\n",
      "HiC has 5000671 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 10.595508337020874\n",
      "HiC added to predictions table. Elapsed time: 1.7283008098602295\n",
      "HiC Complete\n",
      "Completed chromosome: chrX. Elapsed time: 17.810474634170532 \n",
      "\n",
      "Making predictions for chromosome: chr21\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th01/chr21/chr21.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 0.904597282409668\n",
      "HiC Matrix has row sums of 1434.0243901188376, making doubly stochastic...\n",
      "HiC has 1445668 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 2.2500083446502686\n",
      "HiC added to predictions table. Elapsed time: 0.32821083068847656\n",
      "HiC Complete\n",
      "Completed chromosome: chr21. Elapsed time: 3.7603182792663574 \n",
      "\n",
      "Making predictions for chromosome: chr5\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th01/chr5/chr5.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 7.169315338134766\n",
      "HiC Matrix has row sums of 1697.4367414618423, making doubly stochastic...\n",
      "HiC has 7761556 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 14.18867039680481\n",
      "HiC added to predictions table. Elapsed time: 3.1603620052337646\n",
      "HiC Complete\n",
      "Completed chromosome: chr5. Elapsed time: 26.266308784484863 \n",
      "\n",
      "Making predictions for chromosome: chr7\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th01/chr7/chr7.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 6.10869836807251\n",
      "HiC Matrix has row sums of 1626.1438683648717, making doubly stochastic...\n",
      "HiC has 6382703 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 12.184859037399292\n",
      "HiC added to predictions table. Elapsed time: 2.730220079421997\n",
      "HiC Complete\n",
      "Completed chromosome: chr7. Elapsed time: 22.52267575263977 \n",
      "\n",
      "Making predictions for chromosome: chr10\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th01/chr10/chr10.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 5.264261245727539\n",
      "HiC Matrix has row sums of 1728.315837839723, making doubly stochastic...\n",
      "HiC has 5664741 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 10.327777624130249\n",
      "HiC added to predictions table. Elapsed time: 2.078084707260132\n",
      "HiC Complete\n",
      "Completed chromosome: chr10. Elapsed time: 19.13106060028076 \n",
      "\n",
      "Making predictions for chromosome: chr16\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th01/chr16/chr16.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 2.4166016578674316\n",
      "HiC Matrix has row sums of 1633.3944067329098, making doubly stochastic...\n",
      "HiC has 3090105 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 5.612248182296753\n",
      "HiC added to predictions table. Elapsed time: 0.9386165142059326\n",
      "HiC Complete\n",
      "Completed chromosome: chr16. Elapsed time: 9.718830347061157 \n",
      "\n",
      "Writing output files...\n",
      "Done.\n",
      "reading genes\n",
      "reading enhancers\n",
      "Making predictions for chromosome: chr9\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th17/chr9/chr9.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 3.723721504211426\n",
      "HiC Matrix has row sums of 1390.3454375769775, making doubly stochastic...\n",
      "HiC has 4438665 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 7.8922741413116455\n",
      "HiC added to predictions table. Elapsed time: 1.5816800594329834\n",
      "HiC Complete\n",
      "Completed chromosome: chr9. Elapsed time: 14.349733114242554 \n",
      "\n",
      "Making predictions for chromosome: chr17\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th17/chr17/chr17.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 2.1438519954681396\n",
      "HiC Matrix has row sums of 1483.2684323072097, making doubly stochastic...\n",
      "HiC has 2897231 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 5.178173303604126\n",
      "HiC added to predictions table. Elapsed time: 1.0116078853607178\n",
      "HiC Complete\n",
      "Completed chromosome: chr17. Elapsed time: 9.157618284225464 \n",
      "\n",
      "Making predictions for chromosome: chr15\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th17/chr15/chr15.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 2.286987543106079\n",
      "HiC Matrix has row sums of 1445.3765565881868, making doubly stochastic...\n",
      "HiC has 3091119 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 5.376974821090698\n",
      "HiC added to predictions table. Elapsed time: 0.868840217590332\n",
      "HiC Complete\n",
      "Completed chromosome: chr15. Elapsed time: 9.159915208816528 \n",
      "\n",
      "Making predictions for chromosome: chr19\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th17/chr19/chr19.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 1.6013741493225098\n",
      "HiC Matrix has row sums of 1526.2347998480657, making doubly stochastic...\n",
      "HiC has 2234044 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 3.5399997234344482\n",
      "HiC added to predictions table. Elapsed time: 1.072981357574463\n",
      "HiC Complete\n",
      "Completed chromosome: chr19. Elapsed time: 7.235585689544678 \n",
      "\n",
      "Making predictions for chromosome: chr5\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th17/chr5/chr5.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 6.460506439208984\n",
      "HiC Matrix has row sums of 1455.796622666746, making doubly stochastic...\n",
      "HiC has 7067083 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 13.079856395721436\n",
      "HiC added to predictions table. Elapsed time: 2.872908353805542\n",
      "HiC Complete\n",
      "Completed chromosome: chr5. Elapsed time: 23.974833726882935 \n",
      "\n",
      "Making predictions for chromosome: chr22\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th17/chr22/chr22.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 0.8730366230010986\n",
      "HiC Matrix has row sums of 1417.4417063775095, making doubly stochastic...\n",
      "HiC has 1277651 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 2.0893394947052\n",
      "HiC added to predictions table. Elapsed time: 0.3574495315551758\n",
      "HiC Complete\n",
      "Completed chromosome: chr22. Elapsed time: 3.6595773696899414 \n",
      "\n",
      "Making predictions for chromosome: chr21\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th17/chr21/chr21.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 0.8842592239379883\n",
      "HiC Matrix has row sums of 1223.7193572134345, making doubly stochastic...\n",
      "HiC has 1313543 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 2.059053897857666\n",
      "HiC added to predictions table. Elapsed time: 0.3217923641204834\n",
      "HiC Complete\n",
      "Completed chromosome: chr21. Elapsed time: 3.5147948265075684 \n",
      "\n",
      "Making predictions for chromosome: chr7\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th17/chr7/chr7.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 5.276389122009277\n",
      "HiC Matrix has row sums of 1397.4282790987377, making doubly stochastic...\n",
      "HiC has 5791417 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 11.15759801864624\n",
      "HiC added to predictions table. Elapsed time: 2.0024564266204834\n",
      "HiC Complete\n",
      "Completed chromosome: chr7. Elapsed time: 19.787452459335327 \n",
      "\n",
      "Making predictions for chromosome: chr2\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th17/chr2/chr2.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 8.960262060165405\n",
      "HiC Matrix has row sums of 1536.974560245516, making doubly stochastic...\n",
      "HiC has 9317321 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 18.221892595291138\n",
      "HiC added to predictions table. Elapsed time: 3.8291003704071045\n",
      "HiC Complete\n",
      "Completed chromosome: chr2. Elapsed time: 33.20798301696777 \n",
      "\n",
      "Making predictions for chromosome: chr3\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th17/chr3/chr3.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 7.322037696838379\n",
      "HiC Matrix has row sums of 1518.217219048322, making doubly stochastic...\n",
      "HiC has 7955184 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 14.695894241333008\n",
      "HiC added to predictions table. Elapsed time: 3.2464966773986816\n",
      "HiC Complete\n",
      "Completed chromosome: chr3. Elapsed time: 27.103429079055786 \n",
      "\n",
      "Making predictions for chromosome: chr13\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th17/chr13/chr13.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 3.454299211502075\n",
      "HiC Matrix has row sums of 1371.9076569656688, making doubly stochastic...\n",
      "HiC has 3850169 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 6.852335453033447\n",
      "HiC added to predictions table. Elapsed time: 1.3212146759033203\n",
      "HiC Complete\n",
      "Completed chromosome: chr13. Elapsed time: 12.513129472732544 \n",
      "\n",
      "Making predictions for chromosome: chr1\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th17/chr1/chr1.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 7.8138086795806885\n",
      "HiC Matrix has row sums of 1549.9904993515233, making doubly stochastic...\n",
      "HiC has 8850041 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 16.91107439994812\n",
      "HiC added to predictions table. Elapsed time: 4.495526552200317\n",
      "HiC Complete\n",
      "Completed chromosome: chr1. Elapsed time: 32.233598947525024 \n",
      "\n",
      "Making predictions for chromosome: chr11\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th17/chr11/chr11.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 4.538697004318237\n",
      "HiC Matrix has row sums of 1470.015573589537, making doubly stochastic...\n",
      "HiC has 5110354 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 9.516477346420288\n",
      "HiC added to predictions table. Elapsed time: 2.1928622722625732\n",
      "HiC Complete\n",
      "Completed chromosome: chr11. Elapsed time: 18.012093544006348 \n",
      "\n",
      "Making predictions for chromosome: chr12\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th17/chr12/chr12.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 4.783036947250366\n",
      "HiC Matrix has row sums of 1502.404267592939, making doubly stochastic...\n",
      "HiC has 5306512 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 9.582131624221802\n",
      "HiC added to predictions table. Elapsed time: 2.1730899810791016\n",
      "HiC Complete\n",
      "Completed chromosome: chr12. Elapsed time: 18.06832528114319 \n",
      "\n",
      "Making predictions for chromosome: chrX\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th17/chrX/chrX.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 4.124722719192505\n",
      "HiC Matrix has row sums of 961.0704462520184, making doubly stochastic...\n",
      "HiC has 4539179 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 9.932192325592041\n",
      "HiC added to predictions table. Elapsed time: 1.5994281768798828\n",
      "HiC Complete\n",
      "Completed chromosome: chrX. Elapsed time: 16.682201147079468 \n",
      "\n",
      "Making predictions for chromosome: chr8\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th17/chr8/chr8.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 5.17205810546875\n",
      "HiC Matrix has row sums of 1467.1368826893463, making doubly stochastic...\n",
      "HiC has 5555717 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 10.603359460830688\n",
      "HiC added to predictions table. Elapsed time: 1.8901891708374023\n",
      "HiC Complete\n",
      "Completed chromosome: chr8. Elapsed time: 18.92940092086792 \n",
      "\n",
      "Making predictions for chromosome: chr18\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th17/chr18/chr18.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 2.443580389022827\n",
      "HiC Matrix has row sums of 1368.3268768812343, making doubly stochastic...\n",
      "HiC has 2903944 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 5.274777412414551\n",
      "HiC added to predictions table. Elapsed time: 0.7645535469055176\n",
      "HiC Complete\n",
      "Completed chromosome: chr18. Elapsed time: 9.101156949996948 \n",
      "\n",
      "Making predictions for chromosome: chr16\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th17/chr16/chr16.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 2.1839239597320557\n",
      "HiC Matrix has row sums of 1398.0825106876464, making doubly stochastic...\n",
      "HiC has 2821468 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 5.14653468132019\n",
      "HiC added to predictions table. Elapsed time: 0.859835147857666\n",
      "HiC Complete\n",
      "Completed chromosome: chr16. Elapsed time: 8.895794868469238 \n",
      "\n",
      "Making predictions for chromosome: chr4\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th17/chr4/chr4.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 7.19158148765564\n",
      "HiC Matrix has row sums of 1400.7624801900968, making doubly stochastic...\n",
      "HiC has 7248464 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 14.07811450958252\n",
      "HiC added to predictions table. Elapsed time: 2.9371042251586914\n",
      "HiC Complete\n",
      "Completed chromosome: chr4. Elapsed time: 25.865135192871094 \n",
      "\n",
      "Making predictions for chromosome: chr20\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th17/chr20/chr20.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 1.941455602645874\n",
      "HiC Matrix has row sums of 1457.633436399556, making doubly stochastic...\n",
      "HiC has 2456417 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 4.015393018722534\n",
      "HiC added to predictions table. Elapsed time: 0.7260286808013916\n",
      "HiC Complete\n",
      "Completed chromosome: chr20. Elapsed time: 7.220844984054565 \n",
      "\n",
      "Making predictions for chromosome: chr10\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th17/chr10/chr10.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 4.562747478485107\n",
      "HiC Matrix has row sums of 1473.9607341365308, making doubly stochastic...\n",
      "HiC has 5164413 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 9.478096961975098\n",
      "HiC added to predictions table. Elapsed time: 1.961728811264038\n",
      "HiC Complete\n",
      "Completed chromosome: chr10. Elapsed time: 17.343820810317993 \n",
      "\n",
      "Making predictions for chromosome: chr6\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th17/chr6/chr6.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 6.026068449020386\n",
      "HiC Matrix has row sums of 1509.6881820120548, making doubly stochastic...\n",
      "HiC has 6780726 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 12.499083280563354\n",
      "HiC added to predictions table. Elapsed time: 2.832386016845703\n",
      "HiC Complete\n",
      "Completed chromosome: chr6. Elapsed time: 22.995885133743286 \n",
      "\n",
      "Making predictions for chromosome: chr14\n",
      "Making putative predictions table...\n",
      "Using: /gpfs/fs1/data/reddylab/Revathy/dev/Maria/hic/Th17/chr14/chr14.KRobserved.gz\n",
      "Begin HiC\n",
      "Loading HiC\n",
      "hic.to.sparse: Elapsed time: 2.95397686958313\n",
      "HiC Matrix has row sums of 1464.4410384660755, making doubly stochastic...\n",
      "HiC has 3596196 rows after windowing between 0 and 5000000\n",
      "process.hic: Elapsed time: 6.226326942443848\n",
      "HiC added to predictions table. Elapsed time: 1.4025909900665283\n",
      "HiC Complete\n",
      "Completed chromosome: chr14. Elapsed time: 11.543150186538696 \n",
      "\n",
      "Writing output files...\n",
      "Done.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "source /data/reddylab/software/miniconda3/bin/activate jung_py3\n",
    "ABC=/gpfs/fs1/data/reddylab/Revathy/collabs/Maria/ABC/ABC-Enhancer-Gene-Prediction\n",
    "task=ABC_score\n",
    "mkdir -p /gpfs/fs1/data/reddylab/Revathy/collabs/Maria/ABC/${task}\n",
    "Enhancer_activity=/gpfs/fs1/data/reddylab/Revathy/collabs/Maria/ABC/enhancer_acitivity\n",
    "hic=/gpfs/fs1/data/reddylab/Revathy/collabs/Maria/hic\n",
    "\n",
    "cd ${Enhancer_activity}\n",
    "\n",
    "threshold=02\n",
    "LIST=sample_names.txt\n",
    "SAMPLES=$(cat /data/reddylab/Revathy/dev/Maria/ABC/$LIST|tr '\\n' ' ')\n",
    "\n",
    "for sample in $(echo Th01 Th17);\n",
    "do\n",
    "python ${ABC}/src/predict.py \\\n",
    "--enhancers ${Enhancer_activity}/${sample}_abc/EnhancerList.txt \\\n",
    "--genes ${Enhancer_activity}/${sample}_abc/GeneList.txt \\\n",
    "--HiCdir ${hic}/${sample} \\\n",
    "--hic_resolution 5000 \\\n",
    "--scale_hic_using_powerlaw \\\n",
    "--threshold .${threshold} \\\n",
    "--cellType ${sample} \\\n",
    "--outdir /gpfs/fs1/data/reddylab/Revathy/collabs/Maria/ABC/${task}/${sample}_hic.threshold0.${threshold} \\\n",
    "--make_all_putative \n",
    "\n",
    "done\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Create ABC enhancer beds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "module load bedtools2\n",
    "task=enhancers_by_ABC_bed\n",
    "mkdir -p /data/reddylab/Revathy/dev/Maria/ABC/${task}\n",
    "in_dir=/data/reddylab/Revathy/dev/Maria/ABC/ABC_score\n",
    "\n",
    "cd ${in_dir}\n",
    "\n",
    "for i in `ls -d *`;\n",
    "do \n",
    "cat ${in_dir}/${i}/EnhancerPredictions.txt | \\\n",
    "cut -f 1-3,5,7|grep -v end > \\\n",
    "/data/reddylab/Revathy/dev/Maria/ABC/${task}/${i}.bed\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "598211 /data/reddylab/Revathy/dev/Maria/ABC/ABC_score/Th01_hic.threshold0.008/EnhancerPredictions.txt\n",
      "589871 /data/reddylab/Revathy/dev/Maria/ABC/ABC_score/Th17_hic.threshold0.008/EnhancerPredictions.txt\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "wc -l /data/reddylab/Revathy/dev/Maria/ABC/ABC_score/Th01_hic.threshold0.008/EnhancerPredictions.txt\n",
    "wc -l /data/reddylab/Revathy/dev/Maria/ABC/ABC_score/Th17_hic.threshold0.008/EnhancerPredictions.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "195312 /data/reddylab/Revathy/dev/Maria/ABC/ABC_score/Th01_hic.threshold0.02/EnhancerPredictions.txt\n",
      "203522 /data/reddylab/Revathy/dev/Maria/ABC/ABC_score/Th17_hic.threshold0.02/EnhancerPredictions.txt\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "wc -l /data/reddylab/Revathy/dev/Maria/ABC/ABC_score/Th01_hic.threshold0.02/EnhancerPredictions.txt\n",
    "wc -l /data/reddylab/Revathy/dev/Maria/ABC/ABC_score/Th17_hic.threshold0.02/EnhancerPredictions.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
