{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: eu_register.py [-h] [-m DCC_MODE] [-d] [--no-aliases] -p PROFILE_ID -i\n",
      "                      INFILE [--patch] [-w]\n",
      "\n",
      "Given a tab-delimited or JSON input file containing one or more records belonging to one of the profiles\n",
      "listed on the ENCODE Portal (such as https://www.encodeproject.org/profiles/biosample.json),\n",
      "either POSTS or PATCHES the records. The default is to POST each record; to PATCH instead, see\n",
      "the ``--patch`` option.\n",
      "\n",
      "When POSTING file records, the md5sum of each file will be calculated for you if you haven't\n",
      "already provided the `md5sum` property. Then, after the POST operation completes, the actual file\n",
      "will be uploaded to AWS S3. In order for this to work, you must set the `submitted_file_name`\n",
      "property to the full, local path to your file to upload. Alternatively, you can set\n",
      "`submitted_file_name` to and existing S3 object, i.e. s3://mybucket/reads.fastq.\n",
      "\n",
      "Note that there is a special 'trick' defined in the ``encode_utils.connection.Connection()``\n",
      "class that can be taken advantage of to simplify submission under certain profiles.\n",
      "It concerns the `attachment` property in any profile that employs it, such as the `document`\n",
      "profile.  The trick works as follows: instead of constructing the `attachment` propery object\n",
      "value as defined in the schema, simply use a single-key object of the following format::\n",
      "\n",
      "  {\"path\": \"/path/to/myfile\"}\n",
      "\n",
      "and the `attachment` object will be constructed for you.\n",
      "\n",
      "|\n",
      "\n",
      "optional arguments:\n",
      "  -h, --help            show this help message and exit\n",
      "  -m DCC_MODE, --dcc-mode DCC_MODE\n",
      "                        \n",
      "                            The ENCODE Portal site ('prod' or 'dev', or an explicit host name, i.e. 'demo.encodedcc.org') to connect to.\n",
      "  -d, --dry-run         \n",
      "                            Set this option to enable the dry-run feature, such that no modifications are performed on the\n",
      "                            ENCODE Portal.  This is useful if you'd like to inspect the logs or ensure the validity of\n",
      "                            your input file.\n",
      "  --no-aliases          \n",
      "                            Setting this option is NOT advised. Set this option for doing a POST when your input file \n",
      "                            doesn't contain an 'aliases' column, even though this property is supported in the corresponding\n",
      "                            ENCODE profile.\n",
      "                            When POSTING a record to a profile that includes the 'aliases' property, this package requires\n",
      "                            the 'aliases' property be used for traceability purposes and because without this property, \n",
      "                            it'll be very easy to create duplicate objects on the Portal.  For example, you can easily \n",
      "                            create the same biosample as many times as you want on the Portal when not providing an alias.\n",
      "  -p PROFILE_ID, --profile_id PROFILE_ID\n",
      "                        \n",
      "                            The ID of the profile to submit to, i.e. use 'genetic_modification' for\n",
      "                            https://www.encodeproject.org/profiles/genetic_modification.json. The profile will be pulled down for\n",
      "                            type-checking in order to type-cast any values in the input file to the proper type (i.e. some\n",
      "                            values need to be submitted as integers, not strings).\n",
      "  -i INFILE, --infile INFILE\n",
      "                        \n",
      "                            The JSON input file or tab-delimited input file. \n",
      "                        \n",
      "                            **The tab-delimited file format:**\n",
      "                            Must have a field-header line as the first line.\n",
      "                            Any lines after the header line that start with a '#' will be skipped, as well as any empty lines.\n",
      "                            The field names must be exactly equal to the corresponding property names in the corresponding \n",
      "                            profile. Non-scematic fields are allowed as long as they begin with a '#'; they will be \n",
      "                            skipped. If a property has an array data type (as indicated in the profile's documentation \n",
      "                            on the Portal), the array literals '[' and ']' are optional. Values within the array must \n",
      "                            be comma-delimited. For example, if a property takes an array of strings, then you can use \n",
      "                            either of these as the value:\n",
      "                        \n",
      "                            1) str1,str2,str3\n",
      "                            2) [str1,str2,str3]\n",
      "                        \n",
      "                            On the other hand, if a property takes a JSON object as a value, then the value you enter must be\n",
      "                            valid JSON. This is true anytime you have to specify a JSON object.  Thus, if you are submitting a\n",
      "                            genetic_modification and you have two 'introduced_tags' to provide, you can supply them in either\n",
      "                            of the following two ways:\n",
      "                        \n",
      "                            1) {\"name\": \"eGFP\", \"location\": \"C-terminal\"},{\"name\": \"FLAG\",\"C-terminal\"}\n",
      "                            2) [{\"name\": \"eGFP\", \"location\": \"C-terminal\"},{\"name\": \"FLAG\",\"C-terminal\"}]\n",
      "                        \n",
      "                            **The JSON input file**\n",
      "                            Can be a single JSON object, or an array of JSON objects. Key names must match property names of\n",
      "                            an ENCODE record type (profile).\n",
      "                        \n",
      "                            **The following applies to either input file formats**\n",
      "                            When patching objects, you must specify the 'record_id' field to indicate the identifier of the record.\n",
      "                            Note that this a special field that is not present in the ENCODE schema, and doesn't use the '#'\n",
      "                            prefix to mark it as non-schematic. Here you can specify any valid record identifier\n",
      "                            (i.e. UUID, accession, alias).\n",
      "                        \n",
      "                            Some profiles (most) require specification of the 'award' and 'lab' attributes. These may be set\n",
      "                            as fields in the input file, or can be left out, in which case the default values for these\n",
      "                            attributes will be pulled from the environment variables DCC_AWARD and DCC_LAB, respectively.\n",
      "                            \n",
      "  --patch               \n",
      "                            Presence of this option indicates to PATCH an existing DCC record rather than register a new one.\n",
      "  -w, --overwrite-array-values\n",
      "                        \n",
      "                            Only has meaning in combination with the --patch option. When this is specified, it means that\n",
      "                            any keys with array values will be overwritten on the ENCODE Portal with the corresponding value\n",
      "                            to patch. The default action is to extend the array value with the patch value and then to remove\n",
      "                            any duplicates.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "eu_register.py --help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "mkdir -p /data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/metadata\n",
    "source /data/reddylab/software/miniconda3/bin/activate revathy_py3\n",
    "python /data/reddylab/Revathy/scripts/excel_to_text_for_ENCODE.py \\\n",
    "-i /data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/2023_Tcell_CRISPRi_Keith_encode_metadata.xlsx \\\n",
    "-o /data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/metadata \\\n",
    "--sheet-names functional_characterization_ser"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2023-09-25 17:42:21,566:eu_debug:\tConnecting to www.encodeproject.org\n",
      "2023-09-25 17:42:21,592:eu_debug:\tsubmission=False: In non-submission mode.\n",
      "2023-09-25 17:42:22,288:eu_debug:\tsubmission=True: In submission mode.\n",
      "2023-09-25 17:42:22,292:eu_debug:\t\n",
      "IN post().\n",
      "2023-09-25 17:42:22,292:eu_debug:\t<<<<<< POST functional_characterization_series record tim-reddy:mm10-th17-crispri-GITR-series To DCC with URL https://www.encodeproject.org/functional_characterization_series and this payload:\n",
      "\n",
      "{\n",
      "  \"aliases\": [\n",
      "    \"tim-reddy:mm10-th17-crispri-GITR-series\"\n",
      "  ],\n",
      "  \"award\": \"/awards/UM1HG009428/\",\n",
      "  \"lab\": \"/labs/tim-reddy/\",\n",
      "  \"related_datasets\": [\n",
      "    \"tim-reddy:mm10-th17-crispri-GITR-low-bin\",\n",
      "    \"tim-reddy:mm10-th17-crispri-GITR-high-bin\"\n",
      "  ]\n",
      "}\n",
      "\n",
      "\n",
      "2023-09-25 17:42:22,834:eu_debug:\tSuccess.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "source /data/reddylab/software/miniconda3/bin/activate revathy_py3\n",
    "cd /data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/metadata\n",
    "eu_register.py -m prod -p functional_characterization_series -i functional_characterization_ser.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Error [file=/data/reddylab/Keith/collab/220131_TaniaRotation/230921_GITR_Submission/TG2-GITR-Hi-rep1.trim20.alignment.counts.txt, line=1]: found 2 columns, expected 25 [chr4_156026280_156026302_+_peak_10__66_GGCCTGAAGCCCAGTCTGAG\t71]\n",
      "Aborting ... found error.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/data/common/shared_conda_envs/ucsc/bin/validateFiles \\\n",
    "-type=bed3+22 \\\n",
    "-chromInfo=/data/reddylab/software/encValData/hg38/chrom.sizes \\\n",
    "-as=/data/reddylab/Alex/software/encValData/as/element_quant_format.as \\\n",
    "/data/reddylab/Keith/collab/220131_TaniaRotation/230921_GITR_Submission/TG2-GITR-Hi-rep1.trim20.alignment.counts.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Convert .counts.txt files to element_quant_format.as format"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir -p /data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "import warnings\n",
    "import glob\n",
    "import re\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from pandas.core.common import SettingWithCopyWarning\n",
    "warnings.simplefilter(action=\"ignore\", category=SettingWithCopyWarning)\n",
    "\n",
    "files = [f for f in glob.glob('/data/reddylab/Keith/collab/220131_TaniaRotation/230921_GITR_Submission/' + '*.counts.txt')]\n",
    "\n",
    "for file in files:\n",
    "    file_name='.'.join(file.split('/')[-1].split('.')[:3])\n",
    "    \n",
    "    tst = pd.read_csv(file, sep='\\t', names=['name','count'])\n",
    "    tst.tail(40)['name'] = 'NA_NA_NA_NA_' + tst.tail(40)['name']\n",
    "    tst['seq'] = tst['name'].str.split('_').str[-1]\n",
    "    tst['name'] = tst['name'].str.split('_').str[:-1].str.join('_')\n",
    "    tst['chr_start_end_strand'] = tst.name.str.split('_').str[:4].str.join('_')\n",
    "    tst['id'] = tst['name'].str.split('_').str[4:].str.join('_')\n",
    "    tst[['chr','start','end','strand']] = tst.chr_start_end_strand.str.split('_', expand=True)\n",
    "    tst_nt = tst[tst['id'].str.contains('nontargeting')]\n",
    "    tst_t = tst[~tst['id'].str.contains('nontargeting')]\n",
    "    \n",
    "    tst_nt['guidetype'] = 'Non-targeting'\n",
    "    tst_nt['col4'] = 'NA|' + tst['id']\n",
    "    tst_nt['col7'] = tst['id']\n",
    "    \n",
    "    tst_t['guidetype'] = 'targeting'\n",
    "    tst_t['col4'] = 'NA|' + tst_t['chr']+':'+tst_t['start']+'-'+tst_t['end']+':'+tst_t['strand']\n",
    "    tst_t['col7'] = tst_t['chr']+':'+tst_t['start']+'-'+tst_t['end']+':'+tst_t['strand']\n",
    "    \n",
    "    tst_v1 = pd.concat([tst_t, tst_nt], axis=0)\n",
    "\n",
    "    new_col=['col8','col9','col10','col11','col12','col13','col14','col17']\n",
    "    \n",
    "    for i in new_col:\n",
    "        tst_v1[i]='NA'\n",
    "    tst_v2 = tst_v1[['chr','start','end','col4','count','strand','col7','col8','col9','col10','col11','col12','col13','col14','seq','guidetype','col17']]\n",
    "    tst_v2.to_csv('/data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/data/%s.guide_quantifications.tsv'%(file_name), sep='\\t', header=None, index=False)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2023-10-05 11:20:34,712:eu_debug:\tConnecting to www.encodeproject.org\n",
      "2023-10-05 11:20:34,743:eu_debug:\tsubmission=False: In non-submission mode.\n",
      "2023-10-05 11:20:35,427:eu_debug:\tsubmission=True: In submission mode.\n",
      "2023-10-05 11:20:35,435:eu_debug:\t\n",
      "IN post().\n",
      "2023-10-05 11:20:35,463:eu_debug:\t<<<<<< POST file record tim-reddy:mm10-th17-crispri-GITR-low-bin-rep1-guide-quantifications To DCC with URL https://www.encodeproject.org/file and this payload:\n",
      "\n",
      "{\n",
      "  \"aliases\": [\n",
      "    \"tim-reddy:mm10-th17-crispri-GITR-low-bin-rep1-guide-quantifications\"\n",
      "  ],\n",
      "  \"assembly\": \"mm10\",\n",
      "  \"award\": \"/awards/UM1HG009428/\",\n",
      "  \"dataset\": \"tim-reddy:mm10-th17-crispri-GITR-low-bin\",\n",
      "  \"derived_from\": [\n",
      "    \"tim-reddy:mm10-th17-crispri-GITR-low-bin-rep1-fastq\"\n",
      "  ],\n",
      "  \"file_format\": \"tsv\",\n",
      "  \"file_size\": 62161,\n",
      "  \"lab\": \"/labs/tim-reddy/\",\n",
      "  \"md5sum\": \"b174f0e300fdf2d27299e1e9a4c2e540\",\n",
      "  \"output_type\": \"guide quantifications\",\n",
      "  \"replicate\": \"tim-reddy:mm10-th17-crispri-GITR-low-bin-rep1\",\n",
      "  \"submitted_file_name\": \"/data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/data/TG2-GITR-Lo-rep1.trim20.alignment.guide_quantifications.tsv\"\n",
      "}\n",
      "\n",
      "\n",
      "2023-10-05 11:20:36,282:eu_debug:\tSuccess.\n",
      "2023-10-05 11:20:36,290:eu_debug:\t\n",
      "IN upload_file()\n",
      "\n",
      "2023-10-05 11:20:36,290:eu_debug:\tAttempting to generate new file upload credentials\n",
      "2023-10-05 11:20:37,024:eu_debug:\tSuccess: upload credentials for 'ENCFF698PZT' regenerated.\n",
      "2023-10-05 11:20:37,025:eu_debug:\t>>>>>>GET ENCFF698PZT From DCC with URL https://www.encodeproject.org/ENCFF698PZT/?format=json&datastore=database\n",
      "2023-10-05 11:20:37,738:eu_debug:\tRunning command 'aws s3 cp /data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/data/TG2-GITR-Lo-rep1.trim20.alignment.guide_quantifications.tsv s3://encode-files/2023/10/05/1b4b950f-cbbf-4901-ac8d-3ac281212b42/ENCFF698PZT.tsv'.\n",
      "2023-10-05 11:20:44,826:eu_debug:\tAWS upload successful.\n",
      "2023-10-05 11:20:44,827:eu_debug:\t\n",
      "IN post().\n",
      "2023-10-05 11:20:44,857:eu_debug:\t<<<<<< POST file record tim-reddy:mm10-th17-crispri-GITR-low-bin-rep2-guide-quantifications To DCC with URL https://www.encodeproject.org/file and this payload:\n",
      "\n",
      "{\n",
      "  \"aliases\": [\n",
      "    \"tim-reddy:mm10-th17-crispri-GITR-low-bin-rep2-guide-quantifications\"\n",
      "  ],\n",
      "  \"assembly\": \"mm10\",\n",
      "  \"award\": \"/awards/UM1HG009428/\",\n",
      "  \"dataset\": \"tim-reddy:mm10-th17-crispri-GITR-low-bin\",\n",
      "  \"derived_from\": [\n",
      "    \"tim-reddy:mm10-th17-crispri-GITR-low-bin-rep2-fastq\"\n",
      "  ],\n",
      "  \"file_format\": \"tsv\",\n",
      "  \"file_size\": 62161,\n",
      "  \"lab\": \"/labs/tim-reddy/\",\n",
      "  \"md5sum\": \"91e20da1842146756cf3756359942161\",\n",
      "  \"output_type\": \"guide quantifications\",\n",
      "  \"replicate\": \"tim-reddy:mm10-th17-crispri-GITR-low-bin-rep2\",\n",
      "  \"submitted_file_name\": \"/data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/data/TG2-GITR-Lo-rep2.trim20.alignment.guide_quantifications.tsv\"\n",
      "}\n",
      "\n",
      "\n",
      "2023-10-05 11:20:45,623:eu_debug:\tSuccess.\n",
      "2023-10-05 11:20:45,635:eu_debug:\t\n",
      "IN upload_file()\n",
      "\n",
      "2023-10-05 11:20:45,635:eu_debug:\tAttempting to generate new file upload credentials\n",
      "2023-10-05 11:20:46,340:eu_debug:\tSuccess: upload credentials for 'ENCFF478SBX' regenerated.\n",
      "2023-10-05 11:20:46,341:eu_debug:\t>>>>>>GET ENCFF478SBX From DCC with URL https://www.encodeproject.org/ENCFF478SBX/?format=json&datastore=database\n",
      "2023-10-05 11:20:47,055:eu_debug:\tRunning command 'aws s3 cp /data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/data/TG2-GITR-Lo-rep2.trim20.alignment.guide_quantifications.tsv s3://encode-files/2023/10/05/2905c122-148e-4886-9d32-bfefb8c01d94/ENCFF478SBX.tsv'.\n",
      "2023-10-05 11:20:48,866:eu_debug:\tAWS upload successful.\n",
      "2023-10-05 11:20:48,866:eu_debug:\t\n",
      "IN post().\n",
      "2023-10-05 11:20:48,900:eu_debug:\t<<<<<< POST file record tim-reddy:mm10-th17-crispri-GITR-low-bin-rep3-guide-quantifications To DCC with URL https://www.encodeproject.org/file and this payload:\n",
      "\n",
      "{\n",
      "  \"aliases\": [\n",
      "    \"tim-reddy:mm10-th17-crispri-GITR-low-bin-rep3-guide-quantifications\"\n",
      "  ],\n",
      "  \"assembly\": \"mm10\",\n",
      "  \"award\": \"/awards/UM1HG009428/\",\n",
      "  \"dataset\": \"tim-reddy:mm10-th17-crispri-GITR-low-bin\",\n",
      "  \"derived_from\": [\n",
      "    \"tim-reddy:mm10-th17-crispri-GITR-low-bin-rep3-fastq\"\n",
      "  ],\n",
      "  \"file_format\": \"tsv\",\n",
      "  \"file_size\": 62167,\n",
      "  \"lab\": \"/labs/tim-reddy/\",\n",
      "  \"md5sum\": \"26cc39a2f7ed1f1be42760f233bb2767\",\n",
      "  \"output_type\": \"guide quantifications\",\n",
      "  \"replicate\": \"tim-reddy:mm10-th17-crispri-GITR-low-bin-rep3\",\n",
      "  \"submitted_file_name\": \"/data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/data/TG2-GITR-Lo-rep3.trim20.alignment.guide_quantifications.tsv\"\n",
      "}\n",
      "\n",
      "\n",
      "2023-10-05 11:20:49,675:eu_debug:\tSuccess.\n",
      "2023-10-05 11:20:49,675:eu_debug:\t\n",
      "IN upload_file()\n",
      "\n",
      "2023-10-05 11:20:49,675:eu_debug:\tAttempting to generate new file upload credentials\n",
      "2023-10-05 11:20:50,377:eu_debug:\tSuccess: upload credentials for 'ENCFF711ZQR' regenerated.\n",
      "2023-10-05 11:20:50,378:eu_debug:\t>>>>>>GET ENCFF711ZQR From DCC with URL https://www.encodeproject.org/ENCFF711ZQR/?format=json&datastore=database\n",
      "2023-10-05 11:20:51,036:eu_debug:\tRunning command 'aws s3 cp /data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/data/TG2-GITR-Lo-rep3.trim20.alignment.guide_quantifications.tsv s3://encode-files/2023/10/05/b77694b1-41c6-4446-871a-d4ff66fc27b3/ENCFF711ZQR.tsv'.\n",
      "2023-10-05 11:20:52,876:eu_debug:\tAWS upload successful.\n",
      "2023-10-05 11:20:52,877:eu_debug:\t\n",
      "IN post().\n",
      "2023-10-05 11:20:52,900:eu_debug:\t<<<<<< POST file record tim-reddy:mm10-th17-crispri-GITR-low-bin-rep4-guide-quantifications To DCC with URL https://www.encodeproject.org/file and this payload:\n",
      "\n",
      "{\n",
      "  \"aliases\": [\n",
      "    \"tim-reddy:mm10-th17-crispri-GITR-low-bin-rep4-guide-quantifications\"\n",
      "  ],\n",
      "  \"assembly\": \"mm10\",\n",
      "  \"award\": \"/awards/UM1HG009428/\",\n",
      "  \"dataset\": \"tim-reddy:mm10-th17-crispri-GITR-low-bin\",\n",
      "  \"derived_from\": [\n",
      "    \"tim-reddy:mm10-th17-crispri-GITR-low-bin-rep4-fastq\"\n",
      "  ],\n",
      "  \"file_format\": \"tsv\",\n",
      "  \"file_size\": 62156,\n",
      "  \"lab\": \"/labs/tim-reddy/\",\n",
      "  \"md5sum\": \"7e55a8ee992c5911993e557a8e69893d\",\n",
      "  \"output_type\": \"guide quantifications\",\n",
      "  \"replicate\": \"tim-reddy:mm10-th17-crispri-GITR-low-bin-rep4\",\n",
      "  \"submitted_file_name\": \"/data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/data/TG2-GITR-Lo-rep4.trim20.alignment.guide_quantifications.tsv\"\n",
      "}\n",
      "\n",
      "\n",
      "2023-10-05 11:20:53,644:eu_debug:\tSuccess.\n",
      "2023-10-05 11:20:53,645:eu_debug:\t\n",
      "IN upload_file()\n",
      "\n",
      "2023-10-05 11:20:53,646:eu_debug:\tAttempting to generate new file upload credentials\n",
      "2023-10-05 11:20:54,370:eu_debug:\tSuccess: upload credentials for 'ENCFF222YYO' regenerated.\n",
      "2023-10-05 11:20:54,371:eu_debug:\t>>>>>>GET ENCFF222YYO From DCC with URL https://www.encodeproject.org/ENCFF222YYO/?format=json&datastore=database\n",
      "2023-10-05 11:20:55,057:eu_debug:\tRunning command 'aws s3 cp /data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/data/TG2-GITR-Lo-rep4.trim20.alignment.guide_quantifications.tsv s3://encode-files/2023/10/05/180c5043-70a3-48ad-8424-f30a4f634c23/ENCFF222YYO.tsv'.\n",
      "2023-10-05 11:20:56,878:eu_debug:\tAWS upload successful.\n",
      "2023-10-05 11:20:56,879:eu_debug:\t\n",
      "IN post().\n",
      "2023-10-05 11:20:56,901:eu_debug:\t<<<<<< POST file record tim-reddy:mm10-th17-crispri-GITR-high-bin-rep1-guide-quantifications To DCC with URL https://www.encodeproject.org/file and this payload:\n",
      "\n",
      "{\n",
      "  \"aliases\": [\n",
      "    \"tim-reddy:mm10-th17-crispri-GITR-high-bin-rep1-guide-quantifications\"\n",
      "  ],\n",
      "  \"assembly\": \"mm10\",\n",
      "  \"award\": \"/awards/UM1HG009428/\",\n",
      "  \"dataset\": \"tim-reddy:mm10-th17-crispri-GITR-high-bin\",\n",
      "  \"derived_from\": [\n",
      "    \"tim-reddy:mm10-th17-crispri-GITR-high-bin-rep1-fastq\"\n",
      "  ],\n",
      "  \"file_format\": \"tsv\",\n",
      "  \"file_size\": 62148,\n",
      "  \"lab\": \"/labs/tim-reddy/\",\n",
      "  \"md5sum\": \"72ad169774adafd22f90287438345bde\",\n",
      "  \"output_type\": \"guide quantifications\",\n",
      "  \"replicate\": \"tim-reddy:mm10-th17-crispri-GITR-high-bin-rep1\",\n",
      "  \"submitted_file_name\": \"/data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/data/TG2-GITR-Hi-rep1.trim20.alignment.guide_quantifications.tsv\"\n",
      "}\n",
      "\n",
      "\n",
      "2023-10-05 11:20:57,675:eu_debug:\tSuccess.\n",
      "2023-10-05 11:20:57,676:eu_debug:\t\n",
      "IN upload_file()\n",
      "\n",
      "2023-10-05 11:20:57,676:eu_debug:\tAttempting to generate new file upload credentials\n",
      "2023-10-05 11:20:58,410:eu_debug:\tSuccess: upload credentials for 'ENCFF725JNP' regenerated.\n",
      "2023-10-05 11:20:58,410:eu_debug:\t>>>>>>GET ENCFF725JNP From DCC with URL https://www.encodeproject.org/ENCFF725JNP/?format=json&datastore=database\n",
      "2023-10-05 11:20:59,197:eu_debug:\tRunning command 'aws s3 cp /data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/data/TG2-GITR-Hi-rep1.trim20.alignment.guide_quantifications.tsv s3://encode-files/2023/10/05/bff97036-e31f-4285-b101-8a08a5b62c62/ENCFF725JNP.tsv'.\n",
      "2023-10-05 11:21:01,059:eu_debug:\tAWS upload successful.\n",
      "2023-10-05 11:21:01,060:eu_debug:\t\n",
      "IN post().\n",
      "2023-10-05 11:21:01,088:eu_debug:\t<<<<<< POST file record tim-reddy:mm10-th17-crispri-GITR-high-bin-rep2-guide-quantifications To DCC with URL https://www.encodeproject.org/file and this payload:\n",
      "\n",
      "{\n",
      "  \"aliases\": [\n",
      "    \"tim-reddy:mm10-th17-crispri-GITR-high-bin-rep2-guide-quantifications\"\n",
      "  ],\n",
      "  \"assembly\": \"mm10\",\n",
      "  \"award\": \"/awards/UM1HG009428/\",\n",
      "  \"dataset\": \"tim-reddy:mm10-th17-crispri-GITR-high-bin\",\n",
      "  \"derived_from\": [\n",
      "    \"tim-reddy:mm10-th17-crispri-GITR-high-bin-rep2-fastq\"\n",
      "  ],\n",
      "  \"file_format\": \"tsv\",\n",
      "  \"file_size\": 62154,\n",
      "  \"lab\": \"/labs/tim-reddy/\",\n",
      "  \"md5sum\": \"a9738eb0436d69671ad92957b31ff8ab\",\n",
      "  \"output_type\": \"guide quantifications\",\n",
      "  \"replicate\": \"tim-reddy:mm10-th17-crispri-GITR-high-bin-rep2\",\n",
      "  \"submitted_file_name\": \"/data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/data/TG2-GITR-Hi-rep2.trim20.alignment.guide_quantifications.tsv\"\n",
      "}\n",
      "\n",
      "\n",
      "2023-10-05 11:21:01,810:eu_debug:\tSuccess.\n",
      "2023-10-05 11:21:01,811:eu_debug:\t\n",
      "IN upload_file()\n",
      "\n",
      "2023-10-05 11:21:01,811:eu_debug:\tAttempting to generate new file upload credentials\n",
      "2023-10-05 11:21:02,529:eu_debug:\tSuccess: upload credentials for 'ENCFF769COW' regenerated.\n",
      "2023-10-05 11:21:02,530:eu_debug:\t>>>>>>GET ENCFF769COW From DCC with URL https://www.encodeproject.org/ENCFF769COW/?format=json&datastore=database\n",
      "2023-10-05 11:21:03,250:eu_debug:\tRunning command 'aws s3 cp /data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/data/TG2-GITR-Hi-rep2.trim20.alignment.guide_quantifications.tsv s3://encode-files/2023/10/05/87040fc6-af64-4bce-8f81-225aa5574560/ENCFF769COW.tsv'.\n",
      "2023-10-05 11:21:05,046:eu_debug:\tAWS upload successful.\n",
      "2023-10-05 11:21:05,047:eu_debug:\t\n",
      "IN post().\n",
      "2023-10-05 11:21:05,066:eu_debug:\t<<<<<< POST file record tim-reddy:mm10-th17-crispri-GITR-high-bin-rep3-guide-quantifications To DCC with URL https://www.encodeproject.org/file and this payload:\n",
      "\n",
      "{\n",
      "  \"aliases\": [\n",
      "    \"tim-reddy:mm10-th17-crispri-GITR-high-bin-rep3-guide-quantifications\"\n",
      "  ],\n",
      "  \"assembly\": \"mm10\",\n",
      "  \"award\": \"/awards/UM1HG009428/\",\n",
      "  \"dataset\": \"tim-reddy:mm10-th17-crispri-GITR-high-bin\",\n",
      "  \"derived_from\": [\n",
      "    \"tim-reddy:mm10-th17-crispri-GITR-high-bin-rep3-fastq\"\n",
      "  ],\n",
      "  \"file_format\": \"tsv\",\n",
      "  \"file_size\": 62154,\n",
      "  \"lab\": \"/labs/tim-reddy/\",\n",
      "  \"md5sum\": \"55e2be9af1c324e5dd7df8c66427b8d4\",\n",
      "  \"output_type\": \"guide quantifications\",\n",
      "  \"replicate\": \"tim-reddy:mm10-th17-crispri-GITR-high-bin-rep3\",\n",
      "  \"submitted_file_name\": \"/data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/data/TG2-GITR-Hi-rep3.trim20.alignment.guide_quantifications.tsv\"\n",
      "}\n",
      "\n",
      "\n",
      "2023-10-05 11:21:05,813:eu_debug:\tSuccess.\n",
      "2023-10-05 11:21:05,813:eu_debug:\t\n",
      "IN upload_file()\n",
      "\n",
      "2023-10-05 11:21:05,813:eu_debug:\tAttempting to generate new file upload credentials\n",
      "2023-10-05 11:21:06,533:eu_debug:\tSuccess: upload credentials for 'ENCFF858ZEQ' regenerated.\n",
      "2023-10-05 11:21:06,534:eu_debug:\t>>>>>>GET ENCFF858ZEQ From DCC with URL https://www.encodeproject.org/ENCFF858ZEQ/?format=json&datastore=database\n",
      "2023-10-05 11:21:07,289:eu_debug:\tRunning command 'aws s3 cp /data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/data/TG2-GITR-Hi-rep3.trim20.alignment.guide_quantifications.tsv s3://encode-files/2023/10/05/f700452f-7d27-4ab8-b7f8-73cecd9f4c52/ENCFF858ZEQ.tsv'.\n",
      "2023-10-05 11:21:09,217:eu_debug:\tAWS upload successful.\n",
      "2023-10-05 11:21:09,218:eu_debug:\t\n",
      "IN post().\n",
      "2023-10-05 11:21:09,234:eu_debug:\t<<<<<< POST file record tim-reddy:mm10-th17-crispri-GITR-high-bin-rep4-guide-quantifications To DCC with URL https://www.encodeproject.org/file and this payload:\n",
      "\n",
      "{\n",
      "  \"aliases\": [\n",
      "    \"tim-reddy:mm10-th17-crispri-GITR-high-bin-rep4-guide-quantifications\"\n",
      "  ],\n",
      "  \"assembly\": \"mm10\",\n",
      "  \"award\": \"/awards/UM1HG009428/\",\n",
      "  \"dataset\": \"tim-reddy:mm10-th17-crispri-GITR-high-bin\",\n",
      "  \"derived_from\": [\n",
      "    \"tim-reddy:mm10-th17-crispri-GITR-high-bin-rep4-fastq\"\n",
      "  ],\n",
      "  \"file_format\": \"tsv\",\n",
      "  \"file_size\": 62153,\n",
      "  \"lab\": \"/labs/tim-reddy/\",\n",
      "  \"md5sum\": \"ccce7ca77af9cd2ac39cf4a170b8c24a\",\n",
      "  \"output_type\": \"guide quantifications\",\n",
      "  \"replicate\": \"tim-reddy:mm10-th17-crispri-GITR-high-bin-rep4\",\n",
      "  \"submitted_file_name\": \"/data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/data/TG2-GITR-Hi-rep4.trim20.alignment.guide_quantifications.tsv\"\n",
      "}\n",
      "\n",
      "\n",
      "2023-10-05 11:21:09,953:eu_debug:\tSuccess.\n",
      "2023-10-05 11:21:09,953:eu_debug:\t\n",
      "IN upload_file()\n",
      "\n",
      "2023-10-05 11:21:09,953:eu_debug:\tAttempting to generate new file upload credentials\n",
      "2023-10-05 11:21:10,674:eu_debug:\tSuccess: upload credentials for 'ENCFF320TKB' regenerated.\n",
      "2023-10-05 11:21:10,675:eu_debug:\t>>>>>>GET ENCFF320TKB From DCC with URL https://www.encodeproject.org/ENCFF320TKB/?format=json&datastore=database\n",
      "2023-10-05 11:21:11,389:eu_debug:\tRunning command 'aws s3 cp /data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/data/TG2-GITR-Hi-rep4.trim20.alignment.guide_quantifications.tsv s3://encode-files/2023/10/05/536ae662-8c95-4fd4-a219-9cd426e562ee/ENCFF320TKB.tsv'.\n",
      "2023-10-05 11:21:13,246:eu_debug:\tAWS upload successful.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "source /data/reddylab/software/miniconda3/bin/activate revathy_py3\n",
    "cd /data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/metadata\n",
    "eu_register.py -m prod -p file -i crispr_files_guide-quantifications.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TG2-GITR-Hi-rep1.trim20.alignment.guide_quantifications.tsv\r\n",
      "TG2-GITR-Hi-rep2.trim20.alignment.guide_quantifications.tsv\r\n",
      "TG2-GITR-Hi-rep3.trim20.alignment.guide_quantifications.tsv\r\n",
      "TG2-GITR-Hi-rep4.trim20.alignment.guide_quantifications.tsv\r\n",
      "TG2-GITR-Lo-rep1.trim20.alignment.guide_quantifications.tsv\r\n",
      "TG2-GITR-Lo-rep2.trim20.alignment.guide_quantifications.tsv\r\n",
      "TG2-GITR-Lo-rep3.trim20.alignment.guide_quantifications.tsv\r\n",
      "TG2-GITR-Lo-rep4.trim20.alignment.guide_quantifications.tsv\r\n"
     ]
    }
   ],
   "source": [
    "!ls /data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/data/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NA\tNA\tNA\tNA|nontargeting_31\t474\tNA\tnontargeting_31\tNA\tNA\tNA\tNA\tNA\tNA\tNA\tGGGTAGACAGTCTCGGAGAG\tNon-targeting\tNA\r\n",
      "NA\tNA\tNA\tNA|nontargeting_32\t440\tNA\tnontargeting_32\tNA\tNA\tNA\tNA\tNA\tNA\tNA\tAGTGGTGTCGCTCACCTTCG\tNon-targeting\tNA\r\n",
      "NA\tNA\tNA\tNA|nontargeting_33\t551\tNA\tnontargeting_33\tNA\tNA\tNA\tNA\tNA\tNA\tNA\tAGAGTGGCGTCAGGCTGGCG\tNon-targeting\tNA\r\n",
      "NA\tNA\tNA\tNA|nontargeting_34\t198\tNA\tnontargeting_34\tNA\tNA\tNA\tNA\tNA\tNA\tNA\tAAGCGCGGAATTACCCGTCA\tNon-targeting\tNA\r\n",
      "NA\tNA\tNA\tNA|nontargeting_35\t296\tNA\tnontargeting_35\tNA\tNA\tNA\tNA\tNA\tNA\tNA\tTTGTGCCTATAGGTAAAATC\tNon-targeting\tNA\r\n",
      "NA\tNA\tNA\tNA|nontargeting_36\t497\tNA\tnontargeting_36\tNA\tNA\tNA\tNA\tNA\tNA\tNA\tTCGCCCCCCACTACCAAGAA\tNon-targeting\tNA\r\n",
      "NA\tNA\tNA\tNA|nontargeting_37\t375\tNA\tnontargeting_37\tNA\tNA\tNA\tNA\tNA\tNA\tNA\tTCCTGACCTATCCGAAAAAA\tNon-targeting\tNA\r\n",
      "NA\tNA\tNA\tNA|nontargeting_38\t452\tNA\tnontargeting_38\tNA\tNA\tNA\tNA\tNA\tNA\tNA\tCCCACCCCGCTGTTTGCACG\tNon-targeting\tNA\r\n",
      "NA\tNA\tNA\tNA|nontargeting_39\t393\tNA\tnontargeting_39\tNA\tNA\tNA\tNA\tNA\tNA\tNA\tGGAACCTCGCTAGTACCATT\tNon-targeting\tNA\r\n",
      "NA\tNA\tNA\tNA|nontargeting_40\t318\tNA\tnontargeting_40\tNA\tNA\tNA\tNA\tNA\tNA\tNA\tTAGCACTATGTGCAACGTCA\tNon-targeting\tNA\r\n"
     ]
    }
   ],
   "source": [
    "!tail /data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/data/TG2-GITR-Hi-rep1.trim20.alignment.guide_quantifications.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "tail -n 40 /data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/data/TG2-GITR-Hi-rep1.trim20.alignment.guide_quantifications.tsv \\\n",
    "| cut -f7,15 \\\n",
    "> /data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/data/nt_guides.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "220222_GITR_FullLibrary.fa\r\n",
      "220222_TG1_GITR_InputLibrary_guideIDs_CountTable.csv\r\n",
      "220304_TG2_GITR_KRAB_Output_CountTable.txt\r\n",
      "guideScan_output.spec_score_over_0_2.noTTTTnoGGGGG.original.uniqueName.5bp_max_overlap.accessibilityScore_ranked_in_peaks.final.v2.txt\r\n",
      "TG1-GITR-gRNA-InputLibrary_S5_L001_R1_001.fastq.gz\r\n",
      "TG1-GITR-gRNA-InputLibrary.trim20.fastq\r\n",
      "TG2-GITR-Hi-rep1_S6_L001_R1_001.fastq.gz\r\n",
      "TG2-GITR-Hi-rep1.trim20.alignment.counts.txt\r\n",
      "TG2-GITR-Hi-rep2_S8_L001_R1_001.fastq.gz\r\n",
      "TG2-GITR-Hi-rep2.trim20.alignment.counts.txt\r\n",
      "TG2-GITR-Hi-rep3_S10_L001_R1_001.fastq.gz\r\n",
      "TG2-GITR-Hi-rep3.trim20.alignment.counts.txt\r\n",
      "TG2-GITR-Hi-rep4_S12_L001_R1_001.fastq.gz\r\n",
      "TG2-GITR-Hi-rep4.trim20.alignment.counts.txt\r\n",
      "TG2-GITR-Lo-rep1_S5_L001_R1_001.fastq.gz\r\n",
      "TG2-GITR-Lo-rep1.trim20.alignment.counts.txt\r\n",
      "TG2-GITR-Lo-rep2_S7_L001_R1_001.fastq.gz\r\n",
      "TG2-GITR-Lo-rep2.trim20.alignment.counts.txt\r\n",
      "TG2-GITR-Lo-rep3_S9_L001_R1_001.fastq.gz\r\n",
      "TG2-GITR-Lo-rep3.trim20.alignment.counts.txt\r\n",
      "TG2-GITR-Lo-rep4_S11_L001_R1_001.fastq.gz\r\n",
      "TG2-GITR-Lo-rep4.trim20.alignment.counts.txt\r\n"
     ]
    }
   ],
   "source": [
    "!ls /data/reddylab/Keith/collab/220131_TaniaRotation/230921_GITR_Submission/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chromosome\ttarget site start coordinate\ttarget site end coordinate\tgRNA\tcutting efficiency score\tcutting specificity score\tstrand\tofftargets sum\tofftargets summary\tannotation\tgRNA label\taccessibility_cpm\tpeak\trank_in_peak\r\n",
      "chr4\t156026280\t156026302\tGGCCTGAAGCCCAGTCTGAG\t56\t0.24014734\t+\t22\t2:1|3:21\t*\tpeak_10__66\t6.566831376149846\tpeak_10_\t1\r\n",
      "chr4\t156026211\t156026233\tTCTGCTCTACACTTCACAGA\t65\t0.23964306\t+\t25\t2:1|3:24\t*\tpeak_10__67\t6.304071908621422\tpeak_10_\t2\r\n",
      "chr4\t156026240\t156026262\tCAAACACCTCAGATGTCTGC\t56\t0.22785936\t+\t15\t2:2|3:13\t*\tpeak_10__73\t4.398388513167472\tpeak_10_\t3\r\n",
      "chr4\t156026323\t156026345\tGAGACAAGGCAAGTTGGAGC\t54\t0.30288193\t+\t22\t2:1|3:21\t*\tpeak_10__48\t4.253951801960835\tpeak_10_\t4\r\n",
      "chr4\t156026183\t156026205\tGTTAGGAGATGTCCAGAAAG\t63\t0.22126558\t+\t16\t2:2|3:14\t*\tpeak_10__74\t4.206091344460994\tpeak_10_\t5\r\n",
      "chr4\t156026534\t156026556\tACTCTCCTCCTTGCCTTACC\t37\t0.29646987\t-\t10\t2:1|3:9\t\"Tnfrsf18_ENSMUST00000103173.9_exon_1_of_5\tTnfrsf18_ENSMUST00000040274.12_exon_1_of_4\tTnfrsf18_ENSMUST00000122001.2_exon_1_of_4\"\tpeak_10__51\t3.2793636016188685\tpeak_10_\t6\r\n",
      "chr4\t156026371\t156026393\tACTCAGGAGAAGCACTATGG\t68\t0.46283314\t+\t8\t2:1|3:7\t\"Tnfrsf18_ENSMUST00000103173.9_exon_1_of_5\tTnfrsf18_ENSMUST00000040274.12_exon_1_of_4\tTnfrsf18_ENSMUST00000122001.2_exon_1_of_4\"\tpeak_10__18\t3.1816166433911577\tpeak_10_\t7\r\n",
      "chr4\t156026160\t156026182\tGGAGTTGGGAGAGGCATGTA\t41\t0.23432923\t+\t46\t2:1|3:45\t*\tpeak_10__68\t2.810349269867279\tpeak_10_\t8\r\n",
      "chr4\t156025987\t156026009\tGTGCTCTGGACTCCTCTCGC\t51\t0.39180049\t+\t6\t2:0|3:6\t*\tpeak_10__23\t2.8091785653463646\tpeak_10_\t9\r\n"
     ]
    }
   ],
   "source": [
    "!head /data/reddylab/Keith/collab/220131_TaniaRotation/230921_GITR_Submission/guideScan_output.spec_score_over_0_2.noTTTTnoGGGGG.original.uniqueName.5bp_max_overlap.accessibilityScore_ranked_in_peaks.final.v2.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "guide = pd.read_csv('/data/reddylab/Keith/collab/220131_TaniaRotation/230921_GITR_Submission/guideScan_output.spec_score_over_0_2.noTTTTnoGGGGG.original.uniqueName.5bp_max_overlap.accessibilityScore_ranked_in_peaks.final.v2.txt', sep='\\t')\n",
    "guide['name'] = guide['gRNA label']+'_'+guide['chromosome']+':'+guide['target site start coordinate'].astype(str)+'-'+guide['target site end coordinate'].astype(str)+':'+guide['strand']\n",
    "guide['hg38_coordinate'] = guide['chromosome']+':'+guide['target site start coordinate'].astype(str)+'-'+guide['target site end coordinate'].astype(str)+':'+guide['strand']\n",
    "guide_subset=guide[['name', 'hg38_coordinate', 'gRNA', 'cutting efficiency score', 'cutting specificity score', 'offtargets summary']]\n",
    "\n",
    "\n",
    "nt = pd.read_csv('/data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/data/nt_guides.txt', sep='\\t', names=['name','gRNA'])\n",
    "nt['hg38_coordinate'] ='NA'\n",
    "nt['cutting efficiency score']='NA'\n",
    "nt['cutting specificity score']='NA'\n",
    "nt['offtargets summary']='NA'\n",
    "nt = nt[['name', 'hg38_coordinate', 'gRNA', 'cutting efficiency score', 'cutting specificity score', 'offtargets summary']]\n",
    "\n",
    "guide_input = pd.concat([guide_subset, nt], axis=0)\n",
    "guide_input.to_csv('/data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/data/guide_input.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2023-10-16 10:16:20,348:eu_debug:\tConnecting to www.encodeproject.org\n",
      "2023-10-16 10:16:20,349:eu_debug:\tsubmission=False: In non-submission mode.\n",
      "2023-10-16 10:16:21,009:eu_debug:\tsubmission=True: In submission mode.\n",
      "2023-10-16 10:16:21,015:eu_debug:\t\n",
      "IN post().\n",
      "2023-10-16 10:16:21,016:eu_debug:\t<<<<<< POST file record tim-reddy:mm10-th17-crispri-GITR-low-bin-gRNAs To DCC with URL https://www.encodeproject.org/file and this payload:\n",
      "\n",
      "{\n",
      "  \"aliases\": [\n",
      "    \"tim-reddy:mm10-th17-crispri-GITR-low-bin-gRNAs\"\n",
      "  ],\n",
      "  \"award\": \"/awards/UM1HG009428/\",\n",
      "  \"dataset\": \"tim-reddy:mm10-th17-crispri-GITR-low-bin\",\n",
      "  \"file_format\": \"tsv\",\n",
      "  \"file_size\": 47763,\n",
      "  \"lab\": \"/labs/tim-reddy/\",\n",
      "  \"md5sum\": \"9f6bc06651d670d5dd0b5c4af6f229b2\",\n",
      "  \"output_type\": \"gRNAs\",\n",
      "  \"submitted_file_name\": \"/data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/data/guide_input.tsv\"\n",
      "}\n",
      "\n",
      "\n",
      "2023-10-16 10:16:21,738:eu_debug:\t{'code': 409, 'status': 'error', 'detail': \"Keys conflict: [('alias', 'tim-reddy:mm10-th17-crispri-GITR-low-bin-gRNAs'), ('alias', 'md5:9f6bc06651d670d5dd0b5c4af6f229b2')]\", '@type': ['HTTPConflict', 'Error'], 'description': 'There was a conflict when trying to complete your request.', 'title': 'Conflict'}\n",
      "2023-10-16 10:16:21,739:eu_debug:\t>>>>>>GET tim-reddy:mm10-th17-crispri-GITR-low-bin-gRNAs From DCC with URL https://www.encodeproject.org/tim-reddy:mm10-th17-crispri-GITR-low-bin-gRNAs/?format=json&datastore=database\n",
      "2023-10-16 10:16:22,356:eu_debug:\tWill not POST 'tim-reddy:mm10-th17-crispri-GITR-low-bin-gRNAs' since it already exists with aliases '['tim-reddy:mm10-th17-crispri-GITR-low-bin-gRNAs']'.\n",
      "2023-10-16 10:16:22,356:eu_debug:\t\n",
      "IN post().\n",
      "2023-10-16 10:16:22,357:eu_debug:\t<<<<<< POST file record tim-reddy:mm10-th17-crispri-GITR-high-bin-gRNAs To DCC with URL https://www.encodeproject.org/file and this payload:\n",
      "\n",
      "{\n",
      "  \"aliases\": [\n",
      "    \"tim-reddy:mm10-th17-crispri-GITR-high-bin-gRNAs\"\n",
      "  ],\n",
      "  \"award\": \"/awards/UM1HG009428/\",\n",
      "  \"dataset\": \"tim-reddy:mm10-th17-crispri-GITR-high-bin\",\n",
      "  \"file_format\": \"tsv\",\n",
      "  \"file_size\": 47763,\n",
      "  \"lab\": \"/labs/tim-reddy/\",\n",
      "  \"md5sum\": \"9f6bc06651d670d5dd0b5c4af6f229b2\",\n",
      "  \"output_type\": \"gRNAs\",\n",
      "  \"submitted_file_name\": \"/data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/data/guide_input.tsv\"\n",
      "}\n",
      "\n",
      "\n",
      "2023-10-16 10:16:23,068:eu_debug:\t{'code': 409, 'status': 'error', 'detail': \"Keys conflict: [('alias', 'md5:9f6bc06651d670d5dd0b5c4af6f229b2')]\", '@type': ['HTTPConflict', 'Error'], 'description': 'There was a conflict when trying to complete your request.', 'title': 'Conflict'}\n",
      "2023-10-16 10:16:23,069:eu_debug:\t>>>>>>GET tim-reddy:mm10-th17-crispri-GITR-high-bin-gRNAs From DCC with URL https://www.encodeproject.org/tim-reddy:mm10-th17-crispri-GITR-high-bin-gRNAs/?format=json&datastore=database\n",
      "2023-10-16 10:16:23,444:eu_debug:\tNOT FOUND\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"/data/reddylab/software/miniconda3/envs/revathy_py3/bin/eu_register.py\", line 345, in <module>\n",
      "    main()\n",
      "  File \"/data/reddylab/software/miniconda3/envs/revathy_py3/bin/eu_register.py\", line 156, in main\n",
      "    conn.post(payload, require_aliases=not no_aliases)\n",
      "  File \"/data/reddylab/software/miniconda3/envs/revathy_py3/lib/python3.5/site-packages/encode_utils/connection.py\", line 1066, in post\n",
      "    response.raise_for_status()\n",
      "  File \"/data/reddylab/software/miniconda3/envs/revathy_py3/lib/python3.5/site-packages/requests/models.py\", line 943, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 409 Client Error: Conflict for url: https://www.encodeproject.org/file\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "source /data/reddylab/software/miniconda3/bin/activate revathy_py3\n",
    "cd /data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/metadata\n",
    "eu_register.py -m prod -p file -i file_gRNA.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Submit new biosamples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2024-02-26 09:33:11,040:eu_debug:\tConnecting to www.encodeproject.org\n",
      "2024-02-26 09:33:11,068:eu_debug:\tsubmission=False: In non-submission mode.\n",
      "2024-02-26 09:33:11,727:eu_debug:\tsubmission=True: In submission mode.\n",
      "2024-02-26 09:33:11,746:eu_debug:\t\n",
      "IN post().\n",
      "2024-02-26 09:33:11,746:eu_debug:\t<<<<<< POST biosample record tim-reddy:mm10-th17-crispri-GITR-biosample-rep2 To DCC with URL https://www.encodeproject.org/biosample and this payload:\n",
      "\n",
      "{\n",
      "  \"aliases\": [\n",
      "    \"tim-reddy:mm10-th17-crispri-GITR-biosample-rep2\"\n",
      "  ],\n",
      "  \"award\": \"/awards/UM1HG009428/\",\n",
      "  \"biosample_ontology\": \"/biosample-types/primary_cell_CL_0000899/\",\n",
      "  \"donor\": \"encode:C57BL_6NJ\",\n",
      "  \"genetic_modifications\": [\n",
      "    \"tim-reddy:mm10-th17-crispri-GITR\"\n",
      "  ],\n",
      "  \"lab\": \"/labs/tim-reddy/\",\n",
      "  \"organism\": \"mouse\",\n",
      "  \"source\": \"/sources/maria-ciofani/\"\n",
      "}\n",
      "\n",
      "\n",
      "2024-02-26 09:33:12,268:eu_debug:\tSuccess.\n",
      "2024-02-26 09:33:12,279:eu_debug:\t\n",
      "IN post().\n",
      "2024-02-26 09:33:12,280:eu_debug:\t<<<<<< POST biosample record tim-reddy:mm10-th17-crispri-GITR-biosample-rep3 To DCC with URL https://www.encodeproject.org/biosample and this payload:\n",
      "\n",
      "{\n",
      "  \"aliases\": [\n",
      "    \"tim-reddy:mm10-th17-crispri-GITR-biosample-rep3\"\n",
      "  ],\n",
      "  \"award\": \"/awards/UM1HG009428/\",\n",
      "  \"biosample_ontology\": \"/biosample-types/primary_cell_CL_0000899/\",\n",
      "  \"donor\": \"encode:C57BL_6NJ\",\n",
      "  \"genetic_modifications\": [\n",
      "    \"tim-reddy:mm10-th17-crispri-GITR\"\n",
      "  ],\n",
      "  \"lab\": \"/labs/tim-reddy/\",\n",
      "  \"organism\": \"mouse\",\n",
      "  \"source\": \"/sources/maria-ciofani/\"\n",
      "}\n",
      "\n",
      "\n",
      "2024-02-26 09:33:12,722:eu_debug:\tSuccess.\n",
      "2024-02-26 09:33:12,723:eu_debug:\t\n",
      "IN post().\n",
      "2024-02-26 09:33:12,723:eu_debug:\t<<<<<< POST biosample record tim-reddy:mm10-th17-crispri-GITR-biosample-rep4 To DCC with URL https://www.encodeproject.org/biosample and this payload:\n",
      "\n",
      "{\n",
      "  \"aliases\": [\n",
      "    \"tim-reddy:mm10-th17-crispri-GITR-biosample-rep4\"\n",
      "  ],\n",
      "  \"award\": \"/awards/UM1HG009428/\",\n",
      "  \"biosample_ontology\": \"/biosample-types/primary_cell_CL_0000899/\",\n",
      "  \"donor\": \"encode:C57BL_6NJ\",\n",
      "  \"genetic_modifications\": [\n",
      "    \"tim-reddy:mm10-th17-crispri-GITR\"\n",
      "  ],\n",
      "  \"lab\": \"/labs/tim-reddy/\",\n",
      "  \"organism\": \"mouse\",\n",
      "  \"source\": \"/sources/maria-ciofani/\"\n",
      "}\n",
      "\n",
      "\n",
      "2024-02-26 09:33:13,124:eu_debug:\tSuccess.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "source /data/reddylab/software/miniconda3/bin/activate revathy_py3\n",
    "cd /data/reddylab/Revathy/collabs/Keith/ENCODE_submission/CRISPR/metadata\n",
    "eu_register.py -m prod -p biosample -i biosample_v1.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Submit elements reference "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python3 (revathy)",
   "language": "python",
   "name": "revathy_py3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
